---
title: "ELT-2 Regulated Genes"
author: "Rtpw"
date: "2/12/2020"
output:
  pdf_document: default

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Next steps

- perform GO
- figure out rankings of gonad genes, then test most upregulated in elt-2(-)
- figure out rankings of normally expressed genes, then test most upregulated in elt-2(-)
- figure out which are transcription factors, test if combo RNAi of elt-2 and those TFs reduce normally expressed intestine overexpression

# Improvements

Align RNA seq data to ce11 genome with more recent annotation.

# Libraries

```{r}
library(biomaRt)
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(GeneNetworkBuilder)
```



# Differentail Expression

Load data

```{r}
RNAcounts <- read.csv("./01_input/Table_S1_Raw_Read_Counts.csv", header=TRUE, row.names = 1)
```

This count file contains more samples than what I want to analyze. Subset the columns to just have `wt_sorted_*` and `elt2D_sorted_*`. Also select columns that correspond to ce11 genome assembly, since this is the genome used for the ChIP-seq analysis.

```{r}
cts <- RNAcounts %>% select(wt_sorted_1, wt_sorted_2, wt_sorted_3, wt_sorted_4, elt2D_sorted_1, elt2D_sorted_2, elt2D_sorted_3, elt2D_sorted_4)
head(cts)
```

make `coldata`

```{r}
coldata <- data.frame(condition = c("wt", "wt", "wt", "wt", "elt2D", "elt2D", "elt2D", "elt2D"), row.names = colnames(cts))
coldata
```

Check that column matrix and coldata match

```{r}
all(rownames(coldata) == colnames(cts))
```

Generate DESeqDataSet

```{r}
dds <- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ condition)

# add gene names
moreFeatures <- data.frame(gene_name = RNAcounts$gene_id_val, sequence_id = RNAcounts$sequence_id_list)
mcols(dds) <- DataFrame(mcols(dds), moreFeatures)
mcols(dds)
```

Tell DESeq which samples are "control" and which are "control" vs "treatment". This sets up the fold change comparison manually rather than letting the alphabetical determination of factor levels.

with this step: logfoldchange(elt2D/wt)

```{r}
dds$condition <- factor(dds$condition, levels = c("wt", "elt2D"))
```

Perform differential expression analysis

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)


# Convert res to dataframe
res.df <- as.data.frame(res)

# Export the results table
write.csv(res.df, file = "./02_DESeq2/200218_L1_wt_vs_elt2D_results.csv")

# Print results table information
head(res)
```
```{r}
summary(res)
```

Perform vst and rlog transformation of read counts

```{r}
vsd <- vst(dds)
rld <- rlog(dds)
```


# Explore Differential Expression

Determine if glh-1 is upregulated in elt-2(-)  
glh-1 = WBGene00001598  
glh-1 is a germline specific gene  
It is upregulated  

```{r}
res.df[rownames(res.df) == "WBGene00001598",]
```

```{r}
plotCounts(dds, gene = "WBGene00001598")
```

See if elt-2 is depleted   
It is depleted  

```{r}
plotCounts(dds, gene = "WBGene00001250")
```

See if pgl-1 is enriched  
pgl-1 = WBGene00003992  
Looks like it is enriched.  

```{r}
plotCounts(dds, gene = "WBGene00003992")
```

see if egl-20 (ligand of Wnt pathway) is expressed  
Also is depleted  

```{r}
plotCounts(dds, gene = "WBGene00001188")
```


Make an MA plot for all the data.  

```{r}
plotMA(res, ylim = c(-10, 10))
```

Make a heatmap of differentially expressed genes.  
Use variance of rlog transformed read counts to filter the data set for genes that are actually changing.  

```{r}
select <- rowVars(assay(rld)) > 1

rowNormalized <- assay(rld)-rowMeans(assay(rld))

pheatmap(rowNormalized[select,], cluster_cols = FALSE, cluster_rows = TRUE, show_rownames = FALSE)

```


# Determine ELT-2 regulated TFs

Load in datasets

```{r}

res.df <- read.csv(file = "./02_DESeq2/200218_L1_wt_vs_elt2D_results.csv", header = TRUE, sep = ",", row.names = 1)

res.df <- rownames_to_column(res.df, var = "WBGeneID")

wTF3.0 <- read.delim("./TF3-0_namesonly.txt", header = TRUE, sep = "\t") %>% select(Sequence.name, Public_name, WBGeneID, DBD)

```

Subset elt-2(-) differentially expressed genes for transcription factors with wTF3.0 dataset.  

```{r}
elt2_responding_TF <- merge(wTF3.0, res.df, by.x = "WBGeneID", by.y = "WBGeneID")
dim(elt2_responding_TF)
head(elt2_responding_TF)
```
Make an MA plot of differentially expressed transcription factors.  

```{r}
threshold = 0.01
ggplot(data = elt2_responding_TF) + 
  geom_point(aes(x = baseMean, y = log2FoldChange, colour = cut(padj, c(-Inf, threshold, +Inf)))) +
  scale_colour_manual(name = "padj", values = c("red", "black")
  ) +
  scale_x_log10() 
```

Separate dataset into activated and repressed TFs.  

```{r}
elt2_activated_TF <- elt2_responding_TF %>% filter(log2FoldChange <= 0, padj <= threshold)
elt2_repressed_TF <- elt2_responding_TF %>% filter(log2FoldChange >= 0, padj <= threshold)


```

Make heatmap of differentially expressed TFs.  

```{r}
nameselect <- rownames(assay(rld)) %in% wTF3.0$WBGeneID 

nameMatrix <- assay(rld)[nameselect,]

varselect <- rowVars(nameMatrix) > 0.1

namevarMatrix <- nameMatrix[varselect, ]

namevarRowNormalized <- namevarMatrix - rowMeans(namevarMatrix)

pheatmap(namevarRowNormalized, cluster_cols = FALSE, cluster_rows = TRUE, show_rownames = FALSE, border_color = NA)
```

Replace WBGeneID with gene name in row name.  

```{r}
rownames(namevarRowNormalized)

paramart <- useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

name2id = getBM(mart = paramart, 
                     filter=c("species_id_1010", 
                              "biotype"), 
                     value=list(species_id_1010="caelegprjna13758", 
                                biotype="protein_coding"), 
                     attributes = c('external_gene_id', 
                                    'wbps_gene_id'))
head(name2id)
```


# Integrate ChIP-seq data

```{r}
peaks <- read.delim("./01_input/L1_1_L1_2.IDR_0.05.narrowPeak", header = FALSE, sep = "\t")
head(peaks)

peaks_subset <- data.frame(chrom = peaks$V1, start = peaks$V2, end = peaks$V3, name = peaks$V4, score = peaks$V5, strand = peaks$V6)

write.table(peaks_subset, file = "./03_BETA/BETA_Formatted_Input/200218_L1_ELT2_ChIP_Peaks_BETA.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

# Make BETA-friendly GTF file

This annotation file corresponds to annotation version WS271.

```{r}
paramart <- useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

genes_coding = getBM(mart = paramart, 
                     filter=c("species_id_1010", 
                              "biotype"), 
                     value=list(species_id_1010="caelegprjna13758", 
                                biotype="protein_coding"), 
                     attributes = c('refseq_mrna',
                                    'chromosome_name',
                                    'strand',
                                    'start_position', 
                                    'end_position', 
                                    'wbps_gene_id'))

# exclude mitochondria
genes_coding_noMT = genes_coding[genes_coding$chromosome_name != 'MtDNA',]
# add 'chr'
genes_coding_noMT$chromosome_name = paste('chr', genes_coding_noMT$chromosome_name, sep='')
# strands must be '+/-', but paramart returns 1/-1
strand_plus_minus = genes_coding_noMT$strand
strand_plus_minus[ strand_plus_minus == '1'] <- '+'
strand_plus_minus[ strand_plus_minus == '-1'] <- '-'
genes_coding_noMT$strand = strand_plus_minus

head(genes_coding_noMT)
colnames(genes_coding_noMT) <- c("refseqID", "chrom", "strand", "start", "end", "WBGeneID")
genes_coding_noMT_reduced <- genes_coding_noMT %>% filter(refseqID %in% refseqBetaDE$refseqID)

write.table(genes_coding_noMT_reduced, file = "200227_Wormbase_WS271_Genome_Annotation.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

# Replace WBGeneID for Refseq mRNA ID

```{r}
betaDE <- read.table("./02_formatted/200218_L1_wt_vs_elt2D_results_BETA.tsv", sep = "\t")
colnames(betaDE) <- c("WBGeneID", "lfc", "P.value")

# remove rows with NA
betaDE %>% filter(lfc != "NA" , q != "NA")
write.table(betaDE, file = "./03_BETA/BETA_Formatted_Input/200227_L1_wt_vs_elt2D_results_BETA_noNA.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)

refseqBetaDE <- inner_join(betaDE, genes_coding_noMT, by = "WBGeneID") %>% select(refseqID, lfc, q) %>% filter(refseqID != "", lfc != "NA" , q != "NA")

dim(refseqBetaDE)
write.table(refseqBetaDE, file = "./03_BETA/BETA_Formatted_Input/200227_L1_wt_vs_elt2D_results_BETA_refseq.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

Use command line to run BETA. Example code below.

```{bash}
BETA basic -p 03_BETA/BETA_Formatted_Input/200218_L1_ELT2_ChIP_Peaks_BETA.tsv -e 03_BETA/BETA_Formatted_Input/200227_L1_wt_vs_elt2D_results_BETA_refseq.tsv -k BSF --gname2 -r 200227_Wormbase_WS271_Genome_Annotation.tsv -n basic -d 2000
```

Overall having a hard time executing and getting output from BETA. Going to try a different tool.

# Use GeneNetworkBuilder

this software makes plots that incorporate differential expression data (elt-2 (-) vs wt) and gene assignments of ELT-2 ChIP-seq peaks.

```{r}
peaks.df <- read.table(file = "annotated_Peaks.df", header = TRUE)
l1_peak_genes <- peaks.df %>% filter(L1_IDR == TRUE) %>% select(feature)
l1_peak_genes_GNB <- data.frame(from = "WBGene00001250", to = unique(l1_peak_genes$feature))

sifNetwork <- buildNetwork(TFbindingTable = l1_peak_genes_GNB, interactionmap = ce.interactionmap, level = 1)

cifNetwork <- filterNetwork(rootgene = "WBGene00001250", sifNetwork = sifNetwork, exprsData = betaDE, mergeBy = "WBGeneID", remove_miRNA = TRUE, tolerance = 1)

gR <- polishNetwork(cifNetwork=cifNetwork,
                  nodecolor=colorRampPalette(c("green", 
                                               "yellow", 
                                               "red"))(2594))
gR
# this is too big to make
#(html <- browseNetwork(gR, maxNodes = 2594))

```

Make a network with stricter cutoffs

```{r}
data("ce.IDsMap")
data("ce.mapIDs")
sifNetwork <- buildNetwork(TFbindingTable = l1_peak_genes_GNB, interactionmap = ce.interactionmap, level = 2)

cifNetwork <- filterNetwork(rootgene = "WBGene00001250", sifNetwork = sifNetwork, exprsData = betaDE, mergeBy = "WBGeneID", remove_miRNA = TRUE, tolerance = 1, cutoffPVal = 0.01, cutoffLFC = 2)

cifNetwork <- convertID(cifNetwork, ce.mapIDs, ByName = c("from", "to"))

gR <- polishNetwork(na.omit(cifNetwork), nodecolor = colorRampPalette(c("green", "yellow", "red"))(10))
gR

elt2network <- browseNetwork(gR, maxNodes = 528)

exportNetwork(elt2network, "./04_GeneNetworkBuilder/elt2network.html")
```

This is very messy. Subset more

```{r}
cifNetwork %>% filter(from == c("elt-2", "pqm-1", "pha-4"), logFC >= 1)

gR <- polishNetwork(cifNetwork %>% filter(from == c("elt-2", "zip-10"), logFC >= 1),
                    nodecolor = colorRampPalette(c("green", "yellow", "red"))(10))
browseNetwork(gR)

gR_pqm1 <- polishNetwork(na.omit(cifNetwork %>% filter(from == c("elt-2", "pqm-1"))),
                    nodecolor = colorRampPalette(c("green", "yellow", "red"))(10))

browseNetwork(gR_pqm1)

gR_skn1 <- polishNetwork(na.omit(cifNetwork %>% filter(from == c("elt-2", "skn-1"))),
                    nodecolor = colorRampPalette(c("green", "yellow", "red"))(10))

browseNetwork(gR_skn1)

elt2_pqm1_network <- browseNetwork(gR_pqm1)
exportNetwork(elt2_pqm1_network, "elt2_pqm1_network.html")

exportNetwork(gR_skn1, "elt2_skn1_network.html")
```

