---
title: "ELT-2 Regulated Genes"
author: "Rtpw"
date: "2/12/2020"
output:
  pdf_document: default

---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
```

# Next steps
HIGH
- perform GO on up and down regulated genes
LOW
- elt-2 chip or promoter motifs of up and down regulated genes

# Done steps
- Do Z score of row normalization, divide by the standard deviation

# Improvements

Align RNA seq data to ce11 genome with more recent annotation.

# Libraries

```{r}
library(biomaRt)
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(readxl)
library(matrixStats)
```

Source required functions.  

```{r}
source("./RWC23_Functions.R")
```



# Differentail Expression

Load data

```{r}
RNAcounts <- read.csv("./01_input/Table_S1_Raw_Read_Counts.csv", header=TRUE, row.names = 1)
```

This count file contains more samples than what I want to analyze. Subset the columns to just have `wt_sorted_*` and `elt2D_sorted_*`. Also select columns that correspond to ce11 genome assembly, since this is the genome used for the ChIP-seq analysis.

```{r}
cts <- RNAcounts %>% select(wt_sorted_1, wt_sorted_2, wt_sorted_3, wt_sorted_4, elt2D_sorted_1, elt2D_sorted_2, elt2D_sorted_3, elt2D_sorted_4, elt2Delt7D_sorted_1, elt2Delt7D_sorted_2, elt2Delt7D_sorted_3)
head(cts)
```

make `coldata`

```{r}
coldata <- data.frame(condition = c("wt", "wt", "wt", "wt", "elt2D", "elt2D", "elt2D", "elt2D", "elt2Delt7D", "elt2Delt7D", "elt2Delt7D"), row.names = colnames(cts))
coldata
```

Check that column matrix and coldata match

```{r}
all(rownames(coldata) == colnames(cts))
```

Generate DESeqDataSet

```{r}
dds <- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ condition)

# add gene names
moreFeatures <- data.frame(gene_name = RNAcounts$gene_id_val, sequence_id = RNAcounts$sequence_id_list)
mcols(dds) <- DataFrame(mcols(dds), moreFeatures)
mcols(dds)
```

Tell DESeq which samples are "control" and which are "control" vs "treatment". This sets up the fold change comparison manually rather than letting the alphabetical determination of factor levels.

with this step: logfoldchange(elt2D/wt)

```{r}
dds$condition <- factor(dds$condition, levels = c("wt", "elt2D", "elt2Delt7D"))
```

Perform differential expression analysis

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)


# Convert res to dataframe
res.df <- as.data.frame(res)

# Export the results table
#write.csv(res.df, file = "./02_DESeq2/200218_L1_wt_vs_elt2D_results.csv")

# Print results table information
head(res)
```
```{r}
summary(res)
```

Perform vst and rlog transformation of read counts

```{r}
vsd <- vst(dds)
rld <- rlog(dds)
```


# Explore Differential Expression

Determine if glh-1 is upregulated in elt-2(-)  
glh-1 = WBGene00001598  
glh-1 is a germline specific gene  
It is upregulated  

```{r}
res.df[rownames(res.df) == "WBGene00001598",]
```

```{r}
plotCounts(dds, gene = "WBGene00001598")
```

See if elt-2 is depleted   
It is depleted  

```{r}
plotCounts(dds, gene = "WBGene00001250")
```

See if pgl-1 is enriched  
pgl-1 = WBGene00003992  
Looks like it is enriched.  

```{r}
plotCounts(dds, gene = "WBGene00003992")
```

see if egl-20 (ligand of Wnt pathway) is expressed  
Also is depleted  

```{r}
plotCounts(dds, gene = "WBGene00001188")
```


Make an MA plot for all the data.  

```{r}
plotMA(res, ylim = c(-10, 10))
```

## ELT-2 Bound and Reuglated Genes



This section will integrate the L1 stage ELT-2 ChIP data analyzed by David.  

Load in data.

```{r}
elt2_peaks <- read_excel("./01_input/200406_peaksForBigBed.xlsx")
head(elt2_peaks)

names(elt2_peaks)[names(elt2_peaks)=="WBID"] <- "WBGeneID"

names(elt2_peaks)
# Subset for genes bound in the L1 stage
elt2_L1_peaks <- elt2_peaks %>%
  select(WBGeneID, L1) %>% 
  filter(L1 == 1) %>%
  select(WBGeneID) %>% unique()

wTF3.0 <- read.csv("./01_input/TF3-0_namesonly.txt", sep = "\t", header = TRUE) %>% select(WBGeneID)
```

# WT and elt-2 (-) Analysis

First focus the analysis only on genes changing between wildtype and elt-2 (-) samples only.

Use the functions in `RWC23_Functions.R` to subset and row normalize the matrix.  

```{r}

wt_elt2_counts <- subset(assay(rld), select = wt_sorted_1:elt2D_sorted_4)

elt2_bound_matrix <- matrix_select(wt_elt2_counts, elt2_L1_peaks$WBGeneID)

elt2bound_rownormMatrix <- row_normalize_matrix_cutoff(
  count_matrix = elt2_bound_matrix, 
  variance_cutoff =  0.5
  )

head(elt2bound_rownormMatrix)
```

Replace the WBGeneIDs in the row name with gene names. 

```{r}
elt2bound_rowNormGeneNameMatrix <- id2name(elt2bound_rownormMatrix)

head(elt2bound_rowNormGeneNameMatrix)
```

Now plot a heatmap of ELT-2 regulated genes.  

```{r}
myPheatmap(elt2bound_rowNormGeneNameMatrix, "ELT-2 Bound Differentially Expressed Genes\nRow Means Normalized\nVariance Cutoff = 0.5", 2)

```


Do a similar analysis for TFs only.  

```{r}

elt2_bound_TF_matrix <- matrix_select(count_matrix = elt2_bound_matrix, gene_subset_vector = wTF3.0$WBGeneID)

elt2_bound_TF_rowNorm_matrix <- row_normalize_matrix_cutoff(
  count_matrix = elt2_bound_TF_matrix, 
  variance_cutoff =  0.1
  )


elt2_bound_TF_rowNorm_matrix <- id2name(elt2_bound_TF_rowNorm_matrix)

mysmallPheatmap(elt2_bound_TF_rowNorm_matrix, "ELT-2 Bound Differentially Expressed TFs\nRow Means Normalized\nVariance Cutoff = 0.1", 2)


```

Do it for transcription factors.

```{r}
elt2_bound_TF_zscore_matrix <- row_zscore_matrix_cutoff(
  count_matrix = elt2_bound_TF_matrix, 
  variance_cutoff =  0.1
  )

elt2_bound_TF_zscore_matrix <- id2name(elt2_bound_TF_zscore_matrix)

mysmallPheatmap(elt2_bound_TF_zscore_matrix, 
           title = "ELT-2 Bound Differentially Expressed TFs\nRow Z Score Normalized\nVariance Cutoff = 0.1",
           rowspace = 2)

```

Do Z Score normalization for all genes.  

```{r}
elt2bound_zscore_matrix <- row_zscore_matrix_cutoff(
  count_matrix = elt2_bound_matrix, 
  variance_cutoff =  0.5
  )

elt2bound_zscore_matrix <- id2name(elt2bound_zscore_matrix)

myPheatmap(elt2bound_zscore_matrix,
           title = "ELT-2 Bound Differentially Expressed Genes
           Row Z Score
           Variance Cutoff = 0.5",
           rowspace = 2)

```

# Use pairwise differential expression as regulated gene filter

Load in data.  

```{r}
up_in_wt_v_elt2 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "1_up_in_wt_v_elt2", col_names = FALSE)

down_in_wt_v_elt2 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "2_down_in_wt_v_elt2", col_names = FALSE)

up_in_wt_v_elt7 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "3_up_in_wt_v_elt7", col_names = FALSE)

up_in_wt_v_elt7elt2 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "5_up_in_wt_v_elt7elt2", col_names = FALSE)

down_in_wt_v_elt7elt2 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "6_down_in_wt_v_elt7elt2", col_names = FALSE)

up_in_elt2_v_elt7elt2 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "7_up_in_elt2_v_elt7elt2", col_names = FALSE)

down_in_elt2_v_elt7elt2 <- read_excel("01_input/Table_S4_Pairwise_Diff_Expression.xlsx", 
    sheet = "8_down_in_elt2_v_elt7elt2", col_names = FALSE)

colnames(up_in_wt_v_elt2) <- c("WBGeneID")
colnames(down_in_wt_v_elt2) <- c("WBGeneID")
colnames(up_in_wt_v_elt7) <- c("WBGeneID")
colnames(up_in_wt_v_elt7elt2) <- c("WBGeneID")
colnames(down_in_wt_v_elt7elt2) <- c("WBGeneID")
colnames(up_in_elt2_v_elt7elt2) <- c("WBGeneID")
colnames(down_in_elt2_v_elt7elt2) <- c("WBGeneID")
```

Make a union of these lists with unique WBGeneIDs.  

```{r}
union_elt2elt7_DE <- data.frame(WBGeneID = c(up_in_wt_v_elt2$WBGeneID, 
                        down_in_wt_v_elt2$WBGeneID,
                        up_in_wt_v_elt7$WBGeneID,
                        up_in_wt_v_elt7elt2$WBGeneID,
                        down_in_wt_v_elt7elt2$WBGeneID,
                        up_in_elt2_v_elt7elt2$WBGeneID,
                        down_in_elt2_v_elt7elt2$WBGeneID
                        )) %>% unique()
```

Subset count matrix for presence in union of all pairwise comparisons.  

```{r}
all_pairwise_subset <- matrix_select(wt_elt2_counts, union_elt2elt7_DE$WBGeneID)

row_normalize_matrix <- function(count_matrix){
  namevarRowNormalized <- count_matrix - rowMeans(count_matrix)
  return(namevarRowNormalized)
}

all_pairwise_subset_rownorm <- row_normalize_matrix(all_pairwise_subset)

myPheatmap(all_pairwise_subset_rownorm,
           title = "Genes Significantly Differentially Expressed In All
           Pairwise Comparisons
           Row Means Normalized",
           rowspace = 4)
```

Hard to see anything useful with this.  

Do the same thing but use Z score. Maybe there will be more detail.  

```{r}
all_pairwise_subset_Zscore <- row_zscore_matrix(all_pairwise_subset)

# remove columns with NA
all_pairwise_subset_Zscore <- all_pairwise_subset_Zscore[complete.cases(all_pairwise_subset_Zscore), ]

unique(is.na(all_pairwise_subset_Zscore))

myPheatmap(all_pairwise_subset_Zscore,
           title = "Genes with Significant DE In All Pairwise Comparisons
           Row Z Score Normalized",
           rowspace = 3)
```

Clusters are a little more obvious.  

Add annotation to side of heatmap that indicates binding of ELT-2 in L1 stage.  

```{r}

my_row_anno <- data.frame(elt2_L1_bound = ifelse(test = rownames(all_pairwise_subset_Zscore) %in% elt2_L1_peaks$WBGeneID, yes = "bound", no = "not.bound"))


rownames(my_row_anno) <- rownames(all_pairwise_subset_Zscore)

ann_colors = list(
  elt2_L1_bound = c(bound = "green", not.bound = "black")
)

pheatmap(all_pairwise_subset_Zscore, 
         annotation_row = my_row_anno,
         annotation_colors = ann_colors,
           cluster_cols = FALSE, 
           cluster_rows = TRUE, 
           show_rownames = FALSE, 
           border_color = NA,
           cutree_rows = 3,
         main = "Genes with Significant DE In All Pairwise Comparisons
           Row Z Score Normalized",
         width = 6,
         height = 6)#,
         #file = "./03_plots/200406_All_DE_Genes_Elt2_Elt7_rowZscore_Bound_Annotation.pdf")


```

Add 0 or 1 binding value to matrix and use to separate bound and unbound in clustered heatmap.  

```{r}
all_pairwise_subset_Zscore_bindStatus <- cbind(all_pairwise_subset_Zscore, elt2Bound = ifelse(test = rownames(all_pairwise_subset_Zscore) %in% elt2_L1_peaks$WBGeneID, yes = 1, no = 0))

myPheatmap(all_pairwise_subset_Zscore_bindStatus,
           title = "clustered with binding",
           rowspace = 1)
```

Doesn't seem to change clustering.  

Now subset the plot for ELT-2 binding in the L1 stage.  

```{r}
elt2bound_all_pairwise_subset_Zscore <- matrix_select(all_pairwise_subset_Zscore, elt2_L1_peaks$WBGeneID)

myPheatmap(elt2bound_all_pairwise_subset_Zscore,
           title = "Differential Expression of ELT-2 Bound Genes
           Subset: Differentially expressed in all pairwise comparisons
           Z Score Normalized",
           rowspace = 3)
```

Now subset for genes that are transcription factors.  

```{r}
elt2bound_all_pairwise_subset_Zscore_TF <- matrix_select(elt2bound_all_pairwise_subset_Zscore, wTF3.0$WBGeneID)

elt2bound_all_pairwise_subset_Zscore_TF <- id2name(elt2bound_all_pairwise_subset_Zscore_TF)

pheatmap(elt2bound_all_pairwise_subset_Zscore_TF,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         border_color = NA
)
```



# WT, elt-2 (-) and elt-2(-);elt-7(-) Analysis

```{r subset-rownorm-elt2-bound}

wt_elt2_elt7double_counts <-assay(rld)

elt2_bound_matrix <- matrix_select(wt_elt2_elt7double_counts, elt2_L1_peaks$WBGeneID)

elt2bound_rownormMatrix <- row_normalize_matrix_cutoff(
  count_matrix = elt2_bound_matrix, 
  variance_cutoff =  0.5
  )

head(elt2bound_rownormMatrix)
```

Replace the WBGeneIDs in the row name with gene names. 

```{r gene-rowname}
elt2bound_rowNormGeneNameMatrix <- id2name(elt2bound_rownormMatrix)

head(elt2bound_rowNormGeneNameMatrix)
```

Now plot a heatmap of ELT-2 regulated genes.  

```{r plot-elt2-regulated-heatmap}
myPheatmap(elt2bound_rowNormGeneNameMatrix, "ELT-2 Bound Differentially Expressed Genes\nRow Means Normalized\nVariance Cutoff = 0.5", 2)

```


Do a similar analysis for TFs only.  

```{r process-elt2-regulated-tfs}

elt2_bound_TF_matrix <- matrix_select(count_matrix = elt2_bound_matrix, gene_subset_vector = wTF3.0$WBGeneID)

elt2_bound_TF_rowNorm_matrix <- row_normalize_matrix_cutoff(
  count_matrix = elt2_bound_TF_matrix, 
  variance_cutoff =  0.1
  )


elt2_bound_TF_rowNorm_matrix <- id2name(elt2_bound_TF_rowNorm_matrix)

mysmallPheatmap(elt2_bound_TF_rowNorm_matrix, "ELT-2 Bound Differentially Expressed TFs\nRow Means Normalized\nVariance Cutoff = 0.1", 2)


```

Do it for transcription factors.

```{r tf-zscore-heatmap}
elt2_bound_TF_zscore_matrix <- row_zscore_matrix_cutoff(
  count_matrix = elt2_bound_TF_matrix, 
  variance_cutoff =  0.1
  )

elt2_bound_TF_zscore_matrix <- id2name(elt2_bound_TF_zscore_matrix)

mysmallPheatmap(elt2_bound_TF_zscore_matrix, 
           title = "ELT-2 Bound Differentially Expressed TFs\nRow Z Score Normalized\nVariance Cutoff = 0.1",
           rowspace = 3)

```

Do Z Score normalization for all genes.  

```{r genes-zscore-heatmap}
elt2bound_zscore_matrix <- row_zscore_matrix_cutoff(
  count_matrix = elt2_bound_matrix, 
  variance_cutoff =  0.5
  )

elt2bound_zscore_matrix <- id2name(elt2bound_zscore_matrix)

myPheatmap(elt2bound_zscore_matrix,
           title = "ELT-2 Bound Differentially Expressed Genes
           Row Z Score
           Variance Cutoff = 0.5",
           rowspace = 2)

```

# Use pairwise differential expression as regulated gene filter

Data was loaded in the elt-2 (-) section above.  

Subset count matrix for presence in union of all pairwise comparisons.  

```{r}
all_pairwise_subset <- matrix_select(wt_elt2_elt7double_counts, union_elt2elt7_DE$WBGeneID)

row_normalize_matrix <- function(count_matrix){
  namevarRowNormalized <- count_matrix - rowMeans(count_matrix)
  return(namevarRowNormalized)
}

all_pairwise_subset_rownorm <- row_normalize_matrix(all_pairwise_subset)

myPheatmap(all_pairwise_subset_rownorm,
           title = "Genes Significantly Differentially Expressed In All
           Pairwise Comparisons
           Row Means Normalized",
           rowspace = 1)
```

Hard to see anything useful with this.  

Do the same thing but use Z score. Maybe there will be more detail.  

```{r}
all_pairwise_subset_Zscore <- row_zscore_matrix(all_pairwise_subset)

# remove columns with NA
all_pairwise_subset_Zscore <- all_pairwise_subset_Zscore[complete.cases(all_pairwise_subset_Zscore), ]

unique(is.na(all_pairwise_subset_Zscore))

myPheatmap(all_pairwise_subset_Zscore,
           title = "Genes with Significant DE In All Pairwise Comparisons
           Row Z Score Normalized",
           rowspace = 1)
```

Clusters are a little more obvious.  

Add annotation to side of heatmap that indicates binding of ELT-2 in L1 stage.  

```{r}

my_row_anno <- data.frame(elt2_L1_bound = ifelse(test = rownames(all_pairwise_subset_Zscore) %in% elt2_L1_peaks$WBGeneID, yes = "bound", no = "not.bound"))


rownames(my_row_anno) <- rownames(all_pairwise_subset_Zscore)

ann_colors = list(
  elt2_L1_bound = c(bound = "green", not.bound = "black")
)

pheatmap(all_pairwise_subset_Zscore, 
         annotation_row = my_row_anno,
         annotation_colors = ann_colors,
           cluster_cols = FALSE, 
           cluster_rows = TRUE, 
           show_rownames = FALSE, 
           border_color = NA,
           cutree_rows = 3,
         main = "Genes with Significant DE In All Pairwise Comparisons
           Row Z Score Normalized",
         width = 6,
         height = 6)#,
         #file = "./03_plots/200406_All_DE_Genes_Elt2_Elt7_rowZscore_Bound_Annotation.pdf")


```

Add 0 or 1 binding value to matrix and use to separate bound and unbound in clustered heatmap.  

```{r}
all_pairwise_subset_Zscore_bindStatus <- cbind(all_pairwise_subset_Zscore, elt2Bound = ifelse(test = rownames(all_pairwise_subset_Zscore) %in% elt2_L1_peaks$WBGeneID, yes = 1, no = 0))

myPheatmap(all_pairwise_subset_Zscore_bindStatus,
           title = "clustered with binding",
           rowspace = 1)
```

Doesn't seem to change clustering.  

Now subset the plot for ELT-2 binding in the L1 stage.  

```{r}
elt2bound_all_pairwise_subset_Zscore <- matrix_select(all_pairwise_subset_Zscore, elt2_L1_peaks$WBGeneID)

myPheatmap(elt2bound_all_pairwise_subset_Zscore,
           title = "Differential Expression of ELT-2 Bound Genes
           Subset: Differentially expressed in all pairwise comparisons
           Z Score Normalized",
           rowspace = 4)
```

Now subset for genes that are transcription factors.  

```{r}
elt2bound_all_pairwise_subset_Zscore_TF <- matrix_select(elt2bound_all_pairwise_subset_Zscore, wTF3.0$WBGeneID)

elt2bound_all_pairwise_subset_Zscore_TF <- id2name(elt2bound_all_pairwise_subset_Zscore_TF)

pheatmap(elt2bound_all_pairwise_subset_Zscore_TF,
         cluster_rows = TRUE,
         cluster_cols = FALSE,
         show_rownames = TRUE,
         border_color = NA,
         cutree_rows = 2,
         main = "Differential Expression of ELT-2 bound Transcription Factors
         Subset of Significant DE In All Pairwise Comparisons
         Row Z Score Normalization"
)
```

Add intestine expressed annotation to rows. From the project RWC19 aka TF_TEAM.

Load in data.  

```{r}
spencerLEgenes <- read.table("/Users/rtpw/Documents/12_GITHUB_REPO/TF_Team/02_Data/6_Spencer_et_al_2010_FACS_and_pulldown_tilling_array/LE-intestine_enr_vs_ref.WS200.txt", quote="\"", comment.char="", header = TRUE)
colnames(spencerLEgenes) <- str_c("spencer_LE_", colnames(spencerLEgenes))
spencer_LE_subset <- spencerLEgenes %>% select(spencer_LE_ID, spencer_LE_AveExpr, spencer_LE_adj_P_Val, spencer_LE_FC)

spencerL2genes <- read.table("/Users/rtpw/Documents/12_GITHUB_REPO/TF_Team/02_Data/6_Spencer_et_al_2010_FACS_and_pulldown_tilling_array/L2-intestine_enr_vs_ref.WS200.txt", quote="\"", comment.char="", header = TRUE)
colnames(spencerL2genes) <- str_c("spencer_L2_", colnames(spencerL2genes))
spencer_L2_subset <- spencerL2genes %>% select(spencer_L2_ID, spencer_L2_AveExpr, spencer_L2_adj_P_Val, spencer_L2_FC)
```

Add an annotation column for late embryo and larval stage 2 intestine expression.  

```{r}

bound_expressed_annotation <- cbind(my_row_anno, 
                                    LE.Intestine = ifelse(test = rownames(all_pairwise_subset_Zscore) %in% spencer_LE_subset$spencer_LE_ID, yes = "enriched", no = "depleted"),
                                    L2.Intestine = ifelse(test = rownames(all_pairwise_subset_Zscore) %in% spencer_L2_subset$spencer_L2_ID, yes = "enriched", no = "depleted"))

bound_expressed_annotation %>% head()
bound_expressed_ann_colors <- list(
  elt2_L1_bound = c(bound = "green", not.bound = "black"), 
  LE.Intestine = c(enriched = "blue", depleted = "white"), 
  L2.Intestine = c(enriched = "blue", depleted = "white")
  )


pheatmap(all_pairwise_subset_Zscore, 
         annotation_row = bound_expressed_annotation,
         annotation_colors = bound_expressed_ann_colors,
           cluster_cols = FALSE, 
           cluster_rows = TRUE, 
           show_rownames = FALSE, 
           border_color = NA,
         cutree_rows = 3,
         main = "Genes with Significant DE In All Pairwise Comparisons
           Row Z Score Normalized",
         width = 6,
         height = 6)#,
         #file = "./03_plots/200409_All_DE_Genes_Elt2_Elt7_rowZscore_Bound_Expressed_Annotation.pdf")


```

# Session info

Document session info.  

```{r}
sessionInfo()
```


