---
title: "ELT-2 Regulated Genes"
author: "Rtpw"
date: "2/12/2020"
output:
  pdf_document: default

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Next steps
HIGH
- perform GO on up and down regulated genes
LOW
- elt-2 chip or promoter motifs of up and down regulated genes

# Done steps
- Do Z score of row normalization, divide by the standard deviation

# Improvements

Align RNA seq data to ce11 genome with more recent annotation.

# Libraries

```{r}
library(biomaRt)
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(readxl)
library(matrixStats)
```

Source required functions.  

```{r}
source("./RWC23_Functions.R")
```



# Differentail Expression

Load data

```{r}
RNAcounts <- read.csv("./01_input/Table_S1_Raw_Read_Counts.csv", header=TRUE, row.names = 1)
```

This count file contains more samples than what I want to analyze. Subset the columns to just have `wt_sorted_*` and `elt2D_sorted_*`. Also select columns that correspond to ce11 genome assembly, since this is the genome used for the ChIP-seq analysis.

```{r}
cts <- RNAcounts %>% select(wt_sorted_1, wt_sorted_2, wt_sorted_3, wt_sorted_4, elt2D_sorted_1, elt2D_sorted_2, elt2D_sorted_3, elt2D_sorted_4, elt2Delt7D_sorted_1, elt2Delt7D_sorted_2, elt2Delt7D_sorted_3)
head(cts)
```

make `coldata`

```{r}
coldata <- data.frame(condition = c("wt", "wt", "wt", "wt", "elt2D", "elt2D", "elt2D", "elt2D", "elt2Delt7D", "elt2Delt7D", "elt2Delt7D"), row.names = colnames(cts))
coldata
```

Check that column matrix and coldata match

```{r}
all(rownames(coldata) == colnames(cts))
```

Generate DESeqDataSet

```{r}
dds <- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ condition)

# add gene names
moreFeatures <- data.frame(gene_name = RNAcounts$gene_id_val, sequence_id = RNAcounts$sequence_id_list)
mcols(dds) <- DataFrame(mcols(dds), moreFeatures)
mcols(dds)
```

Tell DESeq which samples are "control" and which are "control" vs "treatment". This sets up the fold change comparison manually rather than letting the alphabetical determination of factor levels.

with this step: logfoldchange(elt2D/wt)

```{r}
dds$condition <- factor(dds$condition, levels = c("wt", "elt2D", "elt2Delt7D"))
```

Perform differential expression analysis

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)


# Convert res to dataframe
res.df <- as.data.frame(res)

# Export the results table
#write.csv(res.df, file = "./02_DESeq2/200218_L1_wt_vs_elt2D_results.csv")

# Print results table information
head(res)
```
```{r}
summary(res)
```

Perform vst and rlog transformation of read counts

```{r}
vsd <- vst(dds)
rld <- rlog(dds)
```


# Explore Differential Expression

Determine if glh-1 is upregulated in elt-2(-)  
glh-1 = WBGene00001598  
glh-1 is a germline specific gene  
It is upregulated  

```{r}
res.df[rownames(res.df) == "WBGene00001598",]
```

```{r}
plotCounts(dds, gene = "WBGene00001598")
```

See if elt-2 is depleted   
It is depleted  

```{r}
plotCounts(dds, gene = "WBGene00001250")
```

See if pgl-1 is enriched  
pgl-1 = WBGene00003992  
Looks like it is enriched.  

```{r}
plotCounts(dds, gene = "WBGene00003992")
```

see if egl-20 (ligand of Wnt pathway) is expressed  
Also is depleted  

```{r}
plotCounts(dds, gene = "WBGene00001188")
```


Make an MA plot for all the data.  

```{r}
plotMA(res, ylim = c(-10, 10))
```

Make a heatmap of differentially expressed genes.  
Use variance of rlog transformed read counts to filter the data set for genes that are actually changing.  

```{r}
select <- rowVars(assay(rld)) > 0.5

rowNormalized <- assay(rld)-rowMeans(assay(rld))

pheatmap(rowNormalized[select, ], cluster_cols = FALSE, cluster_rows = TRUE, show_rownames = FALSE)

```


# Determine ELT-2 regulated TFs

Load in datasets

```{r}

res.df <- read.csv(file = "./02_DESeq2/200218_L1_wt_vs_elt2D_results.csv", header = TRUE, sep = ",", row.names = 1)

res.df <- rownames_to_column(res.df, var = "WBGeneID")

wTF3.0 <- read.delim("./TF3-0_namesonly.txt", header = TRUE, sep = "\t") %>% select(Sequence.name, Public_name, WBGeneID, DBD)

```

Subset elt-2(-) differentially expressed genes for transcription factors with wTF3.0 dataset.  

```{r}
elt2_responding_TF <- merge(wTF3.0, res.df, by.x = "WBGeneID", by.y = "WBGeneID")
dim(elt2_responding_TF)
head(elt2_responding_TF)
```
Make an MA plot of differentially expressed transcription factors.  

```{r}
threshold = 0.01
ggplot(data = elt2_responding_TF) + 
  geom_point(aes(x = baseMean, y = log2FoldChange, colour = cut(padj, c(-Inf, threshold, +Inf)))) +
  scale_colour_manual(name = "padj", values = c("red", "black")
  ) +
  scale_x_log10() 
```

Separate dataset into activated and repressed TFs.  

```{r}
elt2_activated_TF <- elt2_responding_TF %>% filter(log2FoldChange <= 0, padj <= threshold)
elt2_repressed_TF <- elt2_responding_TF %>% filter(log2FoldChange >= 0, padj <= threshold)
```

Make heatmap of differentially expressed TFs.  

```{r}
nameselect <- rownames(assay(rld)) %in% wTF3.0$WBGeneID 

nameMatrix <- assay(rld)[nameselect,]

varselect <- rowVars(nameMatrix) > 0.1

namevarMatrix <- nameMatrix[varselect, ]

namevarRowNormalized <- namevarMatrix - rowMeans(namevarMatrix)

pheatmap(namevarRowNormalized, cluster_cols = FALSE, cluster_rows = TRUE, show_rownames = FALSE, border_color = NA)
```

Replace WBGeneID with gene name in row name.  

```{r}
rownames(namevarRowNormalized)

paramart <- useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

name2id = getBM(mart = paramart, 
                     filter=c("species_id_1010", 
                              "biotype"), 
                     value=list(species_id_1010="caelegprjna13758", 
                                biotype="protein_coding"), 
                     attributes = c('external_gene_id', 
                                    'wbps_gene_id'))
head(name2id)

```


# ELT-2 Bound and Reuglated Genes

This section will integrate the L1 stage ELT-2 ChIP data analyzed by David.  

Load in data.

```{r}
elt2_peaks <- read_excel("./01_input/200331_peaksForBigBed.xlsx")

# Subset for genes bound in the L1 stage
elt2_L1_peaks <- elt2_peaks %>% select(mapped_gene, L1) %>% filter(L1 == 1) %>% select(mapped_gene)

```

Now subset the row normalized set expression set for these genes.

<!--
Make a function to subset and process the matrix.  
These functions now live in 

```{r make-matrix-select-function}

matrix_select <- function(count_matrix, gene_subset_vector){
  nameselect <- rownames(count_matrix) %in% gene_subset_vector
  nameMatrix <- count_matrix[nameselect,]
  return(nameMatrix)
}

row_normalize_matrix <- function(count_matrix, variance_cutoff){
  varselect <- rowVars(count_matrix) >= variance_cutoff
  count_matrix <- count_matrix[varselect, ]
  namevarRowNormalized <- count_matrix - rowMeans(count_matrix)
  return(namevarRowNormalized)
  #head(nameMatrix)
}
```
-->

Use the functions to subset and row normalize the matrix.  

```{r subset-rownorm-elt2-bound}
elt2_bound_matrix <- matrix_select(assay(rld), elt2_L1_peaks$mapped_gene)

elt2bound_rownormMatrix <- row_normalize_matrix(
  count_matrix = elt2_bound_matrix, 
  variance_cutoff =  0.5
  )

head(elt2bound_rownormMatrix)
```


<!--
Make a function to replace rowname WBGeneID with Gene Name.  
This function now lives in RWC23_Functions.R
```{r make-id2name-function}

id2name <- function(count_matrix){
  id2name.df <- suppressMessages(biomaRt::getBM(mart = paramart, 
               filter=c("wbps_gene_id"), 
               value=rownames(count_matrix), 
                          attributes = c('wbps_gene_id','external_gene_id')
                          ))
  rownames(count_matrix) <- id2name.df$external_gene_id[ 
    match(rownames(count_matrix), id2name.df$wbps_gene_id)
    ]
  return(count_matrix)
}
```
-->

Replace the WBGeneIDs in the row name with gene names. 

```{r gene-rowname}
elt2bound_rowNormGeneNameMatrix <- id2name(elt2bound_rownormMatrix)

head(elt2bound_rowNormGeneNameMatrix)
```

Now plot a heatmap of ELT-2 regulated genes.  

```{r plot-elt2-regulated-heatmap}
pheatmap(elt2bound_rowNormGeneNameMatrix, 
         cluster_cols = FALSE, 
         cluster_rows = TRUE, 
         show_rownames = FALSE, 
         border_color = NA)#,
         #cellheight = 10,
         #filename = "./03_plots/200331_L1_Elt2_Elt7_Bound_Regulated_Genes_Row_Normalized_NoNames.pdf")

```


Do a similar analysis for TFs only.  

```{r process-elt2-regulated-tfs}

elt2_bound_TF_matrix <- matrix_select(count_matrix = elt2_bound_matrix, gene_subset_vector = wTF3.0$WBGeneID)

elt2_bound_TF_rowNorm_matrix <- row_normalize_matrix(
  count_matrix = elt2_bound_TF_matrix, 
  variance_cutoff =  0.1
  )


elt2_bound_TF_rowNorm_matrix <- id2name(elt2_bound_TF_rowNorm_matrix)

pheatmap(elt2_bound_TF_rowNorm_matrix, 
         cluster_cols = FALSE, 
         cluster_rows = TRUE, 
         show_rownames = TRUE, 
         border_color = NA)#,
         #cellheight = 10,
         #filename = "./03_plots/200331_L1_Elt2_Elt7_Bound_Regulated_TFs_Row_Normalized_Heatmap.pdf")
```

<!--
Now try Z scores.
This function now lives in RWC23_Functions.R  

```{r zscore-function}

row_zscore_matrix <- function(count_matrix, variance_cutoff){
  varselect <- rowVars(count_matrix) >= variance_cutoff
  count_matrix <- count_matrix[varselect, ]
  namevarZscore <- (count_matrix - rowMeans(count_matrix))/rowSds(count_matrix)
  return(namevarZscore)
 
}
```
-->
Do it for transcription factors.

```{r tf-zscore-heatmap}
elt2_bound_TF_zscore_matrix <- row_zscore_matrix(
  count_matrix = elt2_bound_TF_matrix, 
  variance_cutoff =  0.1
  )

elt2_bound_TF_zscore_matrix <- id2name(elt2_bound_TF_zscore_matrix)

pheatmap(elt2_bound_TF_zscore_matrix, 
         cluster_cols = FALSE, 
         cluster_rows = TRUE, 
         show_rownames = TRUE, 
         border_color = NA)#,
         #cellheight = 10,
         #filename = "./03_plots/200328_L1_Elt2_Bound_Regulated_TFs_Zscore_Heatmap.pdf")
```

Do it for all genes.  

```{r genes-zscore-heatmap}
elt2bound_zscore_matrix <- row_zscore_matrix(
  count_matrix = elt2_bound_matrix, 
  variance_cutoff =  0.5
  )

elt2bound_zscore_matrix <- id2name(elt2bound_zscore_matrix)

pheatmap(elt2bound_zscore_matrix, 
         cluster_cols = FALSE, 
         cluster_rows = TRUE, 
         show_rownames = FALSE, 
         border_color = NA)#,
         #cellheight = 10,
         #filename = "./03_plots/200327_L1_Elt2_Bound_Regulated_Genes_Zscore_Heatmap.pdf")
```

# Session info

Document session info.  

```{r}
sessionInfo()
```


