---
title: "RWC23_ELT2_Regulated_Genes"
author: "RTPW"
date: "4/13/2020"
output: 
  pdf_document: default
  fig_crop: no
  
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig_crop = FALSE)
```

# Install Packages

```{r}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install()
# BiocManager::install("biomaRt")
# install.packages("tidyverse")
# install.packages("readxl")
# BiocManager::install("ComplexHeatmap")
# install.packages("matrixStats")
# install.packages("pheatmap")
# install.packages("RVAideMemoire")
# install.packages("dendextend")
# install.packages("binom")
```


# Load Package Libraries

```{r, message=FALSE}
library(biomaRt)
library(tidyverse)
library(readxl)
library(ComplexHeatmap)
library(matrixStats)
library(pheatmap)
library(RVAideMemoire)
library(dendextend)
library(binom)
library(circlize)

```

## Background and Rationale

ELT-2 is the C. elegans intestine master regulator. Deletion of ELT-2 leads to a larval lethal phenotype, and expression of ELT-2 in non-intestine tissue induces an intestine fate.

This documet will generate plots to address the questions outlined below.

For genes differentially expressed during elt-2 (-) and/or elt-7(-):

1) which expression pattern clusters associate with ELT-2 binding?
2) which expression pattern clusters associate with ELT-2 binding categories?
  a) For all genes
  b) For only genes bound by ELT-2
3) Which expression pattern clusters associate with intestine expression? (MA plot for each expression set)
  a) For all genes
  b) For genes only bound by ELT-2

For clusters of transcription factors (TFs) differentially expressed during elt-2 (-) and/or elt-7(-):

1) which transcription factor clusters associate with ELT-2 binding?
2) which transcription factor clusters associate with ELT-2 binding categories
  a) for all TFs
  b) For only TFs bound by ELT-2
3) which transcription factor clusters associate with intestine expression?
  a) for all
  b) for only ELT-2 bound

## Data

I will integrate two RNA-seq experiments and a ChIP-seq experiments.

The first is a set of RNA-seq experiments in L1 stage worms (Dineen and Nishimura, 2018). They were collected from the following genotypes, all in the L1 stage: 

- wildtype (wt)
- elt-7 deleted (elt7D)
- elt-2 deleted (elt2D)
- combination fo elt-7 and elt-2 deleted (elt2Delt7D)

The purpose of including elt-7 and elt-2/elt-7 double deletion is because these two transcription factors have overlapping functionality. Deletion of elt-7 alone does not have a phenotype, but deletion of elt-7 in combination with elt-2 has an enhanced lethal phenotype of just elt-2 alone.

The second RNA-seq experiment is from FACS sorted L1 stage intestine cells. This data is unpublished.  

The ChIP-seq experiments are performed against ELT-2 and are from the following developmental stages:

- late embryo (LE)
- L1
- L3

They were collected as part of the modENCODE consortium and were processed by David King. He has provided gene mapping of ELT-2 targets and categories of ELT-2 binding. The ELT-2 binding categories are as follows:

- Not changing
- Larval
- L3 high
- Embryonic
- Increasing

## Citations

1) Dineen, A., Osborne Nishimura, E., Goszczynski, B., Rothman, J. H., & McGhee, J. D. (2018). Quantitating transcription factor redundancy: The relative roles of the ELT-2 and ELT-7 GATA factors in the C. elegans endoderm. Developmental Biology, 435(2), 150–161. https://doi.org/10.1016/J.YDBIO.2017.12.023

2) Kudron, M. M., Victorsen, A., Gevirtzman, L., Hillier, L. W., Fisher, W. W., Vafeados, D., … Waterston, R. H. (2018). The modern resource: genome-wide binding profiles for hundreds of Drosophila and Caenorhabditis elegans transcription factors. Genetics, 208(3), 937–949. https://doi.org/10.1534/genetics.117.300657

## Code

# Source functions

```{r}
source("./RWC23_Functions.R")
```

## Load and Process Datasets

# Load Dineen and Osborne Nishimura et. al. Data


```{r}
dineen_nishimura_counts <-
  read_xlsx(path = "./01_input/Table_S2_rlog_Stabilized_Read_Counts.xlsx",
            sheet = "Sheet1")

dineen_nishimura_counts_matrix <- dineen_nishimura_counts %>%
  column_to_rownames(var = "WBGeneID")  %>%
  data.matrix()

dineen_nishimura_counts_matrix %>% head
```

list of all dynamically expressed genes

```{r}
dynamic_regulated_genes <-
  read.table(file = "./05_fromErin/2017-11-20_all_changing_genes_0.1alpha_0.8lfc.txt",
             quote = "",
             header = FALSE)
colnames(dynamic_regulated_genes) <- "WBGeneID"

dynamic_regulated_genes %>% head
```


```{r}
dineen_nishimura_clusters <-
  read_xlsx(path = "./01_input/Table_S6_All_Dynamically_Expressed_Genes_Clusters.xlsx",
            sheet = "dataset")

dineen_nishimura_clusters %>% select(WBGeneID, set) %>% head
```

# Load ELT-2 ChIP-seq binding annotations

```{r}
elt2_peaks <-
  read_xlsx("./01_input/200410_peaksForBigBed.xlsx", sheet = "full cluster assignment")

colnames(elt2_peaks) <-
  c(
    "chrom",
    "start",
    "end",
    "peak.name",
    "WBGeneID",
    "mapping",
    "cluster",
    "cluster.description",
    "kweight",
    "LE",
    "L1",
    "L3",
    "peak.summit.agreement"
  )

elt2_peaks$cluster.description <-
  factor(
    elt2_peaks$cluster.description,
    levels = c(
      "Not-changing or not IDR-passing",
      "L3-high",
      "Increasing",
      "Post-embryonic",
      "LE-specific"
    ),
    labels = c(
      "NotChanging",
      "L3high",
      "Increasing",
      "PostEmbryonic",
      "LEspecific"
    )
  )

elt2_peaks %>% head
```

Make a set of genes with ELT-2 binding detected in the L1 stage.  

```{r}


elt2_detected_in_L1 <-
  elt2_peaks %>% select(WBGeneID, L1) %>% filter(L1 == 1) %>% select(WBGeneID) %>% unique()

elt2_detected_in_L1 %>% head
elt2_detected_in_L1 %>% dim

```
Make a dataframe that records the number of peaks per gene that fall in a particular binding catagory.

```{r}
binding_cluster_gene_counts <-
  table(elt2_peaks$WBGeneID, elt2_peaks$cluster.description)
binding_cluster_gene_counts <-
  as.data.frame.matrix(binding_cluster_gene_counts)
binding_cluster_gene_counts %>% head()
```


# Load Spencer et. al. intestine expression

```{r}
spencerLEgenes <-
  read.table(
    "../TF_Team/02_Data/6_Spencer_et_al_2010_FACS_and_pulldown_tilling_array/LE-intestine_enr_vs_ref.WS200.txt",
    quote = "\"",
    comment.char = "",
    header = TRUE
  )
colnames(spencerLEgenes) <-
  str_c("spencer_LE_", colnames(spencerLEgenes))
spencer_LE_subset <-
  spencerLEgenes %>% select(spencer_LE_ID,
                            spencer_LE_AveExpr,
                            spencer_LE_adj_P_Val,
                            spencer_LE_FC)

spencer_LE_subset %>% head
```

```{r}
spencerL2genes <-
  read.table(
    "../TF_Team/02_Data/6_Spencer_et_al_2010_FACS_and_pulldown_tilling_array/L2-intestine_enr_vs_ref.WS200.txt",
    quote = "\"",
    comment.char = "",
    header = TRUE
  )
colnames(spencerL2genes) <-
  str_c("spencer_L2_", colnames(spencerL2genes))
spencer_L2_subset <- spencerL2genes %>%
  select(spencer_L2_ID,
         spencer_L2_AveExpr,
         spencer_L2_adj_P_Val,
         spencer_L2_FC)

spencer_L2_subset %>% head
```

## Process rlog counts

Subset rlog matrix based on presence in list `2017-11-20_all_changing_genes_0.1alpha_0.8lfc.txt`. Row scale and center the rlog counts per genes.  

```{r}
dynamic_counts_matrix <-
  matrix_select(dineen_nishimura_counts_matrix,
                dynamic_regulated_genes$WBGeneID)

dynamic_counts_matrix_scaled <-
  t(apply(unlist(dynamic_counts_matrix), 1, scale))

rownames(dynamic_counts_matrix_scaled) <-
  rownames(dynamic_counts_matrix)
colnames(dynamic_counts_matrix_scaled) <-
  colnames(dynamic_counts_matrix)
dynamic_counts_matrix_scaled %>% head
```

Recreate Supplementary Figure S4a from Dineen and Nishimura et al.    

```{r}


Ha <- Heatmap(
  dynamic_counts_matrix_scaled,
  name = "elt2D-elt7D\nRNAseq",
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = FALSE,
  show_column_names = TRUE,
  row_names_gp = gpar(cex = 0.2),
  column_names_gp = gpar(cex = 0.4),
  heatmap_legend_param = list(color_bar = "continuous"),
  split = 6,
  column_km = 4,
  bottom_annotation = HeatmapAnnotation(foo = anno_block(
    labels = c("wt", "elt7D", "elt2D", "elt7Delt2D"),
    gp = gpar(border = NA, lty = "blank")
  )),
  left_annotation = rowAnnotation(foo = anno_block(
    labels = c("SET1", "SET2", "SET3", "SET4", "SET5", "SET6"),
    labels_rot = 0,
    gp = gpar(border = NA, lty = "blank")
  ))
)
Ha
```



They don't really look the same. Maybe I would try doing these things to figure out how they are different. 
- Correlation of scaled values between the two methods
- plot the row means from each analysis against eachother before and after scaling
- plot the row standard deviation from each analysis per gene  

Rotate the dendrogram to match Erin's order.  

```{r}
c <- cor(t(dynamic_counts_matrix_scaled), method = "spearman")
d <- as.dist(1 - c)
rowdend <- as.dendrogram(hclust(d))

SET1 = 1:276
SET2 = 277:669
SET3 = 670:1906
SET4 = 1907:2010
SET5 = 2011:2076
SET6 = 2077:3092

rowdend_rotate <- rotate(rowdend, c(SET1,
                                    SET3,
                                    SET2,
                                    SET4,
                                    SET5,
                                    SET6))

Ha_rotate <- Heatmap(
  dynamic_counts_matrix_scaled,
  name = "elt2D-elt7D\nRNAseq",
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  column_split = factor(
    c(
      rep("wt", 4),
      rep("elt7D", 3),
      rep("elt2D", 4),
      rep("elt7Delt2D", 3)
    ),
    levels = c("wt", "elt7D", "elt2D", "elt7Delt2D")
  ),
  cluster_rows = rowdend_rotate,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = FALSE,
  show_column_names = TRUE,
  row_names_gp = gpar(cex = 0.2),
  column_names_gp = gpar(cex = 0.4),
  heatmap_legend_param = list(color_bar = "continuous"),
  split = 6,
  bottom_annotation = HeatmapAnnotation(foo = anno_block(
    labels = c("wt", "elt7D", "elt2D", "elt7Delt2D"),
    gp = gpar(border = NA, lty = "blank")
  )),
  left_annotation = rowAnnotation(foo = anno_block(
    labels = c("SET1", "SET2", "SET3", "SET4", "SET5", "SET6"),
    labels_rot = 0,
    gp = gpar(border = NA, lty = "blank")
  ))
)

Ha_rotate
```


# Extract cluster assignments.  

```{r}
gene_expression_clusters <-
  assign_cluster_from_heatmap(mat = dynamic_counts_matrix_scaled, plot = Ha_rotate)
nrow(gene_expression_clusters) == nrow(dynamic_counts_matrix_scaled)
head(gene_expression_clusters)
```

TODO: Make sure cluster assignments match Erin's

```{r}
dineen_nishimura_sets <-
  dineen_nishimura_clusters %>% select(WBGeneID, set)
```

# Add ELT-2 pattern row annotation

In ComplexHeatmap the row order of input matrix and annotation df must be identical to accurately plot data.  

```{r}
elt2_detected_in_L1 %>% dim

elt2_L1_anno <-
  data.frame(
    elt2_detected_in_L1 = ifelse(
      test = rownames(dynamic_counts_matrix_scaled) %in% elt2_detected_in_L1$WBGeneID,
      yes = "bound",
      no = "not.bound"
    )
  )

#spot check that order is preserved. bound/TRUE and not.bound/FALSE should appear
valz = 100
elt2_L1_anno[valz, 1]
rownames(dynamic_counts_matrix_scaled)[valz] %in%
  elt2_detected_in_L1$WBGeneID

elt2_L1_anno <-
  elt2_L1_anno %>% mutate(WBGeneID = rownames(dynamic_counts_matrix_scaled))

elt2_L1_anno %>% dim()
elt2_L1_anno %>% class()
elt2_L1_anno %>% head()
```

Incorporate this into a heatmap annotation

```{r}
Ha_L1chip <-
  Ha_rotate + rowAnnotation(L1bound = elt2_L1_anno$elt2_detected_in_L1,
                            col = list(L1bound = c(
                              "bound" = "green", "not.bound" = "black"
                            )))
Ha_L1chip

# pdf("./03_plots/200428_wt_elt2singledouble_L1elt2bound_heatmap.pdf", height = 5, width = 5)
# Ha_L1chip
# dev.off()
```

Determine enrichment of ELT-2 binding during L1 stage.  

```{r}
expression_L1_binding <-
  merge(elt2_L1_anno, gene_expression_clusters, by = "WBGeneID")
expression_L1_binding %>% head
clust_L1bound_counts <-
  table(
    expression_L1_binding$Expression.Cluster,
    expression_L1_binding$elt2_detected_in_L1
  )

clust_L1bound_prop <- prop.table(clust_L1bound_counts, 1)

clust_L1bound_prop_ggplot <- as.data.frame(clust_L1bound_prop)

colnames(clust_L1bound_prop_ggplot) <- c("SET", "Status", "Freq")

clust_L1bound_prop_ggplot$Status <-
  factor(clust_L1bound_prop_ggplot$Status,
         levels = c("not.bound", "bound"))

clust_L1bound_prop_ggplot$SET <-
  factor(
    clust_L1bound_prop_ggplot$SET,
    levels = c("SET6", "SET5", "SET4", "SET3", "SET2", "SET1")
  )

clust_L1bound_colors <- c("bound" = "green", "not.bound" = "black")

l1bound_percents <-
  ggplot(
    clust_L1bound_prop_ggplot %>% filter(Status == "bound"),
    aes(
      x = SET,
      y = Freq,
      fill = Status,
      order = Status
    )
  ) +
  geom_bar(stat = "identity") +
  scale_color_manual(values = clust_L1bound_colors,
                     aesthetics = c("color", "fill")) +
  ggtitle("Percentage of L1 Stage ELT-2 Binding Per
          Expression Set") +
  xlab("Expression Set") +
  ylab("Percentage") +
  theme_classic() +
  theme(axis.text.x = element_text(size = 13)) +
  ylim(0, 1)

l1bound_percents

# ggsave("./03_plots/200428_proportion_of_l1elt2_per_expression_cluster.pdf", height = 2, width = 5)
```

Plot the number of "bound" vs "not.bound" per cluster

```{r}
clust_L1bound_counts %>% head
clust_L1bound_counts_ggplot <- as.data.frame(clust_L1bound_counts)
colnames(clust_L1bound_counts_ggplot) <- c("SET", "Status", "Freq")
clust_L1bound_counts_ggplot

ggplot(clust_L1bound_counts_ggplot,
       aes(x = SET,
           y = Freq,
           fill = Status)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_color_manual(values = clust_L1bound_colors,
                     aesthetics = c("color", "fill")) +
  ggtitle("Number of L1 Stage ELT-2 Binding Site Per
          Expression Set") +
  xlab("Expression Set") +
  ylab("Count") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 13))

# ggsave("./03_plots/200428_number_of_l1elt2_per_expression_cluster.pdf", height = 2, width = 5)
```


Use hypergeometric to determine enrichment of L1 stage binding per expression cluster.  

Does any set have more ELT-2 bound relative to all the sets?

```{r}
x <- 1
clust_l1bound_dhyper <- data.frame()
for (i in rownames(clust_L1bound_counts)) {
  pval <- phyper(
    q = clust_L1bound_counts[x, 1],
    m = clust_L1bound_counts[x, 1] + clust_L1bound_counts[x, 2],
    k = colSums(clust_L1bound_counts)[1],
    n = colSums(clust_L1bound_counts)[1] + colSums(clust_L1bound_counts)[2]
  )
  toappend <- data.frame(SET = i, pval = (1 - pval))
  clust_l1bound_dhyper <- bind_rows(clust_l1bound_dhyper, toappend)
  x <- x + 1
}
clust_l1bound_dhyper
clust_l1bound_dhyper %>% mutate(bool = pval < 0.05)
```

Use binomial to determine enrichment of L1 stage binding per expression cluster.

```{r}
xbinom = clust_L1bound_counts[1, 1]
nbinom = clust_L1bound_counts[1, 1] + clust_L1bound_counts[1, 2]
binom.bayes(xbinom, nbinom)
binom.bayes.densityplot(binom.bayes(xbinom, nbinom))
# any percentage outside this confidence interval is "significant"
```

Use the binomial test to determine if the different expression clusters are enriched or depleted for ELT-2 binding.  

Use `binom.test` and first two a two-tailed test.  

First calculate the proportion of bound genes over the total number of genes in the analysis.  

```{r}
proportion = as.numeric(colSums(clust_L1bound_counts)[1]) /
  as.numeric(colSums(clust_L1bound_counts)[1] + colSums(clust_L1bound_counts)[2])
proportion
```

Use custom function `ctable_binom()` to calculate p-vaule for each set.

```{r}
l1bound_binom <- ctable_binom(clust_L1bound_counts, "two.sided")
```


This says that all sets but SET5 have a significant difference in genes bound compared to the entire dataset.  

Now use the `less` or `greater` argument of `binom.test` to see if there is more or less binding.  

First two `less`

```{r}
ctable_binom(ctable = clust_L1bound_counts, alt = "less")
```

This says that set 1 and 2 have less ELT-2 binding compared to the entire dataset.  

Now try `greater`.  

```{r}
ctable_binom(clust_L1bound_counts, "greater")
```

Draw line on percentage plots to indicate background percentage of L1 binding.  

```{r}
l1bound_percents +
  geom_hline(yintercept = proportion,
             color = "red",
             linetype = "dashed") +
  geom_errorbar(
    ymax = l1bound_binom$conf.upper,
    ymin = l1bound_binom$conf.lower,
    width = 0.25
  ) +
  coord_flip()

ggsave(
  "./03_plots/200504_percentage_l1bound_per_expression_cluster.pdf",
  width = 4,
  height = 5
)
```


Use the hypergeometric test to determine: Are changing genes (all sets) enriched for L1 binding?

```{r}
N <- 20470
k <- nrow(elt2_detected_in_L1)
x3 <- as.numeric(colSums(clust_L1bound_counts)[1])
m <-
  as.numeric(colSums(clust_L1bound_counts)[1] + colSums(clust_L1bound_counts)[2])
dhyper(x3, m, N, k)

```

Compute pairwise fisher's exact tests

```{r}
fisher.multcomp(clust_L1bound_counts, p.method = "bonferroni")
```

```{r}
fisher.multcomp(clust_L1bound_counts, p.method = "bonferroni")$p.value < 0.05
```


# Row annotation of ELT-2 Binding Pattern Clusters

```{r}
dynamic_counts_matrix_scaled %>% dim

chip_annotation <- dynamic_counts_matrix_scaled %>%
  as.data.frame.matrix() %>%
  rownames_to_column() %>%
  left_join(rownames_to_column(binding_cluster_gene_counts), by = "rowname") %>%
  select(rowname,
         NotChanging,
         L3high,
         Increasing,
         PostEmbryonic,
         LEspecific) %>%
  replace_na(list(
    NotChanging = 0,
    L3high = 0,
    Increasing = 0,
    PostEmbryonic = 0,
    LEspecific = 0
  ))

unique(rownames(dynamic_counts_matrix_scaled) == chip_annotation$rowname)

nrow(dynamic_counts_matrix_scaled) == nrow(chip_annotation)
```


```{r}
Ha_L1chip_bindcluster <- Ha_L1chip +
  rowAnnotation(NotChanging = chip_annotation$NotChanging) +
  rowAnnotation(LESpecific = chip_annotation$LEspecific) +
  rowAnnotation(Increasing = chip_annotation$Increasing) +
  rowAnnotation(PostEmbryonic = chip_annotation$PostEmbryonic) +
  rowAnnotation(L3high = chip_annotation$L3high)
```

Have the colors match plot from David.
```{r}
cluster_colors <-
  data.frame(
    class = c(
      "LEspecific",
      "PostEmbryonic",
      "Increasing",
      "L3high",
      "NotChanging"
    ),
    val = c("#7570B3", "#1B9E77", "#E7298A", "#D95F02", "grey")
  )
cluster_colors$class <-
  factor(
    x = cluster_colors$class,
    levels = c(
      "LEspecific",
      "PostEmbryonic",
      "Increasing",
      "L3high",
      "NotChanging"
    )
  )
```


Convert ChIP binding clusters to a present/absence list.  


```{r}
chip_annotation_present_absent <- chip_annotation %>%
  mutate(
    NotChanging = if_else(
      condition = chip_annotation$NotChanging == 0,
      true = "absent",
      false = "present"
    )
  ) %>%
  mutate(L3high = if_else(
    condition = chip_annotation$L3high == 0,
    true = "absent",
    false = "present"
  )) %>%
  mutate(
    Increasing = if_else(
      condition = chip_annotation$Increasing == 0,
      true = "absent",
      false = "present"
    )
  ) %>%
  mutate(
    PostEmbryonic = if_else(
      condition = chip_annotation$PostEmbryonic == 0,
      true = "absent",
      false = "present"
    )
  ) %>%
  mutate(
    LEspecific = if_else(
      condition = chip_annotation$LEspecific == 0,
      true = "absent",
      false = "present"
    )
  )
```

Plot the heatmap with presence/absence.

```{r}
Ha_L1chip_clusterchip <- Ha_L1chip +
  rowAnnotation(
    NotChanging = chip_annotation_present_absent$NotChanging,
    col = list(NotChanging = c(
      "absent" = "white", "present" = "grey"
    )),
    border = TRUE
  ) +
  rowAnnotation(
    Embryonic = chip_annotation_present_absent$LEspecific,
    col = list(Embryonic = c(
      "absent" = "white", "present" = "#7570B3"
    )),
    border = TRUE
  ) +
  rowAnnotation(
    Increasing = chip_annotation_present_absent$Increasing,
    col = list(Increasing = c(
      "absent" = "white", "present" = "#E7298A"
    )),
    border = TRUE
  ) +
  rowAnnotation(
    Larval = chip_annotation_present_absent$PostEmbryonic,
    col = list(Larval = c(
      "absent" = "white", "present" = "#1B9E77"
    )),
    border = TRUE
  ) +
  rowAnnotation(L3high = chip_annotation_present_absent$L3high,
                col = list(L3high = c(
                  "absent" = "white", "present" = "#D95F02"
                )),
                border = TRUE)

Ha_L1chip_clusterchip


# pdf("./03_plots/200504_wt_elt2singledouble_L1elt2bound_elt2bindclusters_heatmap.pdf", height = 5, width = 5)
# Ha_L1chip_clusterchip
# dev.off()
```

Plot percentage of expression cluster group having binding pattern assignment.  

```{r}
exprclust_bindclust <-
  merge(
    gene_expression_clusters,
    chip_annotation_present_absent,
    by.x = "WBGeneID",
    by.y = "rowname"
  )

exprclust_bindclust %>% head
```

# What is the percentage of genes with annotated ELT2 binding clusters per expression dataset?

Make a dataframe that addresses the question: 

```{r}
elt2_cluster_names <-
  c("NotChanging",
    "L3high",
    "Increasing",
    "PostEmbryonic",
    "LEspecific")

expressionSet_per_BindingCluster <- data.frame()
for (i in elt2_cluster_names) {
  toappend <-
    table(exprclust_bindclust$Expression.Cluster,
          exprclust_bindclust[[i]]) %>%
    as.data.frame.matrix() %>%
    rownames_to_column(var = "Expression.Cluster") %>%
    mutate(ELT2_cluster = i,
           percent = present / (present + absent))
  expressionSet_per_BindingCluster <-
    bind_rows(expressionSet_per_BindingCluster, toappend)
}
expressionSet_per_BindingCluster
```


Make a plot that addresses the question: 

```{r}
ggplot(
  expressionSet_per_BindingCluster,
  aes(x = Expression.Cluster, y = present, fill = ELT2_cluster)
) +
  geom_bar(stat = "identity", position = "fill") +
  theme_classic()
```

# What is the percentage of genes within each Expression Set that are associated with an ELT-2 binding cluster?

```{r}
ggplot(
  expressionSet_per_BindingCluster,
  aes(x = ELT2_cluster, y = present, fill = Expression.Cluster)
) +
  geom_bar(stat = "identity", position = "fill") +
  theme_classic()
```

Make a series of horizontal barplots with percentage of ELT-2 binding cluster per expression cluster.  

First, calculate the percentage of each ELT-2 binding category against the total dataset.  

```{r}
percent_bound_per_ELT2_cluster <-
  expressionSet_per_BindingCluster %>% group_by(ELT2_cluster) %>% summarise(percent = sum(present) /
                                                                              nrow(dynamic_counts_matrix_scaled))
```

Next calculate the the 95% Confidence Interval with the Bionomial Test.

```{r}
expressionSet_per_BindingCluster %>% group_by(Expression.Cluster, ELT2_cluster) %>% summarise(percent = present /
                                                                                                (present + absent))

? binom.test
```
 
Calculate the binomial pvalue and confidence intervals.  

```{r}
# Add a column for the background percentage of ELT2 binding clusters per the whole expression dataset
expression_binding_stats <-
  expressionSet_per_BindingCluster %>% group_by(ELT2_cluster) %>% mutate(background_percent = sum(present) /
                                                                           (sum(present) + sum(absent)))

# Use binomia.test to calculate pvalue and confidence intervales for the percentage of ELT2 binding clusters per expression clusters
expression_binding_stats <- expression_binding_stats %>%
  group_by(ELT2_cluster, Expression.Cluster) %>%
  mutate(
    pval = binom.test(
      x = c(present, absent),
      n = present + absent,
      p = background_percent,
      alternative = "two.sided"
    )$p.value,
    conf.upper = binom.test(
      x = c(present, absent),
      n = present + absent,
      p = background_percent,
      alternative = "two.sided"
    )$conf.int[2],
    conf.lower = binom.test(
      x = c(present, absent),
      n = present + absent,
      p = background_percent,
      alternative = "two.sided"
    )$conf.int[1]
  )

expression_binding_stats$Expression.Cluster <-
  factor(
    expression_binding_stats$Expression.Cluster,
    levels = c("SET6", "SET5", "SET4", "SET3", "SET2", "SET1")
  )

expression_binding_stats %>% head()

```



```{r}
ggplot(expression_binding_stats,
       aes(x = Expression.Cluster,
           y = percent, fill = ELT2_cluster)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(limits = c(0, 0.75)) +
  theme_classic() +
  geom_hline(
    data = percent_bound_per_ELT2_cluster,
    color = "red",
    linetype = "dashed",
    aes(yintercept = percent)
  ) +
  geom_errorbar(
    ymax = expression_binding_stats$conf.upper,
    ymin = expression_binding_stats$conf.lower,
    width = 0.1
  ) +
  coord_flip() +
  facet_grid(. ~ ELT2_cluster) +
  scale_fill_manual(values = as.character(cluster_colors$val))

# ggsave(filename = "./03_plots/200511_Percent_of_ELT2bindClust_per_ExpressionClust.pdf")
```

# L1 Intestine expression row annotation

Import RWC24 L1 Intestine results table

```{r}
RWC24_res <-
  read.csv(
    "../RWC24_L1_Intestine_RNAseq/04_DEseq2/200511_L1_intestine_FACS_gut_vs_gutless.csv",
    row.names = 1
  )
RWC24_res <- RWC24_res %>% rownames_to_column(var = "WBGeneID")
head(RWC24_res)
```

Select log fold change data for row annotation.  

```{r}
L1_gut_lfc <-
  dynamic_counts_matrix_scaled %>% as.data.frame.matrix() %>% rownames_to_column(var = "WBGeneID") %>% inner_join(RWC24_res, by = "WBGeneID") %>% select(WBGeneID, log2FoldChange)
```

Add to heatmap as a row annotation.  

```{r}
col_fun = colorRamp2(c(-10, 0, 10), c("cyan", "black", "yellow"))

Ha_L1chip_L1gutLFC <-
  Ha_L1chip + rowAnnotation(L1_Gut_LFC = L1_gut_lfc$log2FoldChange,
                            col = list(L1_Gut_LFC = col_fun))
Ha_L1chip_L1gutLFC

# pdf(file = "./03_plots/200511_AnyDE_Genes_ELT2_ELT7_L1gutLFC.pdf", width = 5, height = 5)
# Ha_L1chip_L1gutLFC
# dev.off()
```


# Make a TF subset heatmap

```{r, out.width= '\\textwidth', fig.width= 5, fig.height=15, fig.align='center'}
head(dynamic_counts_matrix_scaled)

wTF3.0 <-
  read.csv("./01_input/TF3-0_namesonly.txt",
           sep = "\t",
           header = TRUE) %>% select(WBGeneID)

dynamic_counts_matrix_scaled_TFs <-
  matrix_select(dynamic_counts_matrix_scaled, wTF3.0$WBGeneID)

dynamic_counts_matrix_scaled_TFs_names <-
  id2name(dynamic_counts_matrix_scaled_TFs)

tf_heatmap <- Heatmap(
  dynamic_counts_matrix_scaled_TFs_names,
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = TRUE,
  show_column_names = TRUE,
  column_title = "Differential Expression of All Transcription Factors"
)
tf_heatmap
```

Add row annotation to indicate ELT-2 binding in L1 stage

```{r}
elt2_detected_in_L1 %>% filter(WBGeneID %in% rownames(dynamic_counts_matrix_scaled_TFs))

tf_bound_anno <-
  data.frame(
    elt2_detected_in_L1 = ifelse(
      test = rownames(dynamic_counts_matrix_scaled_TFs) %in% elt2_detected_in_L1$WBGeneID,
      yes = "bound",
      no = "not.bound"
    )
  )

tf_heatmap_L1bound <-
  tf_heatmap + rowAnnotation(L1_bound = tf_bound_anno$elt2_detected_in_L1,
                             col = list(L1_bound = c(
                               "bound" = "green", "not.bound" = "black"
                             )))
tf_heatmap_L1bound
```

Add row annotation of intestine expression

```{r}
tf_lfc <-
  L1_gut_lfc %>% filter(WBGeneID %in% rownames(dynamic_counts_matrix_scaled_TFs)) %>% select(log2FoldChange)

tf_heatmap_L1bound +
  rowAnnotation(log2FC = tf_lfc$log2FoldChange,
                col = list(log2FC = colorRamp2(c(-10, 0, 10), c(
                  "blue", "white", "red"
                ))))
```


Zoom in on only bound TFs

```{r}
dynamic_counts_matrix_scaled_TFs_bound <-
  matrix_select(dynamic_counts_matrix_scaled_TFs,
                elt2_detected_in_L1$WBGeneID)

dynamic_counts_matrix_scaled_TFs_bound_names <-
  id2name(dynamic_counts_matrix_scaled_TFs_bound)

bound_tf_lfc <-
  L1_gut_lfc %>% filter(WBGeneID %in% rownames(dynamic_counts_matrix_scaled_TFs_bound)) %>% select(log2FoldChange)

bound_tf_lfc

Heatmap(
  dynamic_counts_matrix_scaled_TFs_bound_names,
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = TRUE,
  row_names_side = "left",
  show_column_names = TRUE,
  column_title = "Differential Expression of ELT-2 bound Transcription Factors"
) +
  rowAnnotation(lfc = bound_tf_lfc$log2FoldChange,
                col = list(lfc = colorRamp2(c(-10, 0, 10), c(
                  "blue", "white", "red"
                ))))
```

This plot suggests that transcription factors bound by ELT-2 are typically upregulated in the absence of ELT-2. 

Additionally, TFs that are expressed in the L1 intestine are upregulated in absence of ELT-2 alone, but downregulated in the absence of both ELT-2 and ELT-7.  

Futhermore, TFs that are not expressed in the L1 intestine are upregulated only in the absence of both ELT-2 and ELT-7.  

TFs to follow up: pqm-1, zip-10, odd-1 (repressed by elt-2 alone, normally gut expressed). nhr-58 (vulva), zip-2 (neuron), cebp-1 (neuron), gla-3 (germline), zip-11

# old code below

# Transcription factor subset plots

Load in wTF3.0 list

```{r, out.width= '\\textwidth', fig.height= 4.5, fig.align='center'}
wTF3.0 <-
  read.csv("./01_input/TF3-0_namesonly.txt",
           sep = "\t",
           header = TRUE) %>% select(WBGeneID)

dynamic_counts_matrix_TFs <-
  matrix_select(dynamic_counts_matrix, wTF3.0$WBGeneID)
dynamic_counts_matrix_TFs_bound <-
  matrix_select(dynamic_counts_matrix_scaled_TFs,
                elt2_detected_in_L1$WBGeneID)

dynamic_counts_matrix_TFs_bound_sub <-
  subset(
    dynamic_counts_matrix_TFs_bound,
    select = c(
      "wt_sorted_1",
      "wt_sorted_2",
      "wt_sorted_3",
      "wt_sorted_4",
      "elt2D_sorted_1",
      "elt2D_sorted_2",
      "elt2D_sorted_3",
      "elt2D_sorted_4"
    )
  )

dynamic_counts_matrix_TFs_bound_sub_scale <-
  t(apply(unlist(dynamic_counts_matrix_TFs_bound_sub), 1, scale))

# varselect <- rowVars(colsub) >= 0.5
# colsub <- colsub[varselect, ]

dynamic_counts_matrix_TFs_bound_sub_scale_names <-
  id2name(dynamic_counts_matrix_TFs_bound_sub_scale)

bound_TF_heatmap <-
  Heatmap(
    dynamic_counts_matrix_TFs_bound_sub_scale_names,
    col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
    cluster_columns = FALSE,
    #clustering_distance_rows = "spearman",
    row_dend_reorder = TRUE,
    clustering_method_rows = "complete",
    show_row_names = TRUE,
    show_column_names = TRUE,
    row_names_gp = gpar(fontsize = 10),
    column_names_gp = gpar(cex = 0.4),
    heatmap_legend_param = list(color_bar = "continuous"),
    #split = 2,
    width = unit(3, "in"),
    height = unit(4, "in")
  )
bound_TF_heatmap

pdf("./03_plots/200428_TF_only_L1elt2bound",
    height = 5,
    width = 5)
bound_TF_heatmap
dev.off()
```

Add intestine expressed annotation.  

```{r}
bound_TF_rna_anno <- data.frame(
  spencerLE = ifelse(
    test = rownames(dynamic_counts_matrix_TFs_bound_sub_scale) %in% spencer_LE_subset$spencer_LE_ID,
    yes = "enriched",
    no = "depleted"
  ),
  spencerL2 = ifelse(
    test = rownames(dynamic_counts_matrix_TFs_bound_sub_scale) %in% spencer_L2_subset$spencer_L2_ID,
    yes = "enriched",
    no = "depleted"
  )
) 

bound_TF_heatmap +
    rowAnnotation(LE.intestine = bound_TF_rna_anno$spencerLE, col = list(LE.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE) +
  rowAnnotation(L2.intestine = bound_TF_rna_anno$spencerL2, col = list(L2.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE)
```



```{r, out.width= '\\textwidth', fig.height= 3.5, fig.align='center'}
dynamic_counts_matrix_TFs <- matrix_select(dynamic_counts_matrix, wTF3.0$WBGeneID)
dynamic_counts_matrix_TFs_sub <- subset(dynamic_counts_matrix_TFs, select = c("wt_sorted_1", "wt_sorted_2", "wt_sorted_3", "wt_sorted_4", "elt2D_sorted_1", "elt2D_sorted_2", "elt2D_sorted_3", "elt2D_sorted_4"))
dynamic_counts_matrix_TFs_sub_scale <- t(apply(unlist(dynamic_counts_matrix_TFs_sub), 1, scale)) 
dynamic_counts_matrix_TFs_sub_scale_names <- id2name(dynamic_counts_matrix_TFs_sub_scale)
Ha_TF <- Heatmap(dynamic_counts_matrix_TFs_sub_scale_names, 
          col = colorRampPalette(c("cyan","black","yellow"))(1000),    
          cluster_columns = FALSE,
          #clustering_distance_rows = "spearman",
          #row_dend_reorder = TRUE,
          clustering_method_rows = "complete", 
          show_row_names = TRUE,
          show_column_names = TRUE,
          row_names_gp = gpar(cex = 0.4),
          column_names_gp = gpar(cex = 0.4),
          heatmap_legend_param = list(color_bar = "continuous"),
          #split = 2,
           width = unit(2, "in"), 
          # height = unit(3, "in"),
          split = 2)
Ha_TF

# svg("./killme.svg", width = 8, height = 8)
# Ha_TF
# dev.off()
```

```{r}
TF_anno <- data.frame(elt2_detected_in_L1 = ifelse(test = rownames(dynamic_counts_matrix_TFs_sub_scale) %in% elt2_detected_in_L1$WBGeneID, yes = "bound", no = "not.bound")) 

Ha_TF + rowAnnotation(L1bound = TF_anno$elt2_detected_in_L1,
                   col = list(L1bound = c("bound" = "green", "not.bound" = "black")))

# pdf("./tf_heatmap.pdf", height = 8, width = 8)
# Ha_TF + rowAnnotation(L1bound = TF_anno$elt2_detected_in_L1,
#                    col = list(L1bound = c("bound" = "green", "not.bound" = "black")))
# dev.off()
```

Add intestine expression row annotation.  

```{r}
TF_rna_anno <- data.frame(
  spencerLE = ifelse(test = rownames(dynamic_counts_matrix_TFs_sub_scale) %in% spencer_LE_subset$spencer_LE_ID, yes = "enriched", no = "depleted"),
  spencerL2 = ifelse(test = rownames(dynamic_counts_matrix_TFs_sub_scale) %in% spencer_L2_subset$spencer_L2_ID, yes = "enriched", no = "depleted")
    ) 

Ha_TF_spencer <- Ha_TF + rowAnnotation(L1bound = TF_anno$elt2_detected_in_L1,
                   col = list(L1bound = c("bound" = "green", "not.bound" = "black"))) +
    rowAnnotation(LE.intestine = TF_rna_anno$spencerLE, col = list(LE.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE) +
  rowAnnotation(L2.intestine = TF_rna_anno$spencerL2, col = list(L2.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE)

# pdf("./03_plots/200428_DE_TFs_elt2chip_intestineExpression", height = 8, width = 8)
# Ha_TF_spencer
# dev.off()

```

## Results and interpretation

## Session Info

```{r}
sessionInfo()
```

