---
title: "ELT-2 Regulated Genes"
author: "Rtpw"
date: "2/12/2020"
output:
  pdf_document: default

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Next steps

- perform GO
- figure out rankings of gonad genes, then test most upregulated in elt-2(-)
- figure out rankings of normally expressed genes, then test most upregulated in elt-2(-)
- figure out which are transcription factors, test if combo RNAi of elt-2 and those TFs reduce normally expressed intestine overexpression

# Libraries

```{r}
library(DESeq2)
library(dplyr)
library(pheatmap)
```



# Differentail Expression

Load data

```{r}
RNAcounts <- read.csv("./01_input/Table_S1_Raw_Read_Counts.csv", header=TRUE, row.names = 1)
```

This count file contains more samples than what I want to analyze. Subset the columns to just have `wt_sorted_*` and `elt2D_sorted_*`. Also select columns that correspond to ce11 genome assembly, since this is the genome used for the ChIP-seq analysis.

```{r}
cts <- RNAcounts %>% select(wt_sorted_1, wt_sorted_2, wt_sorted_3, wt_sorted_4, elt2D_sorted_1, elt2D_sorted_2, elt2D_sorted_3, elt2D_sorted_4)
head(cts)
```

make `coldata`

```{r}
coldata <- data.frame(condition = c("wt", "wt", "wt", "wt", "elt2D", "elt2D", "elt2D", "elt2D"), row.names = colnames(cts))
coldata
```

Check that column matrix and coldata match

```{r}
all(rownames(coldata) == colnames(cts))
```

Generate DESeqDataSet

```{r}
dds <- DESeqDataSetFromMatrix(countData = cts, colData = coldata, design = ~ condition)

# add gene names
moreFeatures <- data.frame(gene_name = RNAcounts$gene_id_val, sequence_id = RNAcounts$sequence_id_list)
mcols(dds) <- DataFrame(mcols(dds), moreFeatures)
mcols(dds)
```

Tell DESeq which samples are "control and which are "control" vs "treatment". This sets up the fold change comparison manually rather than letting the alphabetical determination of factor levels.

ie: 
without this step: logfoldchange(elt2D/wt)
with this step: logfoldchange(wt/elt2D)

```{r}
dds$condition <- factor(dds$condition, levels = c("wt", "elt2D"))
```

Perform differential expression analysis

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)

#View(as.data.frame(res))

# Convert res to dataframe
res.df <- as.data.frame(res)

# Export the results table
write.csv(res.df, file = "200218_L1_wt_vs_elt2D_results.csv")

# Print results table information
summary(res)
res
```

Determine if glh-1 is upregulated in elt-2(-)
glh-1 = WBGene00001598
glh-1 is a germline specific gene
It is upregulated

```{r}
res.df[rownames(res.df) == "WBGene00001598",]
```

```{r}
plotCounts(dds, gene = "WBGene00001598")
```

See if elt-2 is depleted
It is depleted

```{r}
plotCounts(dds, gene = "WBGene00001250")
```

See if pgl-1 is enriched
Looks like it is depleted.
```{r}
plotCounts(dds, gene = "WBGene00003992")
```

see if egl-20 (ligand of Wnt pathway) is expressed
Also is depleted

```{r}
plotCounts(dds, gene = "WBGene00001188")
```


Make an MA plot cus why not

```{r}
plotMA(res, ylim = c(-10, 10))
```

Reformat for BETA

```{r}
betaDE <- data.frame(WBGeneID = rownames(res.df), lfc = res.df$log2FoldChange, q = res.df$padj)

write.table(betaDE, file = "./200218_L1_wt_vs_elt2D_results_BETA.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

# Import ChIP-seq data

```{r}
peaks <- read.delim("./01_input/L1_1_L1_2.IDR_0.05.narrowPeak", header = FALSE, sep = "\t")
head(peaks)

peaks_subset <- data.frame(chrom = peaks$V1, start = peaks$V2, end = peaks$V3, name = peaks$V4, score = peaks$V5, strand = peaks$V6)

write.table(peaks_subset, file = "200218_L1_ELT2_ChIP_Peaks_BETA.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

