---
title: "RWC23_Unused_Code"
output: html_notebook
---

# BETA

This is code used to make files suitable for BETA that I no longer need.


```{r}
peaks <- read.delim("./01_input/L1_1_L1_2.IDR_0.05.narrowPeak", header = FALSE, sep = "\t")
head(peaks)

peaks_subset <- data.frame(chrom = peaks$V1, start = peaks$V2, end = peaks$V3, name = peaks$V4, score = peaks$V5, strand = peaks$V6)

write.table(peaks_subset, file = "./03_BETA/BETA_Formatted_Input/200218_L1_ELT2_ChIP_Peaks_BETA.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

# Make BETA-friendly GTF file

This annotation file corresponds to annotation version WS271.

```{r}
paramart <- useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

genes_coding = getBM(mart = paramart, 
                     filter=c("species_id_1010", 
                              "biotype"), 
                     value=list(species_id_1010="caelegprjna13758", 
                                biotype="protein_coding"), 
                     attributes = c('refseq_mrna',
                                    'chromosome_name',
                                    'strand',
                                    'start_position', 
                                    'end_position', 
                                    'wbps_gene_id'))

# exclude mitochondria
genes_coding_noMT = genes_coding[genes_coding$chromosome_name != 'MtDNA',]
# add 'chr'
genes_coding_noMT$chromosome_name = paste('chr', genes_coding_noMT$chromosome_name, sep='')
# strands must be '+/-', but paramart returns 1/-1
strand_plus_minus = genes_coding_noMT$strand
strand_plus_minus[ strand_plus_minus == '1'] <- '+'
strand_plus_minus[ strand_plus_minus == '-1'] <- '-'
genes_coding_noMT$strand = strand_plus_minus

head(genes_coding_noMT)
colnames(genes_coding_noMT) <- c("refseqID", "chrom", "strand", "start", "end", "WBGeneID")
genes_coding_noMT_reduced <- genes_coding_noMT %>% filter(refseqID %in% refseqBetaDE$refseqID)

write.table(genes_coding_noMT_reduced, file = "200227_Wormbase_WS271_Genome_Annotation.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

# Replace WBGeneID for Refseq mRNA ID

```{r}
betaDE <- read.table("./02_formatted/200218_L1_wt_vs_elt2D_results_BETA.tsv", sep = "\t")
colnames(betaDE) <- c("WBGeneID", "lfc", "P.value")

# remove rows with NA
betaDE %>% filter(lfc != "NA" , q != "NA")
write.table(betaDE, file = "./03_BETA/BETA_Formatted_Input/200227_L1_wt_vs_elt2D_results_BETA_noNA.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)

refseqBetaDE <- inner_join(betaDE, genes_coding_noMT, by = "WBGeneID") %>% select(refseqID, lfc, q) %>% filter(refseqID != "", lfc != "NA" , q != "NA")

dim(refseqBetaDE)
write.table(refseqBetaDE, file = "./03_BETA/BETA_Formatted_Input/200227_L1_wt_vs_elt2D_results_BETA_refseq.tsv", sep = "\t", col.names = FALSE, row.names = FALSE, quote = FALSE)
```

Use command line to run BETA. Example code below.

```{bash}
BETA basic -p 03_BETA/BETA_Formatted_Input/200218_L1_ELT2_ChIP_Peaks_BETA.tsv -e 03_BETA/BETA_Formatted_Input/200227_L1_wt_vs_elt2D_results_BETA_refseq.tsv -k BSF --gname2 -r 200227_Wormbase_WS271_Genome_Annotation.tsv -n basic -d 2000
```

Overall having a hard time executing and getting output from BETA. Going to try a different tool.