---
title: "RWC23_ELT2_Regulated_Genes"
author: "RTPW"
date: "4/13/2020"
output: 
  pdf_document: default
  fig_crop: no
  
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, fig_crop = FALSE)
```

# Install Packages

```{r}
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install()
# BiocManager::install("biomaRt")
# install.packages("tidyverse")
# install.packages("readxl")
# BiocManager::install("ComplexHeatmap")
# install.packages("matrixStats")
# install.packages("pheatmap")
# install.packages("RVAideMemoire")
# install.packages("dendextend")
# install.packages("binom")
```


# Load Package Libraries

```{r}
library(biomaRt)
library(tidyverse)
library(readxl)
library(ComplexHeatmap)
library(matrixStats)
library(pheatmap)
library(RVAideMemoire)
library(dendextend)
library(binom)
library(circlize)
```

# To do

- Finish adding dialog
- Add spencer LE and L2 intestine expression data to dineen heatmap

# Background and Rationale

ELT-2 is the C. elegans intestine master regulator. Deletion of ELT-2 leads to a larval lethal phenotype, and expression of ELT-2 in non-intestine tissue induces an intestine fate.

This documet will generate plots to address the questions outlined below.

For genes differentially expressed during elt-2 (-) and/or elt-7(-):

1) which expression pattern clusters associate with ELT-2 binding?
2) which expression pattern clusters associate with ELT-2 binding categories?
  + For all genes
  + For only genes bound by ELT-2
3) Which expression pattern clusters associate with intestine expression? (MA plot for each expression set)
  + For all genes
  + For genes only bound by ELT-2

For clusters of transcription factors (TFs) differentially expressed during elt-2 (-) and/or elt-7(-):

1) which transcription factor clusters associate with ELT-2 binding?
2) which transcription factor clusters associate with ELT-2 binding categories
  + for all TFs
  + For only TFs bound by ELT-2
3) which transcription factor clusters associate with intestine expression?
  + for all
  + for only ELT-2 bound

# Description of Data

I will integrate two RNA-seq experiments and a ChIP-seq experiments.

The first is a set of RNA-seq experiments in L1 stage worms (Dineen and Nishimura, 2018). They were collected from the following genotypes, all in the L1 stage: 

- wildtype (wt)
- elt-7 deleted (elt7D)
- elt-2 deleted (elt2D)
- combination fo elt-7 and elt-2 deleted (elt2Delt7D)

The purpose of including elt-7 and elt-2/elt-7 double deletion is because these two transcription factors have overlapping functionality. Deletion of elt-7 alone does not have a phenotype, but deletion of elt-7 in combination with elt-2 has an enhanced lethal phenotype of just elt-2 alone.

The second RNA-seq experiment is from FACS sorted L1 stage intestine cells. This data is unpublished.  

The ChIP-seq experiments are performed against ELT-2 and are from the following developmental stages:

- late embryo (LE)
- L1
- L3

They were collected as part of the modENCODE consortium and were processed by David King. He has provided gene mapping of ELT-2 targets and categories of ELT-2 binding. The ELT-2 binding categories are as follows:

- Not changing
- Larval
- L3 high
- Embryonic
- Increasing


# Citations

1) Dineen, A., Osborne Nishimura, E., Goszczynski, B., Rothman, J. H., & McGhee, J. D. (2018). Quantitating transcription factor redundancy: The relative roles of the ELT-2 and ELT-7 GATA factors in the C. elegans endoderm. Developmental Biology, 435(2), 150–161. https://doi.org/10.1016/J.YDBIO.2017.12.023

2) Kudron, M. M., Victorsen, A., Gevirtzman, L., Hillier, L. W., Fisher, W. W., Vafeados, D., … Waterston, R. H. (2018). The modern resource: genome-wide binding profiles for hundreds of Drosophila and Caenorhabditis elegans transcription factors. Genetics, 208(3), 937–949. https://doi.org/10.1534/genetics.117.300657

3) Spencer, W. C., Zeller, G., Watson, J. D., Henz, S. R., Watkins, K. L., McWhirter, R. D., Petersen, S., Sreedharan, V. T., Widmer, C., Jo, J., Reinke, V., Petrella, L., Strome, S., Von Stetina, S. E., Katz, M., Shaham, S., Rätsch, G., & Miller, D. M. (2011). A spatial and temporal map of C. elegans gene expression. Genome Research, 21(2), 325–341. https://doi.org/10.1101/gr.114595.110

4) Boeck, M. E., Huynh, C., Gevirtzman, L., Thompson, O. A., Wang, G., Kasper, D. M., Reinke, V., Hillier, L. W., & Waterston, R. H. (2016). The time-resolved transcriptome of C. elegans. Genome Research, 26(10), 1441–1450. https://doi.org/10.1101/gr.202663.115

# Code

## Source functions

```{r}
source("./RWC23_Functions.R")
```

# Load and Process Datasets

## Load Dineen and Osborne Nishimura et. al. Data


```{r}
dineen_nishimura_counts <-
  read_xlsx(path = "./01_input/Table_S2_rlog_Stabilized_Read_Counts.xlsx",
            sheet = "Sheet1")

dineen_nishimura_counts_matrix <- dineen_nishimura_counts %>%
  column_to_rownames(var = "WBGeneID")  %>%
  data.matrix()

dineen_nishimura_counts_matrix %>% head
```

list of all dynamically expressed genes

```{r}
dynamic_regulated_genes <-
  read.table(file = "./05_fromErin/2017-11-20_all_changing_genes_0.1alpha_0.8lfc.txt",
             quote = "",
             header = FALSE)
colnames(dynamic_regulated_genes) <- "WBGeneID"

dynamic_regulated_genes %>% head
```

Load differential expression clusters from Dineen and Nishimura et al (2018).  

```{r}
dineen_nishimura_clusters <-
  read_xlsx(path = "./01_input/Table_S6_All_Dynamically_Expressed_Genes_Clusters.xlsx",
            sheet = "dataset")

dineen_nishimura_sets <-
  dineen_nishimura_clusters %>% select(WBGeneID, set)
dineen_nishimura_sets_ascend <-
  arrange(dineen_nishimura_sets, WBGeneID)
dineen_nishimura_sets_ascend$set <-
  toupper(dineen_nishimura_sets_ascend$set)
dineen_nishimura_sets_ascend %>% head
```

## Load ELT-2 ChIP-seq binding annotations

```{r}
elt2_peaks <-
  read_xlsx("./01_input/200410_peaksForBigBed.xlsx", sheet = "full cluster assignment")

# elt2_peaks <- readRDS("./01_input/annotatedPeaks.rds")

colnames(elt2_peaks) <-
  c(
    "chrom",
    "start",
    "end",
    "peak.name",
    "WBGeneID",
    "mapping",
    "cluster",
    "cluster.description",
    "kweight",
    "LE",
    "L1",
    "L3",
    "peak.summit.agreement"
  )

elt2_peaks$cluster.description <-
  factor(
    elt2_peaks$cluster.description,
    levels = c(
      "LE-specific",
      "Post-embryonic",
      "Increasing",
      "L3-high",
      "Not-changing or not IDR-passing"
    ),
    labels = c(
      "Embryo_Specific",
      "Larval",
      "Increasing",
      "L3_High",
      "Not_Changing"
    )
  )

elt2_cluster_names <- c("Embryo_Specific",
                        "Larval",
                        "Increasing",
                        "L3_High",
                        "Not_Changing")

elt2_peaks %>% head
```

Make a set of genes with ELT-2 binding detected in the L1 stage.  

```{r}
elt2_detected_in_L1 <-
  elt2_peaks %>% select(WBGeneID, L1) %>% filter(L1 == 1) %>% select(WBGeneID) %>% unique()

elt2_detected_in_L1 %>% head
elt2_detected_in_L1 %>% dim

```
Make a dataframe that records the number of peaks per gene that fall in a particular binding catagory.

```{r}
binding_cluster_gene_counts <-
  table(elt2_peaks$WBGeneID, elt2_peaks$cluster.description)
binding_cluster_gene_counts <-
  as.data.frame.matrix(binding_cluster_gene_counts)
binding_cluster_gene_counts %>% head()
```


## Load Spencer et. al. intestine expression

This data is from a 2011 paper using FACS sorting of Late Embryo (LE) and Larval Stage 2 (L2) intestine cells, measured with microarray. See [Spencer et. al, (2011)](https://pubmed.ncbi.nlm.nih.gov/21177967/).  

```{r}
spencerLEgenes <-
  read.table(
    "./01_input/Spencer_et_al_2010_FACS_and_pulldown_tilling_array/LE-intestine_enr_vs_ref.WS200.txt",
    quote = "\"",
    comment.char = "",
    header = TRUE
  )
colnames(spencerLEgenes) <-
  str_c("spencer_LE_", colnames(spencerLEgenes))
spencer_LE_subset <-
  spencerLEgenes %>% select(spencer_LE_ID,
                            spencer_LE_AveExpr,
                            spencer_LE_adj_P_Val,
                            spencer_LE_FC)

spencer_LE_subset %>% head
```

```{r}
spencerL2genes <-
  read.table(
    "./01_input/Spencer_et_al_2010_FACS_and_pulldown_tilling_array/L2-intestine_enr_vs_ref.WS200.txt",
    quote = "\"",
    comment.char = "",
    header = TRUE
  )
colnames(spencerL2genes) <-
  str_c("spencer_L2_", colnames(spencerL2genes))
spencer_L2_subset <- spencerL2genes %>%
  select(spencer_L2_ID,
         spencer_L2_AveExpr,
         spencer_L2_adj_P_Val,
         spencer_L2_FC)

spencer_L2_subset %>% head
```

## Process rlog counts

Subset rlog matrix based on presence in list `2017-11-20_all_changing_genes_0.1alpha_0.8lfc.txt`. Row scale and center the rlog counts per genes.  

```{r}
dynamic_counts_matrix <-
  matrix_select(dineen_nishimura_counts_matrix,
                dynamic_regulated_genes$WBGeneID)

dynamic_counts_matrix_scaled <-
  t(apply(unlist(dynamic_counts_matrix), 1, scale))

rownames(dynamic_counts_matrix_scaled) <-
  rownames(dynamic_counts_matrix)
colnames(dynamic_counts_matrix_scaled) <-
  colnames(dynamic_counts_matrix)
dynamic_counts_matrix_scaled %>% head

dynamic_counts_matrix_scaled_ascend <-
  dynamic_counts_matrix_scaled[order(rownames(dynamic_counts_matrix_scaled)),]
```

Must use arrange to sort genes in descending order to ensure row order is preserved

# Recreate Supplementary Figure S4a from Dineen and Nishimura et al.    

Use expression clusters from Dineen and Nishimura et al to split the clusters.  

```{r}
Heatmap(
  dynamic_counts_matrix_scaled_ascend,
  name = "elt2D-elt7D\nRNAseq",
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = FALSE,
  show_column_names = TRUE,
  row_names_gp = gpar(cex = 0.2),
  column_names_gp = gpar(cex = 0.4),
  heatmap_legend_param = list(color_bar = "continuous"),
  row_split = dineen_nishimura_sets_ascend$set
)
```

Add expression set and column labels.  

```{r}
RNA_column_order <-
  factor(c(
    rep("WT", 4),
    rep("elt7D", 3),
    rep("elt2D", 4),
    rep("elt7Delt2D", 3)
  ),
  levels = c("WT", "elt7D", "elt2D", "elt7Delt2D"))
RNA_column_order
```


```{r}
column_labels <- structure(c("rep1", "rep2", "rep3", "rep4", "rep1", "rep2", "rep3", "rep1", "rep2", "rep3", "rep4", "rep1", "rep2", "rep3"),
                           names = colnames(dynamic_counts_matrix_scaled_ascend)
)

column_labels
```


```{r}
Ha <- Heatmap(
  dynamic_counts_matrix_scaled_ascend,
  name = "elt2D-elt7D\nRNAseq",
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = FALSE,
  show_column_names = TRUE,
  column_labels = column_labels[colnames(dynamic_counts_matrix_scaled_ascend)],
  column_names_gp = gpar(cex = 0.7),
  heatmap_legend_param = list(color_bar = "continuous"),
  row_split = dineen_nishimura_sets_ascend$set,
  row_title = NULL,
  column_title = NULL,
  column_split = RNA_column_order,
  bottom_annotation = HeatmapAnnotation(
    foo = anno_block(
      labels = c("WT", "elt7D", "elt2D", "elt7D;elt2D"),
      labels_gp = gpar(cex = .8),
      gp = gpar(border = NA, lty = "blank")
      ),
    foo2 = anno_block(gp = gpar(fill = "black"), height = unit(0.5, "mm"))
  ),
  left_annotation = rowAnnotation(foo = anno_block(
    labels = c("SET1", "SET2", "SET3", "SET4", "SET5", "SET6"),
    labels_rot = 0,
    gp = gpar(border = NA, lty = "blank", cex = 0.4)
  ))
)
Ha
```


Sanity check to ensure that cluster splitting is occuring correctly. Remap the Set assignments back to the heatmap as a row annotation.

```{r}
Ha + rowAnnotation(set = dineen_nishimura_sets_ascend$set)
```





## Add L1 stage ELT-2 binding

This section will add annotation to the rows of the elt2/elt7 differentiall expression heatmap with ELT-2 ChIP-seq binding during the L1 stage. This will determine what differential expression sets associate with ELT-2 binding during the L1 stage. The reason L1 stage ChIP-seq eaks are being used is because the elt2/elt7 RNA-seq experiment was conducted in the L1 stage.  

In ComplexHeatmap the row order of input matrix and annotation df must be identical to accurately plot data.  

```{r}
elt2_detected_in_L1 %>% dim

elt2_L1_anno <-
  data.frame(
    WBGeneID = rownames(dynamic_counts_matrix_scaled_ascend),
    elt2_detected_in_L1 = ifelse(
      test = rownames(dynamic_counts_matrix_scaled_ascend) %in% elt2_detected_in_L1$WBGeneID,
      yes = "bound",
      no = "not.bound"
    ),
    stringsAsFactors = FALSE
  )

elt2_L1_anno %>% head()
```

Incorporate this into a heatmap annotation

```{r}
Ha_L1chip <-
  Ha + rowAnnotation(L1bound = elt2_L1_anno$elt2_detected_in_L1,
                     col = list(L1bound = c(
                       "bound" = "green", "not.bound" = "black"
                     )))

Ha_L1chip

pdf("./03_plots/figure_1/200615_Figure1A_elt2elt7DERNAseq_L1elt2bound_heatmap.pdf", height = 4, width = 4.5)
Ha_L1chip
dev.off()
```

Visually it appears that some elt2/elt7 differential expression clusters have more or less ELT-2 binding associated with the sets. I would like to be more quantitative with this assesment.  

Determine enrichment of ELT-2 binding during L1 stage. I will calculate the percentage of genes with an ELT-2 ChIP-seq peak detected during the L1 stage.  

First use `merge` to combine the ELT-2 binding status and expression set for each gene.  

```{r}
expression_L1_binding <-
  merge(elt2_L1_anno, dineen_nishimura_sets_ascend, by = "WBGeneID")
expression_L1_binding %>% head
```

Next use `table` to tally the number of bound and not.bound genes per expression set.  

```{r}
clust_L1bound_counts <-
  table(expression_L1_binding$set,
        expression_L1_binding$elt2_detected_in_L1)
clust_L1bound_counts
```

Use `prop.table` to convert these values to percentages within each set.  

```{r}
clust_L1bound_prop <- prop.table(clust_L1bound_counts, 1)
clust_L1bound_prop
```

Adjust the percentages object into a dataframe that `ggplot2` can use.  

```{r}
clust_L1bound_prop_ggplot <- as.data.frame(clust_L1bound_prop)

colnames(clust_L1bound_prop_ggplot) <- c("SET", "Status", "Freq")

clust_L1bound_prop_ggplot$Status <-
  factor(clust_L1bound_prop_ggplot$Status,
         levels = c("not.bound", "bound"))

clust_L1bound_prop_ggplot$SET <-
  factor(
    clust_L1bound_prop_ggplot$SET,
    levels = c("SET6", "SET5", "SET4", "SET3", "SET2", "SET1")
  )

clust_L1bound_colors <- c("bound" = "green", "not.bound" = "black")

l1bound_percents <-
  ggplot(
    clust_L1bound_prop_ggplot %>% filter(Status == "bound"),
    aes(
      x = SET,
      y = Freq,
      fill = Status,
      order = Status
    )
  ) +
  geom_bar(stat = "identity") +
  scale_color_manual(values = clust_L1bound_colors,
                     aesthetics = c("color", "fill")) +
  ggtitle("Percentage of L1 Stage ELT-2 Binding Per
          Expression Set") +
  xlab("Expression Set") +
  ylab("Percentage") +
  theme_classic() +
  theme(axis.text.x = element_text(size = 13)) +
  ylim(0, 1)

l1bound_percents

# ggsave("./03_plots/200428_proportion_of_l1elt2_per_expression_cluster.pdf", height = 2, width = 5)
```

This plot shows that all of the differential expression sets have less than 50% of genes bound by ELT-2.  

Rather than viewing percentages of genes bound, what is the number of "bound" vs "not.bound" per cluster?  

```{r}
clust_L1bound_counts_ggplot <- as.data.frame(clust_L1bound_counts)
colnames(clust_L1bound_counts_ggplot) <- c("SET", "Status", "Freq")

ggplot(clust_L1bound_counts_ggplot,
       aes(x = SET,
           y = Freq,
           fill = Status)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_color_manual(values = clust_L1bound_colors,
                     aesthetics = c("color", "fill")) +
  ggtitle("Number of L1 Stage ELT-2 Binding Site Per
          Expression Set") +
  xlab("Expression Set") +
  ylab("Count") +
  theme_bw() +
  theme(axis.text.x = element_text(size = 13))

# ggsave("./03_plots/200428_number_of_l1elt2_per_expression_cluster.pdf", height = 2, width = 5)
```



Use the binomial test to determine if the different expression clusters are enriched or depleted for ELT-2 binding.  

Use `binom.test` and first do a two-tailed test.  

First calculate the proportion of bound genes over the total number of genes in the analysis.  

```{r}
proportion = as.numeric(colSums(clust_L1bound_counts)[1]) /
  as.numeric(colSums(clust_L1bound_counts)[1] + colSums(clust_L1bound_counts)[2])
proportion
```

Use custom function `ctable_binom()` to calculate p-vaule and confidence intervals for each set.

```{r}
l1bound_binom <- ctable_binom(clust_L1bound_counts, "two.sided")
```


This says that all sets but SET5 have a significant difference in genes bound compared to the entire dataset.  

Now use the `less` or `greater` argument of `binom.test` to see if there is more or less binding.  

```{r}
ctable_binom(ctable = clust_L1bound_counts, alt = "less")
```

This says that set 1 and 2 have less ELT-2 binding compared to the entire dataset.  

Now try `greater`.  

```{r}
ctable_binom(clust_L1bound_counts, "greater")
```

This says that SET3, SET4 and SET6 have a higher percentage of genes bound compared the the "background" percent of bound genes for the entire dataset.  

Make a plot that visually depicts this. Draw line on the percentage plot to indicate background percentage of L1 stage ELT-2 binding.  

```{r}
l1bound_percents +
  geom_hline(yintercept = proportion,
             color = "red",
             linetype = "dashed") +
  geom_errorbar(
    ymax = l1bound_binom$conf.upper,
    ymin = l1bound_binom$conf.lower,
    width = 0.25
  ) +
  coord_flip() +
  ggtitle("L1 stage ELT-2 binding per\nexperession set")

# ggsave(
#   "./03_plots/figure_1/200615_Figure1B_percentage_l1bound_per_expression_cluster.pdf",
#   width = 4,
#   height = 5
# )
```


Use the hypergeometric test to determine: Are changing genes (all sets) enriched for L1 binding?

```{r}
N <- 20470
k <- nrow(elt2_detected_in_L1)
x3 <- as.numeric(colSums(clust_L1bound_counts)[1])
m <-
  as.numeric(colSums(clust_L1bound_counts)[1] + colSums(clust_L1bound_counts)[2])
dhyper(x3, m, N, k)

```

A very small p-value for the hypergeometric test suggests that the entire dataset is enriched for ELT-2.  

The next section with compute pairwise fisher's exact tests for the different sets. I have a difficult time interpreting these results.  

```{r}
fisher.multcomp(clust_L1bound_counts, p.method = "bonferroni")
```

```{r}
fisher.multcomp(clust_L1bound_counts, p.method = "bonferroni")$p.value < 0.05
```


## Row annotation of ELT-2 Binding Pattern Clusters

This section will add annotation to the rows of the elt2/elt7 differentiall expression heatmap with ELT-2 ChIP-seq binding pattern clusters. This will determine what differential expression sets associate with ELT-2 binding patters.  

Start by using custom function `make_cluster_annotation()`. This function takes two objects: the matrix of gene expression values and a dataframe of counts ELT-2 binding patterns per genes. It returns a dataframe with the number of ELT-2 binding categories associated with each gene.  

```{r}
chip_annotation <-
  make_cluster_annotation(dynamic_counts_matrix_scaled_ascend,
                          binding_cluster_gene_counts)

chip_annotation %>% head()
```

Sanity check to ensure that the order and number of rows is preserved.  

```{r}
unique(rownames(dynamic_counts_matrix_scaled_ascend) == chip_annotation$WBGeneID)

nrow(dynamic_counts_matrix_scaled) == nrow(chip_annotation)
```

Build add row annotation for the number of ELT-2 binding clusters associated with each gene.  

```{r}
Ha_L1chip_bindcluster <- Ha_L1chip +
  rowAnnotation(Embryo_Specific = chip_annotation$Embryo_Specific) +
  rowAnnotation(Larval = chip_annotation$Larval) +
  rowAnnotation(Increasing = chip_annotation$Increasing) +
  rowAnnotation(L3_High = chip_annotation$L3_High) +
  rowAnnotation(Not_Changing = chip_annotation$Not_Changing)
Ha_L1chip_bindcluster
```

Have the colors match plot from David.
```{r}
cluster_colors <-
  data.frame(
    class = elt2_cluster_names,
    val = c("#7570B3", "#1B9E77", "#E7298A", "#D95F02", "#505050")
  )

cluster_colors$class <-
  factor(x = cluster_colors$class,
         levels = elt2_cluster_names)
```


Convert ChIP binding clusters to a present/absence list.  

```{r}
chip_annotation_present_absent <-
  make_cluster_binary_annotation(chip_annotation)
```

Plot the heatmap with presence/absence.


```{r}
Ha_L1chip_clusterchip <-
  Ha_L1chip + binding_cluster_row_annotation(chip_annotation_present_absent)

Ha_L1chip_clusterchip

# pdf("./03_plots/figure_1/200615_Figure1A_wt_RNA_L1elt2bound_elt2bindclusters_heatmap.pdf", height = 4.5, width = 4)
# Ha_L1chip_clusterchip
# dev.off()
```

Plot percentage of expression cluster group having binding pattern assignment.  

```{r}
exprclust_bindclust <-
  merge(
    dineen_nishimura_sets_ascend,
    chip_annotation_present_absent,
    by.x = "WBGeneID",
    by.y = "rowname"
  )

exprclust_bindclust %>% head
```

# What is the percentage of genes with annotated ELT2 binding clusters per expression dataset?

Make a dataframe that addresses the question: 

```{r}

expressionSet_per_BindingCluster <- data.frame()
for (i in elt2_cluster_names) {
  toappend <-
    table(exprclust_bindclust$set,
          exprclust_bindclust[[i]]) %>%
    as.data.frame.matrix() %>%
    rownames_to_column(var = "set") %>%
    mutate(ELT2_cluster = i,
           percent = present / (present + absent))
  expressionSet_per_BindingCluster <-
    bind_rows(expressionSet_per_BindingCluster, toappend)
}

expressionSet_per_BindingCluster$ELT2_cluster <-
  factor(expressionSet_per_BindingCluster$ELT2_cluster, levels = elt2_cluster_names)

expressionSet_per_BindingCluster
```


Make a plot that addresses the question: 

```{r}
ggplot(expressionSet_per_BindingCluster,
       aes(x = set,
           y = present,
           fill = ELT2_cluster)) +
  geom_bar(stat = "identity", position = "fill") +
  theme_classic() +
  scale_fill_manual(values = as.vector(cluster_colors$val))

# ggsave("./03_plots/figure_1/200615_FigureS1A_Cluster_percent_present_per_Set.pdf")
```

# What is the percentage of genes within each Expression Set that are associated with an ELT-2 binding cluster?

```{r}
ggplot(expressionSet_per_BindingCluster,
       aes(x = ELT2_cluster, y = present, fill = set)) +
  geom_bar(stat = "identity", position = "fill") +
  theme_classic()

# ggsave("./03_plots/figure_1/200615_FigureS1B_Set_percent_present_per_Cluster.pdf")
```

Make a series of horizontal barplots with percentage of ELT-2 binding cluster per expression cluster.  

TODO: Make a function that does all that is below.  


First, calculate the percentage of each ELT-2 binding category against the total dataset.  

```{r}
percent_bound_per_ELT2_cluster <-
  expressionSet_per_BindingCluster %>% group_by(ELT2_cluster) %>% summarise(percent = sum(present) /
                                                                              nrow(dynamic_counts_matrix_scaled))
```

Next calculate the the 95% Confidence Interval with the Bionomial Test.

```{r}
expressionSet_per_BindingCluster %>% group_by(set, ELT2_cluster) %>% summarise(percent = present /
                                                                                 (present + absent))

```
 
Calculate the binomial pvalue and confidence intervals.  

```{r}
# Add a column for the background percentage of ELT2 binding clusters per the whole expression dataset
expression_binding_stats <-
  expressionSet_per_BindingCluster %>% group_by(ELT2_cluster) %>% mutate(background_percent = sum(present) /
                                                                           (sum(present) + sum(absent)))

# Use binom.test to calculate pvalue and confidence intervales for the percentage of ELT2 binding clusters per expression clusters
expression_binding_stats <- expression_binding_stats %>%
  group_by(ELT2_cluster, set) %>%
  mutate(
    pval = binom.test(
      x = c(present, absent),
      n = present + absent,
      p = background_percent,
      alternative = "two.sided"
    )$p.value,
    conf.upper = binom.test(
      x = c(present, absent),
      n = present + absent,
      p = background_percent,
      alternative = "two.sided"
    )$conf.int[2],
    conf.lower = binom.test(
      x = c(present, absent),
      n = present + absent,
      p = background_percent,
      alternative = "two.sided"
    )$conf.int[1]
  )

expression_binding_stats$set <-
  factor(
    expression_binding_stats$set,
    levels = c("SET6", "SET5", "SET4", "SET3", "SET2", "SET1")
  )

expression_binding_stats %>% head()

```



```{r}
ggplot(expression_binding_stats,
       aes(x = set,
           y = percent, fill = ELT2_cluster)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(limits = c(0, 0.75)) +
  theme_classic() +
  geom_hline(
    data = percent_bound_per_ELT2_cluster,
    color = "red",
    linetype = "dashed",
    aes(yintercept = percent)
  ) +
  geom_errorbar(
    ymax = expression_binding_stats$conf.upper,
    ymin = expression_binding_stats$conf.lower,
    width = 0.1
  ) +
  coord_flip() +
  facet_grid(. ~ ELT2_cluster) +
  scale_fill_manual(values = as.character(cluster_colors$val))

# ggsave(filename = "./03_plots/figure_1/200615_Figure1C_Percent_of_ELT2bindClust_per_ExpressionClust.pdf",
#        height = 5,
#        width = 7)
```

# Subset ELT-2/ELT-7 differentially expressed genes based on ELT-2 binding in L1 stage

IAMHERE: FIX RNA_heatmap Function to look pretty like the `Ha` plot

```{r}
RNA_heatmap2(dynamic_counts_matrix_scaled_ascend,
            column_split = RNA_column_order,
            row_split = elt2_L1_anno$elt2_detected_in_L1) +
  elt2_l1_row_annotation(elt2_L1_anno) +
  binding_cluster_row_annotation(chip_annotation_present_absent)
```



```{r}
l1_bound_list <-
  elt2_L1_anno %>% filter(elt2_detected_in_L1 == "bound") %>% select(WBGeneID) %>% arrange(WBGeneID)

dynamic_counts_matrix_scaled_bound_only <-
  matrix_select(dynamic_counts_matrix_scaled_ascend, l1_bound_list$WBGeneID)

bound_only_elt2_clust_anno <-
  make_cluster_binary_annotation(
    make_cluster_annotation(
      dynamic_counts_matrix_scaled_bound_only,
      binding_cluster_gene_counts
    )
  )

bound_only_elt2_clust_anno %>% head()
```

Assign k-means clusters for rows before plotting

```{r}
kclus <- kmeans(dynamic_counts_matrix_scaled_bound_only, 4)
bound_only_sets <-
  data.frame(
    WBGeneID = rownames(dynamic_counts_matrix_scaled_bound_only),
    set = paste("SET", kclus$cluster, sep = "")
  )
head(bound_only_sets)
```

Draw heatmap and check that set assignment is correct.  

```{r}
Ha_bound_only <-
  RNA_heatmap2(mat = dynamic_counts_matrix_scaled_bound_only,
              column_split = RNA_column_order,
              row_split = bound_only_sets$set,
              row_title = c("SET2", "SET3", "SET1", "SET4"))

Ha_bound_only
```

```{r}
Ha_bound_only +
  rowAnnotation(sets = bound_only_sets$set)

```

```{r}

bound_only_annotation <-
  merge(bound_only_elt2_clust_anno,
        bound_only_sets,
        by.x = "WBGeneID",
        by.y = "WBGeneID")
bound_only_annotation_ascend <-
  bound_only_annotation %>% arrange(WBGeneID)
head(bound_only_annotation_ascend)
```

```{r}
Ha_bound_only_chipClust <-
  Ha_bound_only + binding_cluster_row_annotation(bound_only_elt2_clust_anno)
Ha_bound_only_chipClust
```

# Fish out the genes that change the most and look for CGC stains

```{r}
cgc_strains <- readLines("./01_input/200608_cgc-strains.txt")

cgc_strains <- str_trim(cgc_strains)
head(cgc_strains, n = 40)

total_strains <- sum(str_count(cgc_strains, pattern = "-----------------------------------------------------------------------------"))

cgc_strains_df <- data.frame(
  strain = cgc_strains[str_which(cgc_strains, "Strain:")],
  species = cgc_strains[str_which(cgc_strains, "Species:")],
  genotype = cgc_strains[str_which(cgc_strains, "Genotype:")],
  description = cgc_strains[str_which(cgc_strains, "Description:")],
  mutagen = cgc_strains[str_which(cgc_strains, "Mutagen:")],
  outcrossed = cgc_strains[str_which(cgc_strains, "Outcrossed:")],
  made.by = cgc_strains[str_which(cgc_strains, "Made by:")],
  received = cgc_strains[str_which(cgc_strains, "Received:")],
  stringsAsFactors = FALSE,
  
)

counter = 1
cgc_strains_df <- data.frame(stringsAsFactors = FALSE)
for (i in 1:(total_strains)){
  to_append <- data.frame(string = str_c(c(cgc_strains[(counter+1):(counter+8)]), collapse = "\t"), stringsAsFactors = FALSE)
  cgc_strains_df <- bind_rows(cgc_strains_df, to_append)
  counter <- counter + 9
}

# cgc_strains_list <- str_split(cgc_strains_df$string, pattern = "\t")
# 
# str(cgc_strains_list)
# cgc_strains_df$string[str_detect(cgc_strains_df$string, pattern = "elt-2")]
```


```{r}
variance_per_gene <-  as.data.frame(matrix_select(dynamic_counts_matrix, l1_bound_list$WBGeneID)) %>% rownames_to_column(var = "WBGeneID") %>% 
    merge(bound_only_sets, by = "WBGeneID") %>% 
    pivot_longer(wt_sorted_1:elt2Delt7D_sorted_3, names_to = "sample", values_to = "rlogCount") %>%
    group_by(WBGeneID, set) %>%
    summarise(variance = var(rlogCount))

variance_per_gene <- getBM(
  mart = paramart,
  filter = c("wbps_gene_id"),
  value = variance_per_gene$WBGeneID,
  attributes = c('wbps_gene_id', 'external_gene_id')
) %>% inner_join(variance_per_gene, by = c("wbps_gene_id" = "WBGeneID"))

variance_per_gene %>% mutate(cgc_count = 
                               sum(str_count(cgc_strains_df$string, pattern = "elt-2"))
                             ) %>% arrange(desc(cgc_count))

cgc_count_column <- c()
for (i in variance_per_gene$external_gene_id){
  cgc_count_column <- append(cgc_count_column, sum(str_count(cgc_strains_df$string, pattern = paste(i, "(?![:digit:])", sep = "")
                                                             )))
}

variance_per_gene <- variance_per_gene %>% mutate(cgc_count = cgc_count_column)
```

SET1 genes are up in wildtype but down in elt-2 (-) and do not change with elt-7 (-) single or double. In other words, activated alone by ELT-2

```{r}
variance_per_gene %>% filter(set == "SET1") %>% arrange(desc(variance))
```

SET1 strains to order: strain CL2122 (mtl-2)

SET2 genes go up only in elt-2 (-)

```{r}
variance_per_gene %>% filter(set == "SET2") %>% arrange(desc(variance)) %>% filter(cgc_count > 1)
```

SET2 genes to order: OP201 (pqm-1)

SET3 genes go up in elt-2 (-) and go up a little higher in elt-2 (-);elt-7 (-)

```{r}
variance_per_gene %>% filter(set == "SET3") %>% arrange(desc(variance))
```

SET3 order: OP750 (gla-3)

SET4 genes only go down in elt-2 (-);elt-7 (-)

```{r}
variance_per_gene %>% filter(set == "SET4") %>% arrange(desc(variance))
```

SET4 cgc strain: OP509 (ets-4)

# L1 Intestine expression row annotation

Import RWC24 L1 Intestine results table

```{r}
RWC24_res <-
  read.csv(
    "../RWC24_L1_Intestine_RNAseq/04_DEseq2/200511_L1_intestine_FACS_gut_vs_gutless.csv",
    row.names = 1
  )
RWC24_res <- RWC24_res %>% rownames_to_column(var = "WBGeneID")
head(RWC24_res)
```

Select log fold change data for row annotation.  

```{r}
L1_gut_lfc <-
  dynamic_counts_matrix_scaled_ascend %>% as.data.frame.matrix() %>% rownames_to_column(var = "WBGeneID") %>% inner_join(RWC24_res, by = "WBGeneID") %>% select(WBGeneID, log2FoldChange)
```

Add to heatmap as a row annotation.  

```{r}
col_fun = colorRamp2(c(-10, 0, 10), c("blue", "white", "red"))

Ha_L1chip_L1gutLFC <-
  Ha_L1chip + rowAnnotation(L1_Gut_LFC = L1_gut_lfc$log2FoldChange,
                            col = list(L1_Gut_LFC = col_fun))
Ha_L1chip_L1gutLFC

# pdf(file = "./03_plots/200511_AnyDE_Genes_ELT2_ELT7_L1gutLFC.pdf", width = 5, height = 5)
# Ha_L1chip_L1gutLFC
# dev.off()
```

```{r}
RNA_heatmap(dynamic_counts_matrix_scaled_ascend,
            split = elt2_L1_anno$elt2_detected_in_L1) +
  rowAnnotation(L1_Gut_LFC = L1_gut_lfc$log2FoldChange,
                col = list(L1_Gut_LFC = col_fun))
```


# Intestine expression annotation for only ELT-2 bound genes

```{r}
bound_only_L1_gut_lfc <-
  dynamic_counts_matrix_scaled_bound_only %>% as.data.frame.matrix() %>% rownames_to_column(var = "WBGeneID") %>% inner_join(RWC24_res, by = "WBGeneID") %>% select(WBGeneID, log2FoldChange)

identical(
  bound_only_L1_gut_lfc$WBGeneID,
  rownames(dynamic_counts_matrix_scaled_bound_only)
)

Ha_bound_only + rowAnnotation(L1_Gut_LFC = bound_only_L1_gut_lfc$log2FoldChange,
                              col = list(L1_Gut_LFC = col_fun))
```

Next: better display of `L1_Gut_LFC`. Perform GO for the different clusters in this plot. Compare motifs of peaks assigned to genes within each cluster (or activated sets 1 and 2 and repressed sets 3 4 and 5). 

Make an MA plot for each set

```{r}
ggplot(
  RWC24_res %>% rownames_to_column(var = "WBGeneID") %>% inner_join(bound_only_sets, by = "WBGeneID"),
  aes(x = baseMean, y = log2FoldChange, color = pvalue < 0.01)
) +
  geom_point() +
  scale_x_log10() +
  facet_wrap(~ set)
```


# Make a TF subset heatmap

```{r, out.width= '\\textwidth', fig.width= 5, fig.height=15, fig.align='center'}

wTF3.0 <-
  read.csv("./01_input/TF3-0_namesonly.txt",
           sep = "\t",
           header = TRUE) %>% select(WBGeneID)

dynamic_counts_matrix_scaled_TFs <-
  matrix_select(dynamic_counts_matrix_scaled_ascend, wTF3.0$WBGeneID)

dynamic_counts_matrix_scaled_TFs_names <-
  id2name(dynamic_counts_matrix_scaled_TFs)

tf_heatmap <- Heatmap(
  dynamic_counts_matrix_scaled_TFs_names,
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = TRUE,
  show_column_names = TRUE,
  column_title = "Differential Expression of All Transcription Factors"
)
tf_heatmap
```

Add row annotation to indicate ELT-2 binding in L1 stage

```{r}
elt2_detected_in_L1 %>% filter(WBGeneID %in% rownames(dynamic_counts_matrix_scaled_TFs))

tf_bound_anno <-
  data.frame(
    WBGeneID = rownames(dynamic_counts_matrix_scaled_TFs),
    elt2_detected_in_L1 = ifelse(
      test = rownames(dynamic_counts_matrix_scaled_TFs) %in% elt2_detected_in_L1$WBGeneID,
      yes = "bound",
      no = "not.bound"
    )
  )

tf_heatmap_L1bound <-
  tf_heatmap +
  rowAnnotation(L1_bound = tf_bound_anno$elt2_detected_in_L1,
                col = list(L1_bound = c(
                  "bound" = "green", "not.bound" = "black"
                )))
tf_heatmap_L1bound
```

```{r}
RNA_heatmap(dynamic_counts_matrix_scaled_TFs_names,
            split = tf_bound_anno$elt2_detected_in_L1) +
  rowAnnotation(L1_bound = tf_bound_anno$elt2_detected_in_L1,
                col = list(L1_bound = c(
                  "bound" = "green", "not.bound" = "black"
                )))
```


Add row annotation of intestine expression

```{r}
tf_lfc <-
  L1_gut_lfc %>% filter(WBGeneID %in% rownames(dynamic_counts_matrix_scaled_TFs)) %>% select(log2FoldChange)

tf_heatmap_L1bound +
  rowAnnotation(log2FC = tf_lfc$log2FoldChange,
                col = list(log2FC = colorRamp2(c(-10, 0, 10), c(
                  "blue", "white", "red"
                ))))
```

```{r}
RNA_heatmap(dynamic_counts_matrix_scaled_TFs_names,
            split = tf_bound_anno$elt2_detected_in_L1) +
  rowAnnotation(L1_bound = tf_bound_anno$elt2_detected_in_L1,
                col = list(L1_bound = c(
                  "bound" = "green", "not.bound" = "black"
                ))) +  rowAnnotation(log2FC = tf_lfc$log2FoldChange,
                                     col = list(log2FC = colorRamp2(c(-10, 0, 10), c(
                                       "blue", "white", "red"
                                     ))))
```


Zoom in on only bound TFs

```{r}
dynamic_counts_matrix_scaled_TFs_bound <-
  matrix_select(dynamic_counts_matrix_scaled_TFs,
                elt2_detected_in_L1$WBGeneID)

dynamic_counts_matrix_scaled_TFs_bound_names <-
  id2name(dynamic_counts_matrix_scaled_TFs_bound)

bound_tf_lfc <-
  L1_gut_lfc %>% filter(WBGeneID %in% rownames(dynamic_counts_matrix_scaled_TFs_bound)) %>% select(log2FoldChange)

head(bound_tf_lfc)

HAboundTF <- Heatmap(
  dynamic_counts_matrix_scaled_TFs_bound_names,
  col = colorRampPalette(c("cyan", "black", "yellow"))(1000),
  cluster_columns = FALSE,
  clustering_distance_rows = "spearman",
  clustering_method_rows = "complete",
  show_row_names = TRUE,
  row_names_side = "left",
  show_column_names = TRUE,
  column_title = "Differential Expression of\nELT-2 bound Transcription Factors"
)

```

This plot suggests that transcription factors bound by ELT-2 are typically upregulated in the absence of ELT-2. 

Additionally, TFs that are expressed in the L1 intestine are upregulated in absence of ELT-2 alone, but downregulated in the absence of both ELT-2 and ELT-7.  

Futhermore, TFs that are not expressed in the L1 intestine are upregulated only in the absence of both ELT-2 and ELT-7.  

TFs to follow up: pqm-1, zip-10, odd-1 (repressed by elt-2 alone, normally gut expressed). nhr-58 (vulva), zip-2 (neuron), cebp-1 (neuron), gla-3 (germline), zip-11



# Conclusions

Figure 1A

```{r}

```




# old code below

## Transcription factor subset plots

<!-- Load in wTF3.0 list -->

<!-- ```{r, out.width= '\\textwidth', fig.height= 4.5, fig.align='center'} -->
<!-- wTF3.0 <- -->
<!--   read.csv("./01_input/TF3-0_namesonly.txt", -->
<!--            sep = "\t", -->
<!--            header = TRUE) %>% select(WBGeneID) -->

<!-- dynamic_counts_matrix_TFs <- -->
<!--   matrix_select(dynamic_counts_matrix, wTF3.0$WBGeneID) -->
<!-- dynamic_counts_matrix_TFs_bound <- -->
<!--   matrix_select(dynamic_counts_matrix_scaled_TFs, -->
<!--                 elt2_detected_in_L1$WBGeneID) -->

<!-- dynamic_counts_matrix_TFs_bound_sub <- -->
<!--   subset( -->
<!--     dynamic_counts_matrix_TFs_bound, -->
<!--     select = c( -->
<!--       "wt_sorted_1", -->
<!--       "wt_sorted_2", -->
<!--       "wt_sorted_3", -->
<!--       "wt_sorted_4", -->
<!--       "elt2D_sorted_1", -->
<!--       "elt2D_sorted_2", -->
<!--       "elt2D_sorted_3", -->
<!--       "elt2D_sorted_4" -->
<!--     ) -->
<!--   ) -->

<!-- dynamic_counts_matrix_TFs_bound_sub_scale <- -->
<!--   t(apply(unlist(dynamic_counts_matrix_TFs_bound_sub), 1, scale)) -->

<!-- # varselect <- rowVars(colsub) >= 0.5 -->
<!-- # colsub <- colsub[varselect, ] -->

<!-- dynamic_counts_matrix_TFs_bound_sub_scale_names <- -->
<!--   id2name(dynamic_counts_matrix_TFs_bound_sub_scale) -->

<!-- bound_TF_heatmap <- -->
<!--   Heatmap( -->
<!--     dynamic_counts_matrix_TFs_bound_sub_scale_names, -->
<!--     col = colorRampPalette(c("cyan", "black", "yellow"))(1000), -->
<!--     cluster_columns = FALSE, -->
<!--     #clustering_distance_rows = "spearman", -->
<!--     row_dend_reorder = TRUE, -->
<!--     clustering_method_rows = "complete", -->
<!--     show_row_names = TRUE, -->
<!--     show_column_names = TRUE, -->
<!--     row_names_gp = gpar(fontsize = 10), -->
<!--     column_names_gp = gpar(cex = 0.4), -->
<!--     heatmap_legend_param = list(color_bar = "continuous"), -->
<!--     #split = 2, -->
<!--     width = unit(3, "in"), -->
<!--     height = unit(4, "in") -->
<!--   ) -->
<!-- bound_TF_heatmap -->

<!-- pdf("./03_plots/200428_TF_only_L1elt2bound", -->
<!--     height = 5, -->
<!--     width = 5) -->
<!-- bound_TF_heatmap -->
<!-- dev.off() -->
<!-- ``` -->

<!-- Add intestine expressed annotation.   -->

<!-- ```{r} -->
<!-- bound_TF_rna_anno <- data.frame( -->
<!--   spencerLE = ifelse( -->
<!--     test = rownames(dynamic_counts_matrix_TFs_bound_sub_scale) %in% spencer_LE_subset$spencer_LE_ID, -->
<!--     yes = "enriched", -->
<!--     no = "depleted" -->
<!--   ), -->
<!--   spencerL2 = ifelse( -->
<!--     test = rownames(dynamic_counts_matrix_TFs_bound_sub_scale) %in% spencer_L2_subset$spencer_L2_ID, -->
<!--     yes = "enriched", -->
<!--     no = "depleted" -->
<!--   ) -->
<!-- )  -->

<!-- bound_TF_heatmap + -->
<!--     rowAnnotation(LE.intestine = bound_TF_rna_anno$spencerLE, col = list(LE.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE) + -->
<!--   rowAnnotation(L2.intestine = bound_TF_rna_anno$spencerL2, col = list(L2.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE) -->
<!-- ``` -->



<!-- ```{r, out.width= '\\textwidth', fig.height= 3.5, fig.align='center'} -->
<!-- dynamic_counts_matrix_TFs <- matrix_select(dynamic_counts_matrix, wTF3.0$WBGeneID) -->
<!-- dynamic_counts_matrix_TFs_sub <- subset(dynamic_counts_matrix_TFs, select = c("wt_sorted_1", "wt_sorted_2", "wt_sorted_3", "wt_sorted_4", "elt2D_sorted_1", "elt2D_sorted_2", "elt2D_sorted_3", "elt2D_sorted_4")) -->
<!-- dynamic_counts_matrix_TFs_sub_scale <- t(apply(unlist(dynamic_counts_matrix_TFs_sub), 1, scale))  -->
<!-- dynamic_counts_matrix_TFs_sub_scale_names <- id2name(dynamic_counts_matrix_TFs_sub_scale) -->
<!-- Ha_TF <- Heatmap(dynamic_counts_matrix_TFs_sub_scale_names,  -->
<!--           col = colorRampPalette(c("cyan","black","yellow"))(1000),     -->
<!--           cluster_columns = FALSE, -->
<!--           #clustering_distance_rows = "spearman", -->
<!--           #row_dend_reorder = TRUE, -->
<!--           clustering_method_rows = "complete",  -->
<!--           show_row_names = TRUE, -->
<!--           show_column_names = TRUE, -->
<!--           row_names_gp = gpar(cex = 0.4), -->
<!--           column_names_gp = gpar(cex = 0.4), -->
<!--           heatmap_legend_param = list(color_bar = "continuous"), -->
<!--           #split = 2, -->
<!--            width = unit(2, "in"),  -->
<!--           # height = unit(3, "in"), -->
<!--           split = 2) -->
<!-- Ha_TF -->

<!-- # svg("./killme.svg", width = 8, height = 8) -->
<!-- # Ha_TF -->
<!-- # dev.off() -->
<!-- ``` -->

<!-- ```{r} -->
<!-- TF_anno <- data.frame(elt2_detected_in_L1 = ifelse(test = rownames(dynamic_counts_matrix_TFs_sub_scale) %in% elt2_detected_in_L1$WBGeneID, yes = "bound", no = "not.bound"))  -->

<!-- Ha_TF + rowAnnotation(L1bound = TF_anno$elt2_detected_in_L1, -->
<!--                    col = list(L1bound = c("bound" = "green", "not.bound" = "black"))) -->

<!-- # pdf("./tf_heatmap.pdf", height = 8, width = 8) -->
<!-- # Ha_TF + rowAnnotation(L1bound = TF_anno$elt2_detected_in_L1, -->
<!-- #                    col = list(L1bound = c("bound" = "green", "not.bound" = "black"))) -->
<!-- # dev.off() -->
<!-- ``` -->

<!-- Add intestine expression row annotation.   -->

<!-- ```{r} -->
<!-- TF_rna_anno <- data.frame( -->
<!--   spencerLE = ifelse(test = rownames(dynamic_counts_matrix_TFs_sub_scale) %in% spencer_LE_subset$spencer_LE_ID, yes = "enriched", no = "depleted"), -->
<!--   spencerL2 = ifelse(test = rownames(dynamic_counts_matrix_TFs_sub_scale) %in% spencer_L2_subset$spencer_L2_ID, yes = "enriched", no = "depleted") -->
<!--     )  -->

<!-- Ha_TF_spencer <- Ha_TF + rowAnnotation(L1bound = TF_anno$elt2_detected_in_L1, -->
<!--                    col = list(L1bound = c("bound" = "green", "not.bound" = "black"))) + -->
<!--     rowAnnotation(LE.intestine = TF_rna_anno$spencerLE, col = list(LE.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE) + -->
<!--   rowAnnotation(L2.intestine = TF_rna_anno$spencerL2, col = list(L2.intestine = c("enriched" = "blue", "depleted" = "white")), border = TRUE) -->

<!-- # pdf("./03_plots/200428_DE_TFs_elt2chip_intestineExpression", height = 8, width = 8) -->
<!-- # Ha_TF_spencer -->
<!-- # dev.off() -->

<!-- ``` -->

## Results and interpretation

## Session Info

```{r}
sessionInfo()
```

