---
title: "RWC23_ELT2_ChIP_Boeck_Time_Resolved_RNA"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Note: Ensure BioConductor is version 3.10 or above

Install libraries

```{r}
# fill this in
```

Note: you must load `biomaRt` before loading `tidyverse`

Load libraries
```{r}
library(biomaRt)
library(tidyverse)
library(ComplexHeatmap)
library(readxl)
```

Load custom functions

```{r}
source("RWC23_Functions.R")
```


Pseudocode:
- Bring in Boeck Data
- Translate to WBGeneID
- Filter for ELT-2 ChIP bound genes, make heatmap
- Filter for intestine expressed genes (spencer data), make heatmap, add row annotation for binding cluster

# Import Time-resolved RNA

```{r}
time_resolved_rna <- read.delim("~/Dropbox/01_GITHUBREPO/RWC23_elt2_regulated_genes/02_Data/9_Boeck_et_al_2016_time-resolved_transcriptome/Unified_dcpm_per_wormbase_gene.txt", quote="", stringsAsFactors=FALSE)

paramart <- useMart("parasite_mart", dataset = "wbps_gene", host = "https://parasite.wormbase.org", port = 443)

time_resolved_rna <- getBM(
  mart = paramart,
  filter = c("wormbase_gseqname"),
  value = time_resolved_rna$WormbaseName,
  attributes = c("wormbase_gseq","wbps_gene_id", "wikigene_name")
) %>% right_join(time_resolved_rna, by = c("wormbase_gseq" = "WormbaseName"))


time_resolved_rna <- time_resolved_rna %>% drop_na(wbps_gene_id)
```

# Load in public intestine expression datasets

Initalize an empty dataframe.

```{r}
intestine_rna <- data.frame(WBGeneID = time_resolved_rna$wbps_gene_id, Sequence.name = time_resolved_rna$wormbase_gseq, Public_name = time_resolved_rna$wikigene_name)
```

Initalize a seprate empty "metadata" dataframe to store descriptive information

```{r}
RWC23_Metadata <- data.frame(column_ID = "", column_detail = "", source_DOI = "", source_detail = "", seq_method = "")
```

# A Transcriptional Lineage of the Early C . elegans Embryo.
## Tintori SC, Nishimura EO, Golden P, Lieb JD, Goldstein B. Dev Cell. 2016;38(4):430-444. doi:10.1101/047746

```{r}
# Transcripts from embryonic 2Ep intestine from Sophie's single cell RNA-seq paper. See `intestine_tfs.Rmd`
tintori_2Ep <- read.csv("./02_Data/7_Tintori_et_al_2016_Embryo_scRNA-seq/16cell_vs_Ep_p1_00.csv", header = FALSE, stringsAsFactors = FALSE)
tintori_columns <- c("gene", "p.value", "log.CPM", "log.FC")
colnames(tintori_2Ep) <- str_c("tintori_2Ep_", tintori_columns)

intestine_rna <- intestine_rna %>% left_join(tintori_2Ep, by = c("Sequence.name" = "tintori_2Ep_gene"))

# Add boolian column such that it is true if log(FC) is greater than zero and p-value less than 0.05
tintori_2Ep_enriched <- intestine_rna %>% filter(intestine_rna$tintori_2Ep_p.value <= 0.05 & intestine_rna$tintori_2Ep_log.FC < 0)

intestine_rna <- intestine_rna %>% mutate(tintori_2Ep_bool = if_else(intestine_rna$WBGeneID %in% tintori_2Ep_enriched$WBGeneID, TRUE, FALSE))

# Update metadata

RWC23_Metadata <- RWC23_Metadata %>% add_row(
  column_ID = colnames(tintori_2Ep[1]),
  column_detail = "Gene sequence name",
  source_DOI = "10.1016/j.devcel.2016.07.025",
  source_detail = "Single cell RNA-seq of hand dissected embryo. Accessed from tintori.bio.unc.edu.",
  seq_method = "Illumina"
  ) %>% add_row(
  column_ID = colnames(tintori_2Ep[2]),
  column_detail = "Gene p-value",
  source_DOI = "10.1016/j.devcel.2016.07.025",
  source_detail = "Single cell RNA-seq of hand dissected embryo. Accessed from tintori.bio.unc.edu.",
  seq_method = "Illumina"
  ) %>% add_row(
  column_ID = colnames(tintori_2Ep[3]),
  column_detail = "Counts per transcript per million mapped reads.",
  source_DOI = "10.1016/j.devcel.2016.07.025",
  source_detail = "Single cell RNA-seq of hand dissected embryo. Accessed from tintori.bio.unc.edu.",
  seq_method = "Illumina"
  ) %>% add_row(
  column_ID = colnames(tintori_2Ep[4]),
  column_detail = "Log2(Fold Change) of gene in the 2Ep cell when compared to the whole embryo. Less than zero is enriched in the 1E cell. Greater than zero is enriched in the whole 16 cell embryo",
  source_DOI = "10.1016/j.devcel.2016.07.025",
  source_detail = "Single cell RNA-seq of hand dissected embryo. Accessed from tintori.bio.unc.edu.",
  seq_method = "Illumina"
  )
```

# Spatiotemporal transcriptomics reveals the evolutionary history of the endoderm germ layer.
## Hashimshony T, Feder M, Levin M, Hall BK, Yanai I.  Nature. 2015;519(7542):219-222. doi:10.1038/nature13996

```{r}
# TFs from Hashimshony endoderm genes
hashimshony_embryo <- read.csv("./02_Data/1_Hashimshony_et_al_2015_CEL-seq/Hashimshony_Endoderm_Genes.csv", header = TRUE)
colnames(hashimshony_embryo) <- c("hashimshony_embryo_genes")
intestine_rna <- intestine_rna %>% mutate(hashimshony_embryo_bool = if_else(intestine_rna$Public_name %in% hashimshony_embryo$hashimshony_embryo_genes, TRUE, FALSE))

# Update metadata

RWC23_Metadata <- RWC23_Metadata %>% add_row(
  column_ID = colnames(hashimshony_embryo[1]),
  column_detail = "Gene public name",
  source_DOI = "10.1038/nature13996",
  source_detail = "Supplementary Table 4. Endoderm gene set from Single cell RNA-seq of cultured dissociated embryos.",
  seq_method = "Illumina"
  )

```

# Chromosomal clustering and GATA transcriptional regulation of intestine-expressed genes in C. elegans.
## Pauli F, Liu Y, Kim Y a, Chen P-J, Kim SK. Development. 2006;133(2):287-295. doi:10.1242/dev.02185

```{r}
# TFs from Pauli et al 
pauliGenes <- read_excel("./02_Data/4_Pauli_et_al_2006_PAB-1_Pulldown_microarray/SupTable1_intestine_enriched_genes.xlsx")
colnames(pauliGenes) <- str_c("pauli_L4_", c("gene", "percentile", "pvalue", "description"))

# add to master datatable

intestine_rna <- intestine_rna %>% left_join(pauliGenes, by = c("Sequence.name" = "pauli_L4_gene"))

# make boolian column
intestine_rna <- intestine_rna %>% mutate(pauli_L4_bool = if_else(intestine_rna$Sequence.name %in% pauliGenes$pauli_L4_gene, TRUE, FALSE))

# update metadata
RWC23_Metadata <- RWC23_Metadata  %>% add_row(
  column_ID = colnames(pauliGenes[1]),
  column_detail = "Sequence name",
  source_DOI = "10.1242/dev.02185",
  source_detail = "Supplementary Table 1. Microarray of ges-1P::PAB-1 pulldown in L4 worms"
  ) %>% add_row(
  column_ID = colnames(pauliGenes[2]),
  column_detail = "Percentile ranking of transcript abundance",
  source_DOI = "10.1242/dev.02185",
  source_detail = "Supplementary Table 1. Microarray of ges-1P::PAB-1 pulldown in L4 worms"
  ) %>% add_row(
  column_ID = colnames(pauliGenes[3]),
  column_detail = "Student's T-test to calculate significant enrichment of intestine transcripts",
  source_DOI = "10.1242/dev.02185",
  source_detail = "Supplementary Table 1. Microarray of ges-1P::PAB-1 pulldown in L4 worms"
  ) %>% add_row(
  column_ID = colnames(pauliGenes[4]),
  column_detail = "Short descriptions generated by WormBase",
  source_DOI = "10.1242/dev.02185",
  source_detail = "Supplementary Table 1. Microarray of ges-1P::PAB-1 pulldown in L4 worms"
  ) %>% add_row(
  column_ID = "pauli_L4_bool",
  column_detail = "True if included on the list, false if absent on the list",
  source_DOI = "10.1242/dev.02185",
  source_detail = "Supplementary Table 1. Microarray of ges-1P::PAB-1 pulldown in L4 worms"
  ) 

# add sequencing method column

RWC23_Metadata <- mutate(RWC23_Metadata, seq_method = ifelse(RWC23_Metadata$column_ID %in% c(colnames(pauliGenes), "pauli_L4_bool"), "Microarray",RWC23_Metadata$seq_method))

# subset the pauli rows with `ifelse` and `%in%`, replace the NA in `seq_method` with "Microarray"
# if the value does not come from pauli, then place the original string from RWC23_Metadata

```

# A spatial and temporal map of C. elegans gene expression. Genome Research, 21(2), 325–41. https://doi.org/10.1101/gr.114595.110 
## Spencer, W. C., Zeller, G., Watson, J. D., Henz, S. R., Watkins, K. L., McWhirter, R. D., … Miller, D. M. (2011).

```{r}

# Late embryo intestine from Spencer et al
spencerLEgenes <- read.table("./02_Data/6_Spencer_et_al_2010_FACS_and_pulldown_tilling_array/LE-intestine_enr_vs_ref.WS200.txt", quote="\"", comment.char="", header = TRUE)
colnames(spencerLEgenes) <- str_c("spencer_LE_", colnames(spencerLEgenes))
spencer_LE_subset <- spencerLEgenes %>% select(spencer_LE_ID, spencer_LE_AveExpr, spencer_LE_adj_P_Val, spencer_LE_FC)

# add to dataframe
intestine_rna <- merge(x = intestine_rna, y = spencer_LE_subset, by.x = 'WBGeneID', by.y = 'spencer_LE_ID', all.x = TRUE)

# create true/false if gene is on list
intestine_rna <- intestine_rna %>% mutate(spencer_LE_bool = if_else(intestine_rna$WBGeneID %in% spencer_LE_subset$spencer_LE_ID, TRUE, FALSE))


# L2 intestine genes from Spencer et al
spencerL2genes <- read.table("./02_Data/6_Spencer_et_al_2010_FACS_and_pulldown_tilling_array/L2-intestine_enr_vs_ref.WS200.txt", quote="\"", comment.char="", header = TRUE)
colnames(spencerL2genes) <- str_c("spencer_L2_", colnames(spencerL2genes))
spencer_L2_subset <- spencerL2genes %>% select(spencer_L2_ID, spencer_L2_AveExpr, spencer_L2_adj_P_Val, spencer_L2_FC)

intestine_rna <- merge(x = intestine_rna, y = spencer_L2_subset, by.x = 'WBGeneID', by.y = 'spencer_L2_ID', all.x = TRUE)
intestine_rna <- intestine_rna %>% mutate(spencer_L2_bool = if_else(intestine_rna$WBGeneID %in% spencer_L2_subset$spencer_L2_ID, TRUE, FALSE))

# Update metadata

RWC23_Metadata <- RWC23_Metadata  %>% add_row(
  column_ID = colnames(spencer_LE_subset[1]),
  column_detail = "Wormbase Gene ID",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = colnames(spencer_LE_subset[2]),
  column_detail = "??",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = colnames(spencer_LE_subset[3]),
  column_detail = "??",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = colnames(spencer_LE_subset[4]),
  column_detail = "??",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = "spencer_LE_bool",
  column_detail = "True if present in this list",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  )

RWC23_Metadata <- mutate(RWC23_Metadata, seq_method = ifelse(RWC23_Metadata$column_ID %in% c(colnames(spencer_LE_subset), "spencer_LE_bool"), "Microarray",RWC23_Metadata$seq_method))

RWC23_Metadata <- RWC23_Metadata  %>% add_row(
  column_ID = colnames(spencer_L2_subset[1]),
  column_detail = "Wormbase Gene ID",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = colnames(spencer_L2_subset[2]),
  column_detail = "??",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = colnames(spencer_L2_subset[3]),
  column_detail = "??",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = colnames(spencer_L2_subset[4]),
  column_detail = "??",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  ) %>% add_row(
  column_ID = "spencer_L2_bool",
  column_detail = "True if present in this list",
  source_DOI = "10.1101/gr.114595.110",
  source_detail = "FACS Sorted ELT-2::GFP worms. https://www.vanderbilt.edu/wormdoc/wormmap"
  )

RWC23_Metadata <- mutate(RWC23_Metadata, seq_method = ifelse(RWC23_Metadata$column_ID %in% c(colnames(spencer_L2_subset), "spencer_L2_bool"), "Microarray",RWC23_Metadata$seq_method))
```

Make a vector that contains intestine expressed genes from any stage

```{r}
intestine_gene_list <- intestine_rna %>% filter(
  tintori_2Ep_bool == TRUE |
    hashimshony_embryo_bool == TRUE |
    pauli_L4_bool == TRUE |
    spencer_LE_bool == TRUE |
    spencer_L2_bool == TRUE
) %>% select(WBGeneID)

dim(intestine_gene_list)
```

Import wTF3.0 worm transcription factor database

```{r}
wTF3.0 <-
  read.csv("./01_input/TF3-0_namesonly.txt",
           sep = "\t",
           header = TRUE) %>% select(WBGeneID)
```


# Filter time-resolved RNA-seq based on intestine expression

```{r}
time_resolved_rna_intestine_df <- time_resolved_rna %>%
  remove_rownames() %>%
  arrange(wbps_gene_id) %>%
  filter(wbps_gene_id %in% intestine_gene_list$WBGeneID) %>%
  select(
    -(emb_4cell:emb_471min),
    -DE,
    -D,
    -DX,
    -Soma,
    -Male,
    -AdultSPE9,
    -gonad,
    -LENGTH
  )
head(time_resolved_rna_intestine_df)
```


```{r}
time_resolved_rna_intestine_matrix <- time_resolved_rna_intestine_df %>%
  select(-wormbase_gseq,-wikigene_name) %>%
  remove_rownames() %>%
  arrange(wbps_gene_id) %>%
  column_to_rownames(var = "wbps_gene_id") %>%
  as.matrix()
head(time_resolved_rna_matrix_intestine)
```

Perform row normalization

```{r}
time_resolved_rna_intestine_matrix_scaled <- t(apply(unlist(time_resolved_rna_intestine_matrix), 1, scale))
colnames(time_resolved_rna_intestine_matrix_scaled) <- colnames(time_resolved_rna_intestine_matrix)
```

Store index of relavent genes for row annotations. Use custom function 

```{r}
gene_names <- c("end-1", "elt-2", "elt-7", "elt-4", "pqm-1", "mtl-2", "ets-4")
GOI_df <- GOI_annotate_heatmap(gene_names, time_resolved_rna_intestine_df$wikigene_name)
GOI_df
```


```{r}
time_resolved_rna_intestine_df %>% filter(wikigene_name %in% GOI_df$name)
```


```{r}
Boeck_intestine_RNA <- Heatmap(time_resolved_rna_intestine_matrix_scaled,
        cluster_columns = FALSE,
        show_row_names = FALSE,
        row_km = 5) +
  rowAnnotation(foo = anno_mark(GOI_df$index, labels = GOI_df$name))
Boeck_intestine_RNA
# pdf(file = "./03_plots/200915_Boeck_RNA_Intestine.pdf", width = 7, height = 7)
# Boeck_intestine_RNA
# dev.off()
```

Filter heatmap for only transcription factors. This is very ugly, fix later.

```{r}
time_resolved_rna_intestine_matrix_scaled_TFONLY <-
  matrix_select(time_resolved_rna_intestine_matrix_scaled, wTF3.0$WBGeneID)

tf_GOI_df <- GOI_df %>% left_join(time_resolved_rna_intestine_df, by = c("name" = "wikigene_name")) %>% select(name:wbps_gene_id, -index) %>% filter(wbps_gene_id %in% wTF3.0$WBGeneID) 
tf_GOI_df
tf_GOI_df<-GOI_annotate_heatmap(tf_GOI_df$wbps_gene_id, rownames(time_resolved_rna_intestine_matrix_scaled_TFONLY)) %>% full_join(tf_GOI_df, by = c("name" = "wbps_gene_id"))

```


```{r}
Heatmap(
  time_resolved_rna_intestine_matrix_scaled_TFONLY,
  cluster_columns = FALSE,
  show_row_names = FALSE,
  row_km = 5
) +
  rowAnnotation(foo = anno_mark(
    at = tf_GOI_df$index,
    labels = tf_GOI_df$name.y
  ))
```


# Import ELT-2 ChIP-seq binding data

```{r}
ELT2_GRange <- readRDS(file = "./01_input/200719_annotatedPeaks.rds")
chip_df <- mcols(ELT2_GRange)
head(chip_df)
```

# Subset ELT-2 ChIP with literature Intestine Expression

Do this earlier in the code to have k4labels stored in the time_resolved_rna dataframe and subsequent subsetting

```{r}
 chip_rna_df <-
   as.data.frame(chip_df) %>% remove_rownames() %>%
   select(name, k4labels, feature)  %>%
   right_join(time_resolved_rna_intestine_df, by = c("feature" = "wbps_gene_id")) %>%
   replace_na(list("k4labels" = "Not_Bound", "name" = "Not_Bound"))

 chip_rna_df$k4labels <-
   factor(
     chip_rna_df$k4labels,
     levels = c(
       "LE-specific",
       "Post-embryonic",
       "Increasing",
       "L3-high",
       "Not-changing or not IDR-passing",
       "Not_Bound"
     ),
     labels = c(
       "Embryo_Specific",
       "Larval",
       "Increasing",
       "L3_High",
       "Not_Changing",
       "Not_Bound"
     )
   )
```

# Subset heatmap based on ELT-2 binding pattern




```{r}
chip_rna_matrix <- chip_rna_df %>% select(emb_510min:YA) %>% as.matrix() 
rownames(chip_rna_matrix) <- chip_rna_df$wikigene_name
chip_rna_matrix_scaled <- row_scale(chip_rna_matrix)
head(chip_rna_matrix_scaled)
```
```{r}
which(rownames(chip_rna_matrix_scaled) == "mtl-2")
```

```{r}
for (name in gene_names){
  index <- which(rownames(chip_rna_matrix_scaled) == name)
  for (i in 1:length(index)){
    print(c(name, index[i]))
  }
}
```



```{r}
chip_GOI_df <- data.frame(name = NULL, index = NULL)
for (name in gene_names){
  index <- which(rownames(chip_rna_matrix_scaled) == name)
  for (i in 1:length(index)){
    toappend <- data.frame(name = name, index = index[i])
    chip_GOI_df <- bind_rows(chip_GOI_df, toappend)
  }
}
chip_GOI_df
```

```{r}
BoeckRNA_ELT2_chip <- Heatmap(chip_rna_matrix_scaled,
        name = "Boeck Time Resolved RNA",
        row_split = chip_rna_df$k4labels,
        row_title = NULL,
        cluster_columns = FALSE) +
  rowAnnotation(ELT2_peak = chip_rna_df$k4labels, 
                col = list(ELT2_peak = c(
      "Embryo_Specific" = "#7570B3",
      "Larval" = "#1B9E77",
      "Increasing" = "#E7298A",
      "L3_High" = "#D95F02",
      "Not_Changing" = "#505050",
      "Not_Bound" = "white"
    )),
    border = TRUE
  ) +
  rowAnnotation(foo = anno_mark(
    at = chip_GOI_df$index,
    labels = chip_GOI_df$name
  )
  )
BoeckRNA_ELT2_chip

# pdf(file = "./03_plots/200915_Boeck_RNA_ELT2ChIP_Patterns.pdf", width = 7, height = 7)
# BoeckRNA_ELT2_chip
# dev.off()
```


