---
title: "RWC23_ELT2_ChIP_Boeck_Time_Resolved_RNA"
output:
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Note: Ensure BioConductor is version 3.10 or above

Install libraries

```{r}
# fill this in
```

Note: you must load `biomaRt` before loading `tidyverse`

Load libraries
```{r}
library(biomaRt)
library(tidyverse)
library(ComplexHeatmap)

```

Load custom functions

```{r}
source("../RWC23_Functions.R")
```


Pseudocode:
- Bring in Boeck Data
- Translate to WBGeneID
- Filter for ELT-2 ChIP bound genes, make heatmap
- Filter for intestine expressed genes (spencer data), make heatmap, add row annotation for binding cluster

# Import Time-resolved RNA

```{r}
time_resolved_rna <-
  read.delim(
    "../02_Public_Intesine_RNA/01_input/9_Boeck_et_al_2016_time-resolved_transcriptome/Unified_dcpm_per_wormbase_gene.txt",
    quote = "",
    stringsAsFactors = FALSE
  )

paramart <-
  useMart("parasite_mart",
          dataset = "wbps_gene",
          host = "https://parasite.wormbase.org",
          port = 443)

time_resolved_rna <- getBM(
  mart = paramart,
  filter = c("wormbase_gseqname"),
  value = time_resolved_rna$WormbaseName,
  attributes = c("wormbase_gseq", "wbps_gene_id", "wikigene_name")
) %>% right_join(time_resolved_rna, by = c("wormbase_gseq" = "WormbaseName"))


time_resolved_rna <- time_resolved_rna %>% drop_na(wbps_gene_id)
```

```{r}
intestine_gene_list <-
  read_csv("../02_Public_Intesine_RNA/02_output/RWC23_Public_Intestine_RNA_Data.csv")
```


# Import wTF3.0 worm transcription factor database

```{r}
wTF3.0 <-
  read.csv(
    "../01_ChIPseq_RNAseq_Integration/01_input/TF3-0_namesonly.txt",
    sep = "\t",
    header = TRUE
  ) %>% select(WBGeneID)
```


# Filter time-resolved RNA-seq based on intestine expression

```{r}
time_resolved_rna_intestine_df <- time_resolved_rna %>%
  remove_rownames() %>%
  arrange(wbps_gene_id) %>%
  filter(wbps_gene_id %in% intestine_gene_list$WBGeneID) %>%
  select(-(emb_4cell:emb_471min),-DE,-D,-DX,-Soma,-Male,-AdultSPE9,-gonad,-LENGTH)
head(time_resolved_rna_intestine_df)
```


```{r}
time_resolved_rna_intestine_matrix <-
  time_resolved_rna_intestine_df %>%
  select(-wormbase_gseq, -wikigene_name) %>%
  remove_rownames() %>%
  arrange(wbps_gene_id) %>%
  column_to_rownames(var = "wbps_gene_id") %>%
  as.matrix()
head(time_resolved_rna_intestine_matrix)
```
```{r}


rowvariances=apply(time_resolved_rna_intestine_matrix,1,var)
plot(ecdf(rowvariances))
```
Perform row normalization

```{r}
time_resolved_rna_intestine_matrix_scaled <-
  t(apply(unlist(time_resolved_rna_intestine_matrix), 1, scale))
colnames(time_resolved_rna_intestine_matrix_scaled) <-
  colnames(time_resolved_rna_intestine_matrix)
```

Store index of relavent genes for row annotations. Use custom function 

```{r}
gene_names <-
  c("elt-2", "elt-7", "elt-4", "pqm-1", "mtl-2", "ets-4", "aat-6")
GOI_df <-
  GOI_annotate_heatmap(gene_names, time_resolved_rna_intestine_df$wikigene_name)
GOI_df
```


```{r}
time_resolved_rna_intestine_df %>% filter(wikigene_name %in% GOI_df$name)
```


```{r}
Boeck_intestine_RNA <-
  Heatmap(
    time_resolved_rna_intestine_matrix_scaled,
    cluster_columns = FALSE,
    show_row_names = FALSE,
    row_km = 5
  ) +
  rowAnnotation(foo = anno_mark(GOI_df$index, labels = GOI_df$name))
Boeck_intestine_RNA
# pdf(file = "./03_plots/200915_Boeck_RNA_Intestine.pdf", width = 7, height = 7)
# Boeck_intestine_RNA
# dev.off()
```

Filter heatmap for only transcription factors. This is very ugly, fix later.

```{r}
time_resolved_rna_intestine_matrix_scaled_TFONLY <-
  matrix_select(time_resolved_rna_intestine_matrix_scaled, wTF3.0$WBGeneID)

tf_GOI_df <-
  GOI_df %>% 
  left_join(time_resolved_rna_intestine_df, by = c("name" = "wikigene_name")) %>% 
  select(name:wbps_gene_id, -index) %>% filter(wbps_gene_id %in% wTF3.0$WBGeneID)
tf_GOI_df
tf_GOI_df <-
  GOI_annotate_heatmap(
    tf_GOI_df$wbps_gene_id,
    rownames(time_resolved_rna_intestine_matrix_scaled_TFONLY)
  ) %>% full_join(tf_GOI_df, by = c("name" = "wbps_gene_id"))

```


```{r}
Heatmap(
  time_resolved_rna_intestine_matrix_scaled_TFONLY,
  cluster_columns = FALSE,
  show_row_names = FALSE,
  row_km = 5
) +
  rowAnnotation(foo = anno_mark(at = tf_GOI_df$index,
                                labels = tf_GOI_df$name.y))
```


# Import ELT-2 ChIP-seq binding data

```{r}
chip_df <-
  read_csv(file = "../01_ChIPseq_RNAseq_Integration/01_input/200719_annotatedPeaks.csv")
head(chip_df)
```

# Subset ELT-2 ChIP with literature Intestine Expression

Do this earlier in the code to have k4labels stored in the time_resolved_rna dataframe and subsequent subsetting

```{r}
chip_rna_df <- chip_df %>%
  select(name, cluster.description, WBGeneID)  %>%
  right_join(time_resolved_rna_intestine_df,
             by = c("WBGeneID" = "wbps_gene_id")) %>%
  replace_na(list("cluster.description" = "Not_Bound", "name" = "Not_Bound"))

chip_rna_df$cluster.description <-
  factor(
    chip_rna_df$cluster.description,
    levels = c(
      "Embryo_Specific",
      "Larval",
      "Increasing",
      "L3_High",
      "Not_Changing",
      "Not_Bound"
    )
  )
```

# Subset heatmap based on ELT-2 binding pattern

```{r chip_rna_matrix}
#### Handle duplicate rows created by 1:many gene:peak mapping

# match will return the first index of each non-redundant gene
nr_gene_name_ixs = match(unique(chip_rna_df$wikigene_name),chip_rna_df$wikigene_name)
#length(nr_gene_name_ixs)
#[1] 3286

chip_rna_df = chip_rna_df[nr_gene_name_ixs,]


chip_rna_matrix <-
  chip_rna_df %>% select(emb_510min:YA) %>% as.matrix()
  #chip_rna_df %>% select(emb_548min,emb_626min,L1,L2,L3,L4) %>% as.matrix()


#### Handle 0's and take the log
# 1. Just replace 0's as NAs so we can apply log(). Alternatively, we could do log(x + .01), but there are only ~300
chip_rna_matrix_na = chip_rna_matrix; 
chip_rna_matrix_na[0 == chip_rna_matrix_na] <- NA
# 2. Apply log()
chip_rna_matrix_log = log( chip_rna_matrix_na )
# 3. Do variances row-wise, make sure to set na.rm=T
rowvariances = apply(chip_rna_matrix_log, 1, var, na.rm=T)

range(rowvariances) # no NaNs
# 4. Plot distribution of row variances of the log data...
# You can do hist with 10,100,1000 breaks, and there will always be
# a dominant spike all the way on the lowest value. This is because the data elicit no natural lowest break, and can be demonstrated by ecdf, which is binless.
plot(ecdf(rowvariances)) # no "steps" anywhere, just a smooth curve
# therefore, we will choose to exclude the lowest 5% of the rows by their variance

changing = rowvariances > 1  # quantile(rowvariances,.1) # or .05
chip_rna_matrix = chip_rna_matrix_log
rownames(chip_rna_matrix) <- chip_rna_df$wikigene_name
chip_rna_matrix_scaled <- row_scale(chip_rna_matrix) # calls base::scale() via RWC23_Functions.R



```

```{r print-gene-names-indices}
# for (name in gene_names) {
#   index <- which(rownames(chip_rna_matrix_scaled) == name)
#   for (i in 1:length(index)) {
#     print(c(name, index[i]))
#   }
# }
```
```{r chip_GOI_df}
# This is now done directly before plotting to incorporate filtering

# chip_GOI_df <- data.frame(name = NULL, index = NULL)
# for (name in gene_names) {
#   index <- which(rownames(chip_rna_matrix_scaled) == name)
#   for (i in 1:length(index)) {
#     toappend <-
#       data.frame(name = name,
#                  index = index[i],
#                  stringsAsFactors = FALSE)
#     chip_GOI_df <- bind_rows(chip_GOI_df, toappend)
#   }
# }
# chip_GOI_df
```
```{r gower_dist, include=F}
# chip_rna_matrix_scaled_df = as.data.frame(chip_rna_matrix_scaled)
# chip_rna_matrix_scaled_df$elt2 = factor(chip_rna_df$cluster.description)
# gower_dist <- daisy(chip_rna_matrix_scaled_df, metric="gower", type=list(logratio=3))
# summary(gower_dist)
# gower_mat <- as.matrix(gower_dist)
# chip_rna_matrix_scaled_df[which(gower_mat == min(gower_mat[gower_mat != min(gower_mat)]), arr.ind = TRUE)[1, ], ]
# chip_rna_matrix_scaled_df[
#   which(gower_mat == max(gower_mat[gower_mat != max(gower_mat)]),
#         arr.ind = TRUE)[1, ], ]
# 
# # Do the silhouette widths to find the optimum number of partitions.
# # This procedure is very time-consuming.
# sil_width <- c(NA)
# nclusts=11
# for(i in 2:nclusts){
#   print(i)
#   pam_fit <- pam(gower_dist,
#                  diss = TRUE,
#                  k = i)
# 
#   sil_width[i] <- pam_fit$silinfo$avg.width
#   print(sil_width[i])
# }
# 
# # Plot silhouette width (higher is better)
# #
# plot(1:nclusts, sil_width,
#      xlab = "Number of clusters",
#      ylab = "Silhouette Width")
# lines(1:nclusts, sil_width)
# # all data- 9 is the peak at 0.3373983
# # Embryo_548min, L1, L3 continues to increase, but 14 is a local peak at 0.5625833
# pam_fit <- pam(gower_dist, diss = TRUE, k = 8)

```

```{r cstats}
library(fpc)
cstats.table <- function(dist, tree, k) {
clust.assess <- c("cluster.number","n","within.cluster.ss","average.within","average.between",
                  "wb.ratio","dunn2","avg.silwidth")
clust.size <- c("cluster.size")
stats.names <- c()
row.clust <- c()
output.stats <- matrix(ncol = k, nrow = length(clust.assess))
cluster.sizes <- matrix(ncol = k, nrow = k)
for(i in c(1:k)){
  row.clust[i] <- paste("Cluster-", i, " size")
}
for(i in c(2:k)){
  cat("Doing k",i,"...")
  stats.names[i] <- paste("Test", i-1)
  
  for(j in seq_along(clust.assess)){
    output.stats[j, i] <- unlist(cluster.stats(d = dist, clustering = cutree(tree, k = i))[clust.assess])[j]
    
  }
  
  for(d in 1:k) {
    cluster.sizes[d, i] <- unlist(cluster.stats(d = dist, clustering = cutree(tree, k = i))[clust.size])[d]
    dim(cluster.sizes[d, i]) <- c(length(cluster.sizes[i]), 1)
    cluster.sizes[d, i]
    
  }
  cat("done,  ")
}
output.stats.df <- data.frame(output.stats)
cluster.sizes <- data.frame(cluster.sizes)
cluster.sizes[is.na(cluster.sizes)] <- 0
rows.all <- c(clust.assess, row.clust)
# rownames(output.stats.df) <- clust.assess
output <- rbind(output.stats.df, cluster.sizes)[ ,-1]
colnames(output) <- stats.names[2:k]
rownames(output) <- rows.all
is.num <- sapply(output, is.numeric)
output[is.num] <- lapply(output[is.num], round, 2)
output
}

```

```{r BoeckRNA_ELT2_chip}
real = apply(chip_rna_matrix_scaled, 1, function(x) { ! any(is.na(x)) }) # NA's introduced by log transform don't jive here
changing = rowvariances > quantile(rowvariances,.25)
# euc.dist = dist(chip_rna_matrix_scaled[real & changing,])
# aggl.clust.c <- hclust(euc.dist, method = "complete")
# 
# #stats.df.aggl <-cstats.table(euc.dist, aggl.clust.c, 15)
# stats.df.aggl
# ggplot(data = data.frame(t(stats.df.aggl)), 
#   aes(x=cluster.number, y=within.cluster.ss)) + 
#   geom_point()+
#   geom_line()+
#   ggtitle("Agglomerative clustering") +
#   labs(x = "Num.of clusters", y = "Within clusters sum of squares (SS)") +
#   theme(plot.title = element_text(hjust = 0.5))
# 
# # Silhouette
# ggplot(data = data.frame(t(stats.df.aggl)), 
#   aes(x=cluster.number, y=avg.silwidth)) + 
#   geom_point()+
#   geom_line()+
#   ggtitle("Agglomerative clustering") +
#   labs(x = "Num.of clusters", y = "Average silhouette width") +
#   theme(plot.title = element_text(hjust = 0.5))
# 
# library("dendextend")
# dendro <- as.dendrogram(aggl.clust.c)
# dendro.col <- dendro %>%
#   set("branches_k_color", k = 4, value =   c("darkslategray","darkslategray3", "gold3", "darkcyan"))%>%
#   set("branches_lwd", 0.6) 
#   # set("labels_colors", 
#   #     value = c("darkslategray")) %>% 
#   # set("labels_cex", 0.5)
# ggd1 <- as.ggdend(dendro.col)
# ggplot(ggd1, theme = theme_minimal()) +
#   labs(x = "Num. observations", y = "Height", title = "Dendrogram, k = 4")
# 
# 
# ix=which(rownames(chip_rna_matrix_scaled)[real & changing] %in% gene_names)
# chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing][ix],
#                          index =ix )
# ### ALL ###
# BoeckRNA_ELT2_chip <- Heatmap(
#   chip_rna_matrix_scaled[real & changing ,],
#   name = "Boeck Time Resolved RNA",
#   #row_split = chip_rna_df$cluster.description[real & changing ],
#   column_title = sprintf("Log(RNA Timecourse), exclude lowest 25%% variance"),
#   row_title = NULL,
#   row_dend_width = unit(20, "mm"),
#   row_km = 4,
#   #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
#   #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
#   cluster_columns = FALSE
# ) +
#   rowAnnotation(
#     ELT2_cluster = chip_rna_df$cluster.description[real & changing ],
#     col = list(
#       ELT2_cluster = c(
#         "Embryo_Specific" = "#7570B3",
#         "Larval" = "#1B9E77",
#         "Increasing" = "#E7298A",
#         "L3_High" = "#D95F02",
#         "Not_Changing" = "#505050",
#         "Not_Bound" = "yellow"
#       )
#     ),
#     border = TRUE
#   ) +  rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
#                                labels = chip_GOI_df$name))
# BoeckRNA_ELT2_chip
 ```
 
 ```{r}
gene_names <-
  c("flh-3", "elt-7", "clec-258", "pqm-1", "mtl-2", "ets-4", "aat-6")


embryo_specific = chip_rna_df$cluster.description == "Embryo_Specific"
larval = chip_rna_df$cluster.description == "Larval"

#changing = rowvariances > 0.1355294 # .05 thresh from chipseq

####                          #####
####                          #####
#### NO THRESHOLD ON VARIANCE #####
changing = rowvariances > 0
####                          #####
####                          #####
# find Genes Of Interest in subsetted table
ix=which(rownames(chip_rna_matrix_scaled)[real & changing] %in% gene_names)
chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing][ix],
                         index =ix )
### ALL ###
BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[real & changing ,],
  name = "Boeck Time Resolved RNA",
  row_split = chip_rna_df$cluster.description[real & changing ],
  column_title = sprintf("Log(RNA Timecourse), No threshold on variance"),
  row_title = NULL,
  #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
  #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
  cluster_columns = FALSE
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real & changing ],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  ) +  rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
                               labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip

#### EMBRYO ONLY ####

ix=which(rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific] %in% gene_names)
chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific][ix], index =ix )

BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[real & changing & embryo_specific,],
  name = "Boeck Time Resolved RNA",
  row_split = chip_rna_df$cluster.description[real & changing & embryo_specific],
  column_title = sprintf("Log(RNA Timecourse) no threshold on variance", quantile(rowvariances,.05)),
  row_title = NULL,
  #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
  #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
  cluster_columns = FALSE
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real & changing & embryo_specific],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  ) + rowAnnotation(foo = anno_mark(at = chip_GOI_df$index, labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip


####                          #####
####                          #####
####  VARIANCE > lowest .05   #####
changing = rowvariances > quantile(rowvariances,.05)
####                          #####
####                          #####
# find Genes Of Interest in subsetted table
ix=which(rownames(chip_rna_matrix_scaled)[real & changing] %in% gene_names)
chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing ][ix],
                         index =ix )

BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[real & changing, ],
  name = "Boeck Time Resolved RNA",
  row_split = chip_rna_df$cluster.description[real & changing],
  column_title = sprintf("Log(RNA Timecourse), var > %.2f (q.05)", quantile(rowvariances,.05)),
  row_title = NULL,
  #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
  #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
  cluster_columns = FALSE
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real & changing],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  ) +  rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
                               labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip

ix=which(rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific] %in% gene_names)
chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific][ix],
                         index =ix )

BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[real & changing & embryo_specific,],
  name = "Boeck Time Resolved RNA",
  row_split = chip_rna_df$cluster.description[real & changing & embryo_specific],
  column_title = sprintf("Log(RNA Timecourse), var > %.2f (q.05)", quantile(rowvariances,.05)),
  row_title = NULL,
  #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
  #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
  cluster_columns = FALSE
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real & changing & embryo_specific],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  )  + rowAnnotation(foo = anno_mark(at = chip_GOI_df$index, labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip


####                          #####
####                          #####
####  VARIANCE > 1 q(~.8)     #####
changing = rowvariances > 1
####                          #####
####                          #####
# find Genes Of Interest in subsetted table
ix=which(rownames(chip_rna_matrix_scaled)[real & changing] %in% gene_names)
chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing ][ix],
                         index =ix )

BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[real & changing, ],
  name = "Boeck Time Resolved RNA",
  row_split = chip_rna_df$cluster.description[real & changing],
  column_title = sprintf("Log(RNA Timecourse), variance > 1 (top 20%% data)"),
  row_title = NULL,
  #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
  #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
  cluster_columns = FALSE
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real & changing],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  ) +  rowAnnotation(foo = anno_mark(at = chip_GOI_df$index,
                               labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip

ix=which(rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific] %in% gene_names)
chip_GOI_df = data.frame(name=rownames(chip_rna_matrix_scaled)[real & changing & embryo_specific][ix],
                         index =ix )

BoeckRNA_ELT2_chip <- Heatmap(
  chip_rna_matrix_scaled[real & changing & embryo_specific,],
  name = "Boeck Time Resolved RNA",
  row_split = chip_rna_df$cluster.description[real & changing & embryo_specific],
  column_title = sprintf("Log(RNA Timecourse), variance > 1 (top 20%% data)"),
  row_title = NULL,
  #clustering_distance_rows = function(x) {philentropy::distance(x,method="euclidean", test.na=F)},
  #clustering_distance_rows = function(x) daisy(x, metric="gower", type=list(logratio=3)),
  cluster_columns = FALSE
) +
  rowAnnotation(
    ELT2_cluster = chip_rna_df$cluster.description[real & changing & embryo_specific],
    col = list(
      ELT2_cluster = c(
        "Embryo_Specific" = "#7570B3",
        "Larval" = "#1B9E77",
        "Increasing" = "#E7298A",
        "L3_High" = "#D95F02",
        "Not_Changing" = "#505050",
        "Not_Bound" = "yellow"
      )
    ),
    border = TRUE
  )  + rowAnnotation(foo = anno_mark(at = chip_GOI_df$index, labels = chip_GOI_df$name))
BoeckRNA_ELT2_chip












# pdf(file = "./03_plots/201008_Boeck_RNA_ELT2ChIP_Patterns_Subset.pdf", 
#     width = 7, height = 7)
# BoeckRNA_ELT2_chip
# dev.off()
```

