---
title: "Step6 -- DESeq2 -- Clustering"
author: "Erin Osborne Nishimura"
date: "November 21, 2017"
output: pdf_document
---

#######################################

__Date:__ November 21, 2017

__Author:__ Erin Osborne Nishimura

__Script:__ step6\_171121\_DESeq2\_Clusteringanalysis.Rmd

__Project:__ To analyze Aidan's RNA-seq datset from FACS sorted and filtered worms of the genotypes N2, elt-2D, elt-7D, and elt-2Delt-7D


__Requires:__ 
  + 3.3.2 
  + Bioconductor v. 3.2. for some reason 3.3 won't work.
  + RColorBrewer (1.1.2)   
  + dplyr
  + GenomicRanges
  + gplots
  + ComplexHeatmap
  + circlize
              
######################################


## Load required libraries:   
  + DESeq2   
  + RColorBrewer   
  + dplyr
  + GenomicRanges
  + gplots
  + ComplexHeatmap
  
```{r, message = FALSE, warning = FALSE}
# #First time loading of DESeq2:
# source("http://bioconductor.org/biocLite.R")
# biocValid() 
# biocLite("BiocUpgrade") 
# biocLite("dplyr")
# biocLite("ComplexHeatmap")


library(ComplexHeatmap)
library(RColorBrewer)
library(dplyr)
library(GenomicRanges)
library(circlize)

```


## Upload datasets from previous steps
I'll upload: 
  + The list of changing genes
  + The annotated list of rld log counts
  


# Get the annotated list of r-stabilized log transformed counts
```{r}
# Get r-stabilized log transformed counts
projectdir <- "~/Dropbox/labwork/2016_ELT2_PROJECT/07_DESeq2_Analysis_EOP215"
setwd(paste(projectdir, "03_output", "step5_datasets", sep = "/"))
rlog_annot <- read.table(file="2017-07-12_step5_rlog_counts.txt", sep = "\t", na.strings = c(""), header = TRUE)
dim(rlog_annot)

# Get list of changing genes
setwd(paste(projectdir, "03_output", "step5_datasets", sep = "/"))
changing_genes <- read.table(file = "2017-07-12_all_changing_genes_0.1alpha_0.8lfc.txt", header = FALSE)
dim(changing_genes)
```



# Incorporate in the Mann & Weisenfahrt ChIP-seq data

# Produce bed files. 
```{r}
rlog_annot_bed <- rlog_annot[,c("chr_ce10", "start_min_ce10", "stop_max_ce10", "Row.names", "strand_ce10")]
names(rlog_annot_bed) <- c("chr", "start", "end", "id", "strand")
dim(rlog_annot_bed)

#Remove a few entries that have "+,-" strands
rlog_annot_bed <- rlog_annot_bed[setdiff(1:dim(rlog_annot_bed)[1], grep(",", rlog_annot_bed$strand )),]
rlog_annot_bed <- rlog_annot_bed[setdiff(1:dim(rlog_annot_bed)[1], grep(",", rlog_annot_bed$chr )),]

#Remove entries that have no start or end
rlog_annot_bed <- rlog_annot_bed[setdiff(1:dim(rlog_annot_bed)[1], grep("NA", rlog_annot_bed$end )),]
rlog_annot_bed <- rlog_annot_bed[setdiff(1:dim(rlog_annot_bed)[1], grep("NA", rlog_annot_bed$start )),]
dim(rlog_annot_bed)

#Change character and numeric classes
rlog_annot_bed$strand <- as.character(rlog_annot_bed$strand)
rlog_annot_bed$chr <- as.character(rlog_annot_bed$chr)
rlog_annot_bed$end <- as.numeric(as.character(rlog_annot_bed$end))
rlog_annot_bed$start <- as.numeric(as.character(rlog_annot_bed$start))

#Make Granges
norm_counts_annot_gr <- with(rlog_annot_bed, GRanges(chr, IRanges(start+1, end), strand=strand, id=id))

length(norm_counts_annot_gr)
head(norm_counts_annot_gr)

```

## Import ChIP-seq dataset from Weisenfahrt et al.

This information was downloaded from the paper: 
The function and regulation of the GATA factor ELT-2 in the C. elegans endoderm  
Tobias Wiesenfahrt, Janette Y. Berg, Erin Osborne Nishimura, Adam G. Robinson, Barbara Goszczynski, Jason D. Lieb, James D. McGhee  
http://dev.biologists.org/content/143/3/483.long  

Supplemental datasets: 14\_ELT2\_IggBlackHOTMinus\_gt30_summits.bed. 

They will be imported into a GRagnes Object called peaks_bed:

```{r}

#Import the ChIP-seq peaks from Weisenfahrt et al., 2016
setwd(paste(projectdir, "01_input", "02_from_weisenfahrt", sep = "/"))
peaks_bed <- read.table(file = "14_ELT2_IggBlackHOTMinus_gt30_summits.bed", sep = "\t", header = FALSE)
#Format the weisenfahrt peaks into a peaks_bed object that is GRanges compatible
colnames(peaks_bed) <- c("chr", "start", "end", "peakname", "MACS2_score")
peaks_bed <- with(peaks_bed, GRanges(chr, IRanges(start+1, end), id=peakname, score=MACS2_score))
head(peaks_bed)

```
There are `r length(peaks_bed$id)` peaks in the Weisenfahrt et al. dataset.


## Import ChIP-seq dataset from Mann et al.

This dataset is ELT-2 ChIP-seq data from the paper:  
Deactivation of the GATA Transcription Factor ELT-2 Is a Major Driver of Normal Aging in C. elegans  
Frederick G. Mann, Eric L. Van Nostrand, Ari E. Friedland, Xiao Liu, Stuart K. Kim  
http://journals.plos.org/plosgenetics/article?id=10.1371%2Fjournal.pgen.1005956 

Supplemental Dataset:  
S1 Table. Low-complexity ELT-2 ChIP-seq targets.  
doi:10.1371/journal.pgen.1005956.s011 (XLSX)  
mann\_etal\_journal.pgen.1005956.s011.txt  

```{r}


#Import the ChIP-seq peaks from Mann et al., 2016
setwd(paste(projectdir, "01_input", "03_from_mann", sep = "/"))
mannPeaks <- read.table(file = "mann_etal_journal.pgen.1005956.s011.txt", sep = "\t", header = TRUE, skip = 1, stringsAsFactors = FALSE)

#Format Mann peaks into a GRanges compatible object called mannPeaks_bed
mannPeaks <- mutate(mannPeaks, chr = sapply(strsplit(mannPeaks$Genome.Position, ":"), "[", 1),
       range = sapply(strsplit(mannPeaks$Genome.Position, ":"), "[", 2),
       score = -log(q.value, base = 10))

mannPeaks <- mutate(mannPeaks, 
       start = sapply(strsplit(mannPeaks$range, "-"), "[", 1),
       stop = sapply(strsplit(mannPeaks$range, "-"), "[", 2)
       )

mannPeaks$chr <- paste("chr", mannPeaks$chr, sep = "")

#Convert mannPeaks into a bed and Granges object:
mannPeaks_bed <- mannPeaks[,c(5,8,9,1,7,4)]
colnames(mannPeaks_bed) <- c("chr", "start", "end", "id", "score", "npeaks")
mannPeaks_bed$start <- as.integer(mannPeaks_bed$start)
mannPeaks_bed$end <- as.integer(mannPeaks_bed$end)
mannPeaks_bed <- with(mannPeaks_bed, GRanges(chr, IRanges(start+1, end), id=id, score=score, npeaks=npeaks))


```

There are `r length(mannPeaks_bed$id)` peaks in the Mann et al. dataset.

## Use GRanges to find overlaps between the promoters of annotated genes and the peaks in both the Weisenfahrt et al. and Mann et al. datasets.

Look for peaks that fall within 3000 bp upstream of the transcription start site and within 1000 bp downstream of the transcription start site.

Let's do the Weisenfahrt dataset first:

```{r}

#Line up norm_counts_annot_bed with peaks and mann_peaks

#Find overlaps
#Make promoter regions; find overlaps between promoters and peaks
all_promoters <- promoters(norm_counts_annot_gr, upstream=3000, downstream=1000, use.names=TRUE)
#The above code didn't work on a more recent installation. try again
all_promoters <- promoters(norm_counts_annot_gr, upstream=3000, downstream=1000)

hits <- findOverlaps(all_promoters, peaks_bed)
head(hits)
#length(peaks_bed)
#length(all_promoters)
#str(hits)
length(queryHits(hits))
```

There are `r length(queryHits(hits))` of `r length(peaks_bed$id)` ELT-2 ChIP-seq peaks (from Weisenfahrt et al) fall within 3 kb upstream and 1 kb downstream of an annotated transcriptional start site.

Save this in an object called counts\_annot\_peaks1



```{r}

#Pull out the genes that match with these peaks, preserve peak scores from Weisenfahrt et al.
match_peaks <- cbind.data.frame(query_id = as.vector(all_promoters$id[queryHits(hits)]), score = as.vector(peaks_bed$score[subjectHits(hits)]))
weisenfahrt_match_peaks_annot <- cbind.data.frame(query_id = as.vector(all_promoters$id[queryHits(hits)]), score = as.vector(peaks_bed$score[subjectHits(hits)]))
#Compress these down into individual genes (some genes may have multiple peaks in their promoters)
unique_match_peaks <- match_peaks %>%
  group_by(query_id) %>%
  summarize(score_sum = sum(score),
            peaks_num = n()) %>%
  arrange(desc(score_sum))
  
#How many do we have now
dim(unique_match_peaks)
#head(unique_match_peaks)
#as.data.frame(unique_match_peaks)[1:100,]

#OK, now go back and re-merge this with the list of genes:
#head(rlog_annot)
#head(unique_match_peaks)
#dim(rlog_annot)
#dim(unique_match_peaks)
counts_annot_peaks1 <- merge(rlog_annot, unique_match_peaks, by.x = "Row.names", by.y = "query_id", all.x = TRUE)
colnames(counts_annot_peaks1)[which(colnames(counts_annot_peaks1) == "score_sum")] <- "ELT2chip_scoresum_Weisenfahrt"
colnames(counts_annot_peaks1)[which(colnames(counts_annot_peaks1) == "peaks_num")] <- "ELT2chip_peaksnum_Weisenfahrt"
dim(counts_annot_peaks1)
#head(counts_annot_peaks1)

```



#OK, now do the same for the Mann et al peaks data:
```{r, echo = FALSE}

hits2 <- findOverlaps(all_promoters, mannPeaks_bed)

```

There are `r length(queryHits(hits2))` of `r length(mannPeaks_bed$id)` ELT-2 ChIP-seq peaks (from Mann et al) that fall within 3 kb upstream and 1 kb downstream of an annotated transcriptional start site.

Now I'll pull out the unique genes associated with these peaks. This will be saved as an object called counts\_annot\_peaks2

```{r, echo = FALSE}
match_peaks2 <- cbind.data.frame(query_id = as.vector(all_promoters$id[queryHits(hits2)]), 
                                 score = as.vector(mannPeaks_bed$score[subjectHits(hits2)]))
#head(match_peaks2)

unique_match_peaks2 <- match_peaks2 %>%
  group_by(query_id) %>%
  summarize(score_sum = sum(score),
            peaks_num = n()) %>%
  arrange(desc(score_sum))
  

dim(unique_match_peaks2)
#head(unique_match_peaks2)
#as.data.frame(unique_match_peaks2)[1:100,]

#OK, now go back and re-merge this with the list of genes:
#head(counts_annot_peaks1)
#head(unique_match_peaks2)
#dim(counts_annot_peaks1)
#dim(unique_match_peaks2)
#mannPeaks_bed

```

Merge counts\_annot_peaks1 with the Mann et al datsets (unique\_match\_peaks2) to create the object: counts\_annot\_peaks2.
```{r, echo = FALSE}
counts_annot_peaks2 <- merge(counts_annot_peaks1, unique_match_peaks2, by.x = "Row.names", by.y = "query_id", all.x = TRUE)
colnames(counts_annot_peaks2)[which(colnames(counts_annot_peaks2) == "score_sum")] <- "ELT2chip_scoresum_Mann"
colnames(counts_annot_peaks2)[which(colnames(counts_annot_peaks2) == "peaks_num")] <- "ELT2chip_peaksnum_Mann"
head(counts_annot_peaks2)
dim(counts_annot_peaks2)



```


Ok, now I have a large datastructure: counts\_annot\_peak2 with information on...  
1) r-log transformed count data from the RNA-seq assay  
2) annotation information for ce10 and ce11  
3) ChIP-seq info from Weisenfahrt et al.  
4) ChIP-seq info from Mann et al.  


# Import Intestine Specific and Intestine-enriched gene lists. Merge those into the dataset and incorporate the information into the clustering analysis:

```{r}
# Import the intestine enriched list
setwd(paste(projectdir, "01_input", "04_intestineSp", sep = "/"))
intestineEnriched <- read.table(file = "IEG_170104.txt", sep = "\t", header = FALSE)
dim(intestineEnriched)


# Import the intestine specific list
setwd(paste(projectdir, "01_input", "04_intestineSp", sep = "/"))
intestineSpecific <- read.table(file = "ISG_170104.txt", sep = "\t", header = FALSE)
dim(intestineSpecific)

intestine_union <- union(intestineSpecific$V1, intestineEnriched$V1)

# Annotate whether a gene is intestine enriched or specific in rlog counts data frame:
all_annot_rlog_counts_peaks_int <- counts_annot_peaks2 %>%
        mutate(intestine_enr = (sequence_id_list %in% intestine_union)) %>%
        mutate(intestine_sp = (sequence_id_list %in% intestineSpecific$V1))

#dim(all_annot_rlog_counts_peaks_int)
#head(all_annot_rlog_counts_peaks_int)
#sum(all_annot_rlog_counts_peaks_int$intestine_enr)
#sum(all_annot_rlog_counts_peaks_int$intestine_sp)

```

# Save the dataset

```{r}

setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
filename = paste(Sys.Date(), "counts_annot_peaks_int.txt", sep = "_")
write.table(all_annot_rlog_counts_peaks_int, file = filename, sep = "\t")
colnames(all_annot_rlog_counts_peaks_int)
colnames(all_annot_rlog_counts_peaks_int[1:100,c(1,16:39)])

setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
filename = paste(Sys.Date(), "genes_ELT2peaks_intestineExpression.txt", sep = "_")

write.table(all_annot_rlog_counts_peaks_int[1:100,c(1,16:39)], file = filename, sep = "\t", quote = FALSE)

```

# Import the list of changing genes and filter: 

```{r}
# Get list of changing genes
setwd(paste(projectdir, "03_output", "step5_datasets", sep = "/"))
changing_genes <- read.table(file = "2017-07-12_all_changing_genes_0.1alpha_0.8lfc.txt", header = FALSE)
dim(changing_genes)
head(changing_genes)
```

# Merge the changing genes list with the list of rlog counts
```{r}
changing_pairwise_rlog_counts <- all_annot_rlog_counts_peaks_int[all_annot_rlog_counts_peaks_int$Row.names %in% changing_genes$V1,]
dim(changing_pairwise_rlog_counts)

# Changing NA's to 0's:
# Changing NA's to 0
# Changing Inf to MaxFinite
changing_pairwise_rlog_counts$ELT2chip_scoresum_Weisenfahrt[which(is.na(changing_pairwise_rlog_counts$ELT2chip_scoresum_Weisenfahrt))] <- 0
changing_pairwise_rlog_counts$ELT2chip_scoresum_Mann[which(is.na(changing_pairwise_rlog_counts$ELT2chip_scoresum_Mann))] <- 0
max_finite <- max(changing_pairwise_rlog_counts$ELT2chip_scoresum_Mann[is.finite(changing_pairwise_rlog_counts$ELT2chip_scoresum_Mann)])
changing_pairwise_rlog_counts$ELT2chip_scoresum_Mann[which(changing_pairwise_rlog_counts$ELT2chip_scoresum_Mann == "Inf")] <- max_finite
#head(changing_pairwise_rlog_counts)

# setwd(paste(projectdir, "03_output", "step5_datasets", sep = "/"))
# changing_genes <- read.table(file = "2017-07-12_all_changing_genes_0.1alpha_0.8lfc.txt", header = FALSE)
# dim(changing_genes)
# head(changing_genes)
```


# Cluster with annotation:

```{r}
# Set up a matrix for a clustered heatmap:
rld_pairwise_matrix <- as.matrix(changing_pairwise_rlog_counts[,c(2:15)])
rld_pairwise_matrix <- rld_pairwise_matrix[,c(8:14, 1:7)]
row.names(rld_pairwise_matrix) <- changing_pairwise_rlog_counts$gene_id_val
#head(rld_pairwise_matrix)
dim(rld_pairwise_matrix)

#scaling, no centering:
mat_scaled = t(apply(unlist(rld_pairwise_matrix), 1, scale))
colnames(mat_scaled) <- colnames(rld_pairwise_matrix)
head(mat_scaled)



```


# Functionalize the clustering

```{r}

return_cluster <- function(dataframe, num){
        #dataframe <- changing_pairwise_rlog_counts_intSp
        # Convert dataframe to a matrix
        rld_pairwise_matrix <- as.matrix(dataframe[,c(2:15)])
        rld_pairwise_matrix <- rld_pairwise_matrix[,c(8:14, 1:7)]
        row.names(rld_pairwise_matrix) <- dataframe$gene_id_val
        
        # scaling, no centering:
        mat_scaled = t(apply(unlist(rld_pairwise_matrix), 1, scale))
        colnames(mat_scaled) <- colnames(rld_pairwise_matrix)
        
        # Draw main RNA-seq heatmap
        ht1 <- Heatmap(mat_scaled,
          col = colorRampPalette(c("cyan","black","yellow"))(1000),    
          cluster_columns = FALSE,
          clustering_distance_rows = "spearman",
          clustering_method_rows = "complete", 
          #show_row_names = FALSE,
          show_column_names = TRUE,
          row_names_gp = gpar(cex = 0.2),
          column_names_gp = gpar(cex = 0.4),
          heatmap_legend_param = list(color_bar = "continuous"), split = num)

        # Draw ChIP-seq Weisenfahrt heatmap
        maxWeis <- max(dataframe$ELT2chip_scoresum_Weisenfahrt)
        maxWeis1 <- maxWeis*0.125
        maxWeis2 <- maxWeis*0.25
        maxWeis3 <- maxWeis*0.5
        ht2 <- Heatmap(dataframe$ELT2chip_scoresum_Weisenfahrt, 
                       name = "ELT-2 ChIP (Weisenfahrt)", 
                       show_row_names = FALSE, width = unit(2, "mm"), 
                       col = colorRamp2(c(0, maxWeis1, maxWeis2, maxWeis3), 
                                        c("#000000", "#006400", "#00af00", "#00fb00")), 
                       heatmap_legend_param = list(color_bar = "continuous"))

        # Draw ChIP-seq Mann heatmap
        maxMann <- max(dataframe$ELT2chip_scoresum_Mann)
        maxMann1 <- maxMann*0.125
        maxMann2 <- maxMann*0.25
        maxMann3 <- maxMann*0.5
        ht3 <- Heatmap(dataframe$ELT2chip_scoresum_Mann, 
                       name = "ELT-2 ChIP peak (Mann et al)", 
                       show_row_names = FALSE, 
                       width = unit(2, "mm"), 
                       col = colorRamp2(c(0, maxMann1, maxMann2, maxMann3), 
                                      c("#000000", "#006400", "#00af00", "#00fb00")), 
                       heatmap_legend_param = list(color_bar = "continuous"))
        
        # Annotate with intestine-enriched
        ha <- rowAnnotation(intestine_enriched = dataframe$intestine_enr, 
                            col = list(intestine_enriched = c("TRUE" = "red", "FALSE" = "gray")), 
                            gap = unit(0, "mm"),
                            width = unit(0.25, "cm"))
        
        # Annotate with intestine-specific
        ha2 <- rowAnnotation(intestine_specific = dataframe$intestine_sp, 
                             col = list(intestine_specific = c("TRUE" = "green", "FALSE" = "gray")), 
                             gap = unit(0, "mm"),
                             width = unit(0.25, "cm"))
        
        ht_list2 = ht1 + ht2 + ht3 + ha + ha2
        return(ht_list2)
}

# Generate the clustered heatmap for all genes
draw(return_cluster(changing_pairwise_rlog_counts, 6))
changing_cluster_list <- return_cluster(changing_pairwise_rlog_counts, 5)

# str(changing_cluster_list)
# as.vector(changing_pairwise_rlog_counts_intSp[row_order(ht1)[[1]],]$WBGene)
# row_order(changing_cluster_list)[[1]]

#Save it as a figure
setwd(paste(projectdir, "03_output", sep = "/"))
dir.create(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"), showWarnings = FALSE)
setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
pdf(paste(Sys.Date(), "changing_genes_cluster_annotated_all.pdf", sep = "_" ), width=5, height=8)
draw(changing_cluster_list)
dev.off()


# Generate the clustered heatmap for intestine-enriched genes
changing_pairwise_rlog_counts_intEnr <- changing_pairwise_rlog_counts %>%
         filter(intestine_enr == TRUE)

dim(changing_pairwise_rlog_counts_intEnr)
#head(changing_pairwise_rlog_counts_intEnr)

draw(return_cluster(changing_pairwise_rlog_counts_intEnr, 5))
enriched_changing_cluster_list <- return_cluster(changing_pairwise_rlog_counts_intEnr, 5)

setwd(paste(projectdir, "03_output", sep = "/"))
dir.create(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"), showWarnings = FALSE)
setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
pdf(paste(Sys.Date(), "all_changing_genes_cluster_annotated_enriched.pdf", sep = "_" ), width=5, height=8)
draw(enriched_changing_cluster_list)
dev.off()


# Generate the clustered heatmap for intestine-specific genes
changing_pairwise_rlog_counts_intSp <- changing_pairwise_rlog_counts %>%
         filter(intestine_sp == TRUE)

dim(changing_pairwise_rlog_counts_intSp)
#head(changing_pairwise_rlog_counts_intSp)

draw(return_cluster(changing_pairwise_rlog_counts_intSp, 2))
specific_changing_cluster_list <- return_cluster(changing_pairwise_rlog_counts_intSp, 2)

setwd(paste(projectdir, "03_output", sep = "/"))
dir.create(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"), showWarnings = FALSE)
setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
pdf(paste(Sys.Date(), "ungrouped_changing_genes_cluster_annotated_specific.pdf", sep = "_" ), width=5, height=8)
draw(specific_changing_cluster_list)
dev.off()


```


# Save genelists over each subcluster 
```{r}

# Write a script to automate the heatmap generation:

return_clusterlist <- function(dataframe, num){
        #dataframe <- changing_pairwise_rlog_counts_intSp
        # Convert dataframe to a matrix
        rld_pairwise_matrix <- as.matrix(dataframe[,c(2:15)])
        rld_pairwise_matrix <- rld_pairwise_matrix[,c(8:14, 1:7)]
        row.names(rld_pairwise_matrix) <- dataframe$gene_id_val
        
        # scaling, no centering:
        mat_scaled = t(apply(unlist(rld_pairwise_matrix), 1, scale))
        colnames(mat_scaled) <- colnames(rld_pairwise_matrix)
        
        # Draw main RNA-seq heatmap
        ht1 <- Heatmap(mat_scaled,
          col = colorRampPalette(c("cyan","black","yellow"))(1000),    
          cluster_columns = FALSE,
          clustering_distance_rows = "spearman",
          clustering_method_rows = "complete", 
          #show_row_names = FALSE,
          show_column_names = TRUE,
          row_names_gp = gpar(cex = 0.2),
          column_names_gp = gpar(cex = 0.4),
          heatmap_legend_param = list(color_bar = "continuous"), split = num)
        
        return(ht1)
}

# Generate the clustered heatmap list for all genes
dim(changing_pairwise_rlog_counts)
changing_cluster_list <- return_clusterlist(changing_pairwise_rlog_counts, 6)

#Extract WBGene identifiers for each cluster:
set1 <- as.vector(changing_pairwise_rlog_counts[row_order(changing_cluster_list)[[1]],]$Row.names)
set3 <- as.vector(changing_pairwise_rlog_counts[row_order(changing_cluster_list)[[2]],]$Row.names)
set2 <- as.vector(changing_pairwise_rlog_counts[row_order(changing_cluster_list)[[3]],]$Row.names)
set4 <- as.vector(changing_pairwise_rlog_counts[row_order(changing_cluster_list)[[4]],]$Row.names)
set5 <- as.vector(changing_pairwise_rlog_counts[row_order(changing_cluster_list)[[5]],]$Row.names)
set6 <- as.vector(changing_pairwise_rlog_counts[row_order(changing_cluster_list)[[6]],]$Row.names)
length(set1)
length(set2)
length(set3)
length(set4)
length(set5)
```


# Save a Supplemental Data Table containing all changing genes, their annotations, and their set category....
```{r}
# Merge changing_pairwise_r_log_counts with the set ID lists...
changing_pairwise_rlog_counts_plusSet <- changing_pairwise_rlog_counts %>%
  mutate(set = ifelse(changing_pairwise_rlog_counts$Row.names %in% set1, "set1", 
                      ifelse(changing_pairwise_rlog_counts$Row.names %in% set2, "set2",
                             ifelse(changing_pairwise_rlog_counts$Row.names %in% set3, "set3",
                                    ifelse(changing_pairwise_rlog_counts$Row.names %in% set4, "set4",
                                           ifelse(changing_pairwise_rlog_counts$Row.names %in% set5, "set5",
                                                  ifelse(changing_pairwise_rlog_counts$Row.names %in% set6, "set6", "NA")))))))

# Save to a file 
setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
filename = paste(Sys.Date(), "Supplemental_Dataset_all_changing_genes_annotated_sets.txt", sep = "_")

write.table(changing_pairwise_rlog_counts_plusSet, file = filename, sep = "\t", quote = FALSE)
```

# Save lists for GO ontology
```{r}

#Save all clusters in a set list:
# notes, set2 and 3 were switched in illustrator, so I'll amend the code to reflect that here...
setlist <- list(set1 = set1, 
                set2 = set2, 
                set3 = set3,
                set4 = set4,
                set5 = set5,
                set6 = set6)


#Write set lists to file:
setwd(paste(projectdir, "03_output", sep = "/"))
dir.create(paste(projectdir, "03_output", "step6_clusters_for_GO", sep = "/"), showWarnings = FALSE)
setwd(paste(projectdir, "03_output", "step6_clusters_for_GO", sep = "/"))
getwd()

for (n in c(1:length(setlist))) {
  write(setlist[[n]], file = paste(Sys.Date(), "Geneset_cluster", n, "all.txt", sep = "_" ))
}

# Write background list:
write(as.vector(rlog_annot$Row.names), file = paste(Sys.Date(), "background", "all.txt", sep = "_" ))


```

```{r}
# write a list for homer for clustering information:

#load annotation information 
setwd(paste(projectdir, "03_output", sep = "/"))
lookup <- read.table(file = "2016-07-04_lookup_table_ce10_ce11.pdf", 
                     sep = "\t", header = TRUE)

#Some Lookup table entries have two WBGene entries:
duplicated_names <- names(which(table(lookup$WBGene) >1 ))
length(duplicated_names)
duplicated_lookup_entries <- lookup[lookup$WBGene %in% duplicated_names,]

#Resolve the lookup table entries by taking the first entry of each:
duplicated_genes <- lookup[lookup$WBGene %in% duplicated_names,]
dim(duplicated_genes)
non_duplicated_genes <- lookup[!(lookup$WBGene %in% duplicated_names),]
dim(non_duplicated_genes)

lookup <- rbind(non_duplicated_genes, duplicated_genes[seq(1,82,2),])
dim(lookup)


convert_names_homer <- function(vector) {
        newnames <- lookup[which(lookup$WBGene %in% vector),]$transcript
        return(as.vector(newnames))
        print(length(newnames))
        print(length(vector))

}

setlist_homer <- lapply(setlist, convert_names_homer)

#Write set lists to file:
setwd(paste(projectdir, "03_output", sep = "/"))
dir.create(paste(projectdir, "03_output", "step6_clusters_for_homer", sep = "/"), showWarnings = FALSE)
setwd(paste(projectdir, "03_output", "step6_clusters_for_homer", sep = "/"))
getwd()

for (n in c(1:length(setlist))) {
  setlist_homer[[n]]
  write(unlist(strsplit(setlist_homer[[n]], split = ",")), file = paste(Sys.Date(), "Geneset_cluster_hm", n, "all.txt", sep = "_" ))
}

# Write background list:
write(unlist(strsplit(as.character(rlog_annot$transcripts), split = ",")), file = paste(Sys.Date(), "background_hm", "all.txt", sep = "_" ))
```

# Find transcription factors in the different sets:
Don't evaluate
```{r, eval = FALSE, echo = FALSE}
setwd(paste(projectdir, "01_input", "fromJohn_Fuxman_bass_2016", sep = "/"))
TF3 <- read.table("TF3-0_namesonly.txt", header = TRUE, fill = TRUE, sep = "\t")
TF3 <- TF3[,1:4]
dim(TF3)
head(TF3)

TF3[which(TF3$WBGeneID %in% setlist[[1]]),]
TF2_set <- TF3[which(TF3$WBGeneID %in% setlist[[2]]),]
TF3[which(TF3$WBGeneID %in% setlist[[3]]),]
TF3[which(TF3$WBGeneID %in% setlist[[4]]),]
TF3[which(TF3$WBGeneID %in% setlist[[5]]),]
TF3[which(TF3$WBGeneID %in% setlist[[6]]),]

TF2_set[which(TF2_set %in% intestine_union),]

TF2_set[which(TF2_set$Public_name %in% intestine_union),]

```

# Draw plots of elt-2 ChIP-seq data 
Don't evaluate
```{r, eval = FALSE, echo = FALSE}

#head(all_annot_rlog_counts_peaks_int)

all_annot_rlog_counts_peak_int_eitherpeak <- all_annot_rlog_counts_peaks_int %>%
  mutate(either_peak = ifelse(ELT2chip_peaksnum_Weisenfahrt > 0, "TRUE", "FALSE") )

all_annot_rlog_counts_peak_int_eitherpeak <- all_annot_rlog_counts_peaks_int %>%
  mutate(either_peak = ifelse((ELT2chip_scoresum_Weisenfahrt > 0) | (ELT2chip_peaksnum_Mann > 0), "TRUE", "FALSE"))

help(ifelse)
head(all_annot_rlog_counts_peak_int_eitherpeak)    
all_annot_rlog_counts_peaks_int[1:100,]
all_annot_rlog_counts_peak_int_eitherpeak[1:100,]
all_annot_rlog_counts_peak_int_eitherpeak[55,]
par(mfrow = c(1,1))
find_peaks_num <- function(names_vector, dataframe, chip_peaks_colname){
  sum(table(dataframe[which(dataframe$Row.names %in% names_vector),chip_peaks_colname]))/length(names_vector)*100
}

find_peaks_num(set1, all_annot_rlog_counts_peak_int_eitherpeak,"ELT2chip_peaksnum_Mann")
Mannpeaks_list <- lapply(setlist, find_peaks_num, all_annot_rlog_counts_peaks_int, "ELT2chip_peaksnum_Mann")
barplot(as.matrix(as.numeric(Mannpeaks_list)), beside = TRUE, names.arg = names(Mannpeaks_list), space = 0.1, ylim = c(0,100))

find_intestine_num <- function(names_vector, dataframe, intestine_colname){
  sum(dataframe[which(dataframe$Row.names %in% names_vector),intestine_colname])/length(names_vector)*100
}

intenr_list <- lapply(setlist, find_intestine_num, all_annot_rlog_counts_peaks_int, "intestine_enr")
barplot(as.matrix(as.numeric(intenr_list)), beside = TRUE, names.arg = names(Mannpeaks_list), space = 0.1, ylim = c(0,20))


intestine_specific_list <- lapply(setlist, find_intestine_num, all_annot_rlog_counts_peaks_int, "intestine_enr")
barplot(as.matrix(as.numeric(intestine_specific_list)), beside = TRUE, names.arg = names(intestine_specific_list), space = 0.1, ylim = c(0,20))

```


# Save the genesets for the intestine specific cluster
```{r}


# Perform the clustering to divide the intestine specific genes into four clusters
intspecific_cluster_list <- return_clusterlist(changing_pairwise_rlog_counts_intSp, 4)


#Extract WBGene identifiers for each cluster:
classB <- as.vector(changing_pairwise_rlog_counts_intSp[row_order(intspecific_cluster_list)[[1]],]$Row.names)
classC <- as.vector(changing_pairwise_rlog_counts_intSp[row_order(intspecific_cluster_list)[[2]],]$Row.names)
classD <- as.vector(changing_pairwise_rlog_counts_intSp[row_order(intspecific_cluster_list)[[3]],]$Row.names)
classA <- as.vector(changing_pairwise_rlog_counts_intSp[row_order(intspecific_cluster_list)[[4]],]$Row.names)

length(classA)
length(classB)
length(classC)
length(classD)

#Save all clusters in a set list:
# notes, set2 and 3 were switched in illustrator, so I'll amend the code to reflect that here...
sp_setlist <- list(setA = classA, 
                setB = classB, 
                setC = classC,
                setD = classD)


#Write set lists to file:
setwd(paste(projectdir, "03_output", sep = "/"))
dir.create(paste(projectdir, "03_output", "step6_clusters_for_bifrucationPlot", sep = "/"), showWarnings = FALSE)
setwd(paste(projectdir, "03_output", "step6_clusters_for_bifrucationPlot", sep = "/"))
getwd()

for (n in c(1:length(sp_setlist))) {
  write(sp_setlist[[n]], file = paste(Sys.Date(), "_geneset_cluster", n, "intsp.txt", sep = "_" ))
}



```

# Make a new Supplemental Dataset with just the intestine specific information and the different classes...
```{r}

# head(changing_pairwise_rlog_counts_intSp)
dim(changing_pairwise_rlog_counts_intSp)

# Merge changing_pairwise_r_log_counts with the set ID lists...
changing_intestineSp_rlog_counts_plusSet <- changing_pairwise_rlog_counts_intSp %>%
  mutate(class = ifelse(changing_pairwise_rlog_counts_intSp$Row.names %in% classA, "classA", 
                      ifelse(changing_pairwise_rlog_counts_intSp$Row.names %in% classB, "ClassB",
                             ifelse(changing_pairwise_rlog_counts_intSp$Row.names %in% classC, "ClassC",
                                    ifelse(changing_pairwise_rlog_counts_intSp$Row.names %in% classD, "ClassD", "NA")))))


# head(changing_intestineSp_rlog_counts_plusSet)
dim(changing_intestineSp_rlog_counts_plusSet)
# Save to a file 
setwd(paste(projectdir, "03_output", "step6_cluster_analysis", sep = "/"))
filename = paste(Sys.Date(), "Supplemental_Dataset_intestineSp_changing_annotated_sets.txt", sep = "_")

write.table(changing_intestineSp_rlog_counts_plusSet, file = filename, sep = "\t", quote = FALSE)

```




```{r}
sessionInfo()
```